// apeman-react-select/shim/node/ap_select_label.js
/**
 * Label component
 * @class ApSelectLabel
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ApSelectLabel = undefined;

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactTouchable = require('apeman-react-touchable');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApSelectLabel */
var ApSelectLabel = function (_Component) {
  (0, _inherits3.default)(ApSelectLabel, _Component);

  function ApSelectLabel() {
    (0, _classCallCheck3.default)(this, ApSelectLabel);
    return (0, _possibleConstructorReturn3.default)(this, (ApSelectLabel.__proto__ || (0, _getPrototypeOf2.default)(ApSelectLabel)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApSelectLabel, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var hasVal = !!props.value;
      var className = (0, _classnames2.default)('ap-select-label-value', {
        'ap-select-label-placeholder': !hasVal
      });
      return _react2.default.createElement(
        'a',
        { className: 'ap-select-label' },
        _react2.default.createElement(
          'span',
          { className: className },
          hasVal ? props.value : props.placeholder
        ),
        _react2.default.createElement(_apemanReactIcon.ApIcon, { className: props.icon })
      );
    }
  }]);
  return ApSelectLabel;
}(_react.Component);

(0, _assign2.default)(ApSelectLabel, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    value: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.element]),
    placeholder: _react.PropTypes.string,
    icon: _react.PropTypes.string
  },

  defaultProps: {
    value: null
  }

});

exports.ApSelectLabel = ApSelectLabel;
exports.default = (0, _apemanReactTouchable.withTouch)(ApSelectLabel, {
  getTouchData: function getTouchData() {
    var s = this;
    var props = s.props;

    return props.data;
  }
});
