// apeman-react-select/shim/node/ap_select_item.js
/**
 * Item component
 * @class ApSelectItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ApSelectItem = undefined;

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _asobj = require('asobj');

var _apemanReactTouchable = require('apeman-react-touchable');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApSelectItem */
var ApSelectItem = function (_Component) {
  (0, _inherits3.default)(ApSelectItem, _Component);

  function ApSelectItem() {
    (0, _classCallCheck3.default)(this, ApSelectItem);
    return (0, _possibleConstructorReturn3.default)(this, (ApSelectItem.__proto__ || (0, _getPrototypeOf2.default)(ApSelectItem)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApSelectItem, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var hasChildren = _react2.default.Children.count(props.children) > 0;
      return _react2.default.createElement(
        'a',
        { className: (0, _classnames2.default)('ap-select-item', {
            'ap-select-item-focused': props.focused
          }) },
        hasChildren ? props.children : props.label
      );
    }

    // --------------------
    // Lifecycle
    // --------------------

  }, {
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      var s = this;
      var props = s.props,
          state = s.state;

      return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
    }
  }]);
  return ApSelectItem;
}(_react.Component);

(0, _assign2.default)(ApSelectItem, {

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Data for event */
    data: _react.PropTypes.string,
    /** Item label text */
    label: _react.PropTypes.string,
    /** Focused or not */
    focused: _react.PropTypes.bool
  },

  defaultProps: {
    label: null,
    data: null,
    focused: false
  }
});

exports.ApSelectItem = ApSelectItem;
exports.default = (0, _apemanReactTouchable.withTouch)(ApSelectItem, {
  getTouchData: function getTouchData() {
    var s = this;
    var props = s.props;

    return props.data;
  }
});
