// apeman-react-done/shim/node/ap_done_style.js
/**
 * Style for ApDone.
 * @class ApDoneStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApDoneStyle */
var ApDoneStyle = function (_Component) {
  _inherits(ApDoneStyle, _Component);

  function ApDoneStyle() {
    _classCallCheck(this, ApDoneStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApDoneStyle).apply(this, arguments));
  }

  _createClass(ApDoneStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApDoneStyle$styleDat = ApDoneStyle.styleData(props);

      var all = _ApDoneStyle$styleDat.all;
      var small = _ApDoneStyle$styleDat.small;
      var medium = _ApDoneStyle$styleDat.medium;
      var large = _ApDoneStyle$styleDat.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApDoneStyle;
}(_react.Component);

Object.assign(ApDoneStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },

  defaultProps: {
    style: {},
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;
    var backgroundColor = config.backgroundColor;

    return {
      all: {
        '.ap-done': {
          textAlign: 'center'
        },
        '.ap-done-button-container': {
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center'
        },
        '.ap-done-button': {
          fontSize: '11px',
          padding: '4px 16px'
        },
        '.ap-done-link': {
          display: 'inline-block',
          padding: '8px 16px',
          color: '' + highlightColor,
          textDecoration: 'underline',
          fontSize: '14px'
        },
        '.ap-done-link:hover': {},
        '.ap-done-link:active': {
          opacity: '0.8'
        }
      }

    };
  }
});

exports.default = ApDoneStyle;
