// apeman-react-done/shim/node/ap_done.js
/**
 * apeman react package for done components.
 * @class ApDone
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactButton = require('apeman-react-button');

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApDone */
var ApDone = function (_Component) {
  _inherits(ApDone, _Component);

  function ApDone() {
    _classCallCheck(this, ApDone);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApDone).apply(this, arguments));
  }

  _createClass(ApDone, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;


      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-done', props.className),
          style: Object.assign({}, props.style) },
        _react2.default.createElement(
          'p',
          { className: 'ap-done-message' },
          props.message
        ),
        props.children,
        _react2.default.createElement(
          'div',
          { className: 'ap-done-button-container' },
          _react2.default.createElement(ApDone.Button, { text: props.subActionText,
            action: props.onSubAction
          }),
          _react2.default.createElement(ApDone.Button, { text: props.actionText,
            action: props.onAction,
            primary: true
          })
        ),
        _react2.default.createElement(
          'div',
          null,
          _react2.default.createElement(ApDone.Link, { text: props.linkText,
            href: props.linkHref
          })
        )
      );
    }

    // --------------------
    // Lifecycle
    // --------------------

  }, {
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      var s = this;
      var props = s.props;
      var state = s.state;

      return !(0, _asobj.shallowEqual)(props, nextProps) || !(0, _asobj.shallowEqual)(state, nextState);
    }
  }]);

  return ApDone;
}(_react.Component);

Object.assign(ApDone, {

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Done message string */
    message: _react.PropTypes.string,
    /** Action button text */
    actionText: _react.PropTypes.string,
    /** Handler for action */
    onAction: _react.PropTypes.func,
    /** Sub action button text */
    subActionText: _react.PropTypes.string,
    /** Handler for sub action */
    onSubAction: _react.PropTypes.func,
    /** Link url */
    link: _react.PropTypes.string,
    /** Text of link */
    linkText: _react.PropTypes.string
  },

  defaultProps: {
    message: null,
    actionText: null,
    onAction: null,
    subActionText: null,
    onSubAction: null,
    linkHref: null,
    linkText: null
  },

  Button: function Button(_ref) {
    var text = _ref.text;
    var action = _ref.action;
    var primary = _ref.primary;

    if (!text) {
      return null;
    }
    return _react2.default.createElement(
      _apemanReactButton.ApButton,
      { className: 'ap-done-button',
        onTap: action,
        primary: primary
      },
      text
    );
  },
  Link: function Link(_ref2) {
    var text = _ref2.text;
    var href = _ref2.href;

    if (!text) {
      return null;
    }
    return _react2.default.createElement(
      'a',
      { href: href, className: 'ap-done-link' },
      text
    );
  }
});

exports.default = ApDone;
