// apeman-react-dialog/shim/node/ap_confirm_dialog_style.js
/**
 * Style for ApConfirmDialog.
 * @class ApConfirmDialogStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApConfirmDialogStyle */
var ApConfirmDialogStyle = function (_Component) {
  _inherits(ApConfirmDialogStyle, _Component);

  function ApConfirmDialogStyle() {
    _classCallCheck(this, ApConfirmDialogStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApConfirmDialogStyle).apply(this, arguments));
  }

  _createClass(ApConfirmDialogStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApConfirmDialogStyle = ApConfirmDialogStyle.styleData(props);

      var all = _ApConfirmDialogStyle.all;
      var small = _ApConfirmDialogStyle.small;
      var medium = _ApConfirmDialogStyle.medium;
      var large = _ApConfirmDialogStyle.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApConfirmDialogStyle;
}(_react.Component);

Object.assign(ApConfirmDialogStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    buttonHeight: _react.PropTypes.number,
    highlightColor: _react.PropTypes.string
  },

  defaultProps: {
    style: {},
    buttonHeight: 32,
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    errorColor: _apemanReactStyle.ApStyle.ERROR_COLOR
  },
  styleData: function styleData(config) {
    var buttonHeight = config.buttonHeight;
    var highlightColor = config.highlightColor;
    var errorColor = config.errorColor;

    var checkboxHeight = 32;
    return {
      all: {
        '.ap-confirm-dialog .ap-dialog-body': {
          paddingBottom: buttonHeight + 8 + 'px',
          textAlign: 'center'
        },
        '.ap-confirm-dialog .ap-dialog-close-button-icon': {},
        '.ap-confirm-dialog-control': {
          display: 'block',
          textAlign: 'center',
          position: 'relative',
          left: 0,
          right: 0,
          bottom: 0,
          backgroundColor: 'rgba(255,255,255,0.9)',
          padding: '16px 0'
        },
        '.ap-confirm-dialog-button': {
          display: 'inline-block',
          width: '80%',
          textAlign: 'center',
          border: '1px solid #EEE',
          margin: '0 auto',
          background: '' + highlightColor,
          color: 'white',
          boxSizing: 'border-box',
          maxWidth: '320px'
        },
        '.ap-confirm-dialog-button-disabled, .ap-confirm-dialog-button-disabled:hover, .ap-confirm-dialog-button-disabled:active': {
          cursor: 'default',
          boxShadow: 'none',
          color: '#888',
          borderColor: '#888',
          backgroundColor: '#F0F0F0'
        },
        '.ap-confirm-dialog-checkbox': {
          margin: '8px auto',
          lineHeight: checkboxHeight + 'px',
          textAlign: 'left'
        },
        '.ap-confirm-dialog-checkbox i:before': {
          lineHeight: checkboxHeight + 'px'
        },
        '.ap-confirm-dialog-message': {
          fontSize: 'smaller',
          padding: '16px 0'
        },
        '.ap-confirm-dialog-err': {
          fontSize: 'smaller',
          textAlign: 'center',
          color: '' + errorColor,
          fontStyle: 'italic'
        }
      },
      small: {},
      medium: {},
      large: {}
    };
  }
});

exports.default = ApConfirmDialogStyle;
