// apeman-react-dialog/shim/node/ap_yesno_dialog.js
/**
 * Yes/No dialog.
 * @class ApYesnoDialog
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactTouchable = require('apeman-react-touchable');

var _ap_dialog = require('./ap_dialog');

var _ap_dialog2 = _interopRequireDefault(_ap_dialog);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApYesnoDialog */
var ApYesnoDialog = function (_Component) {
  _inherits(ApYesnoDialog, _Component);

  function ApYesnoDialog() {
    _classCallCheck(this, ApYesnoDialog);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApYesnoDialog).apply(this, arguments));
  }

  _createClass(ApYesnoDialog, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      if (!props.present) {
        return null;
      }
      return _react2.default.createElement(
        _ap_dialog2.default,
        _extends({ className: (0, _classnames2.default)('ap-yesno-dialog', props.className),
          style: Object.assign({}, props.style)
        }, props),
        _react2.default.createElement(
          'div',
          null,
          props.children
        ),
        _react2.default.createElement(
          'div',
          { className: 'ap-yesno-dialog-control' },
          _react2.default.createElement(ApYesnoDialog.Button, { text: props.noText, onTap: props.onNo }),
          _react2.default.createElement(ApYesnoDialog.Button, { text: props.yesText, onTap: props.onYes })
        )
      );
    }
  }]);

  return ApYesnoDialog;
}(_react.Component);

Object.assign(ApYesnoDialog, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    present: _react.PropTypes.bool.isRequired,
    /** Handler for tapping YES */
    onYes: _react.PropTypes.func,
    /** Handler for tapping NO */
    onNo: _react.PropTypes.func,
    /** Dialog title */
    title: _react.PropTypes.string,
    /** Text of YES button */
    yesText: _react.PropTypes.string,
    /** Text of NO button */
    noText: _react.PropTypes.string
  },

  defaultProps: {
    present: false,
    onYes: null,
    onNo: null,
    title: null,
    yesText: 'Yes',
    noText: 'No'
  },

  // --------------------
  // Sub components
  // --------------------
  Button: function Button(_ref) {
    var text = _ref.text;
    var onTap = _ref.onTap;

    var s = this;
    return _react2.default.createElement(
      'a',
      { className: 'ap-yesno-dialog-button' },
      _react2.default.createElement(
        _apemanReactTouchable.ApTouchable,
        { onTap: onTap },
        _react2.default.createElement(
          'span',
          { className: 'ap-yesno-dialog-button-text' },
          text
        )
      )
    );
  }
});

exports.default = ApYesnoDialog;
