// apeman-react-dialog/shim/node/ap_yesno_dialog_style.js
/**
 * Style for ApYesnoDialog.
 * @class ApYesnoDialogStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApYesnoDialogStyle */
var ApYesnoDialogStyle = function (_Component) {
  _inherits(ApYesnoDialogStyle, _Component);

  function ApYesnoDialogStyle() {
    _classCallCheck(this, ApYesnoDialogStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApYesnoDialogStyle).apply(this, arguments));
  }

  _createClass(ApYesnoDialogStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApYesnoDialogStyle$s = ApYesnoDialogStyle.styleData(props);

      var all = _ApYesnoDialogStyle$s.all;
      var small = _ApYesnoDialogStyle$s.small;
      var medium = _ApYesnoDialogStyle$s.medium;
      var large = _ApYesnoDialogStyle$s.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApYesnoDialogStyle;
}(_react.Component);

Object.assign(ApYesnoDialogStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    buttonHeight: _react.PropTypes.number,
    highlightColor: _react.PropTypes.string
  },
  defaultProps: {
    style: {},
    buttonHeight: 40,
    borderColor: '#D0D0D0',
    highlightColor: '#38E'
  },
  styleData: function styleData(config) {
    var buttonHeight = config.buttonHeight;
    var borderColor = config.borderColor;
    var highlightColor = config.highlightColor;

    return {
      all: {
        '.ap-yesno-dialog .ap-dialog-body': {
          paddingBottom: buttonHeight + 8 + 'px'
        },
        '.ap-yesno-dialog .ap-dialog-close-button-icon': {
          display: 'none'
        },
        '.ap-yesno-dialog-control': {
          display: 'flex',
          position: 'absolute',
          left: 0,
          right: 0,
          bottom: 0,
          boxSizing: 'border-box',
          overflowX: 'hidden',
          justifyContent: 'flex-end',
          padding: '0 8px',
          backgroundColor: 'rgba(255,255,255,0.9)'
        },
        '.ap-yesno-dialog-button': {
          display: 'block',
          maxWidth: '50%',
          textAlign: 'center',
          margin: '0 -1px',
          color: '' + highlightColor
        },
        '.ap-yesno-dialog-button .ap-touchable': {
          padding: '4px 16px'
        },
        '.ap-yesno-dialog-button-text': {
          display: 'inline-block',
          width: '100%',
          boxSizing: 'border-box',
          lineHeight: buttonHeight + 'px'
        },
        '.ap-yesno-dialog-button:hover': {
          cursor: 'pointer',
          opacity: 0.9
        },
        '.ap-yesno-dialog-button:active': {
          opacity: 0.6
        }
      },
      small: {},
      medium: {},
      large: {}
    };
  }
});

exports.default = ApYesnoDialogStyle;
