// apeman-react-dialog/shim/node/with_dialog.js
/**
 * @function withDialog
 */
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _breact = require('breact');

var _asobj = require('asobj');

var _ap_yesno_dialog = require('./ap_yesno_dialog');

var _ap_yesno_dialog2 = _interopRequireDefault(_ap_yesno_dialog);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function withDialog(Component) {
  return (0, _breact.wrap)(Component, {
    getInitialState: function getInitialState() {
      return {
        dialog: null
      };
    },
    render: function render() {
      var s = this;
      var props = s.props;
      var state = s.state;
      var dialog = state.dialog;

      var componentProps = (0, _asobj.clone)(props, {
        without: []
      });
      return _react2.default.createElement(Component, _extends({}, componentProps, {
        askYesNoWithDialog: s.askYesNo,
        dialog: dialog
      }));
    },


    // --------------------
    // Custom
    // --------------------
    askYesNo: function askYesNo(_ref) {
      var title = _ref.title;
      var text = _ref.text;
      var yesText = _ref.yesText;
      var noText = _ref.noText;
      var closeIcon = _ref.closeIcon;

      var s = this;
      return new Promise(function (resolve, reject) {
        var dialog = _react2.default.createElement(
          _ap_yesno_dialog2.default,
          _extends({ title: title, yesText: yesText, noText: noText, closeIcon: closeIcon }, {
            present: true,
            onYes: function onYes() {
              return resolve(true);
            },
            onNo: function onNo() {
              return resolve(false);
            },
            onClose: function onClose() {
              return resolve();
            }
          }),
          text,
          ' '
        );
        s.setState({ dialog: dialog });
      }).then(function (answer) {
        s.setState({ dialog: null });
        return answer;
      }).catch(function (err) {
        s.setState({ dialog: null });
        return Promise.reject(err);
      });
    }
  });
}

exports.default = withDialog;
