// apeman-react-captcha/shim/node/ap_captcha_style.js
/**
 * Style for ApCaptcha.
 * @class ApCaptchaStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApCaptchaStyle */
var ApCaptchaStyle = function (_Component) {
  (0, _inherits3.default)(ApCaptchaStyle, _Component);

  function ApCaptchaStyle() {
    (0, _classCallCheck3.default)(this, ApCaptchaStyle);
    return (0, _possibleConstructorReturn3.default)(this, (ApCaptchaStyle.__proto__ || (0, _getPrototypeOf2.default)(ApCaptchaStyle)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApCaptchaStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApCaptchaStyle$style = ApCaptchaStyle.styleData(props);

      var all = _ApCaptchaStyle$style.all;
      var small = _ApCaptchaStyle$style.small;
      var medium = _ApCaptchaStyle$style.medium;
      var large = _ApCaptchaStyle$style.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: (0, _assign2.default)(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);
  return ApCaptchaStyle;
}(_react.Component);

(0, _assign2.default)(ApCaptchaStyle, {
  propTypes: {
    style: _react.PropTypes.object
  },
  defaultProps: {
    style: {}
  },
  styleData: function styleData(config) {
    return {
      all: {
        '.ap-captcha': {
          display: 'block',
          position: 'relative',
          background: '#FAFAFA',
          padding: '8px',
          boxSizing: 'border-box',
          textAlign: 'center'
        },
        '.ap-captcha-image': {
          background: 'white',
          border: '1px solid #E0E0E0'
        },
        '.ap-captcha-image .ap-image-spinner': {
          background: 'transparent'
        },
        '.ap-captcha-refresh-button': {
          display: 'block',
          textAlign: 'right',
          cursor: 'pointer',
          padding: '4px 8px',
          position: 'absolute',
          fontSize: '12px',
          right: 0,
          bottom: 0,
          background: 'rgba(255,255,255,0.9)'
        },
        '.ap-captcha-refresh-button:active': {
          opacity: 0.8
        },
        '.ap-captcha-spinner': {
          position: 'absolute',
          left: 0,
          top: 0,
          right: 0,
          bottom: 0,
          zIndex: 4,
          background: 'rgba(255,255,255,0.9)',
          color: '#CCC'
        }
      }
    };
  }
});

exports.default = ApCaptchaStyle;
