// apeman-react-captcha/shim/node/ap_captcha.js
/**
 * Captcha component.
 * @class ApCaptcha
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

var _apemanReactImage = require('apeman-react-image');

var _apemanReactSpinner = require('apeman-react-spinner');

var _apemanReactTouchable = require('apeman-react-touchable');

var _breact = require('breact');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApCaptcha */
var ApCaptcha = function (_Component) {
  _inherits(ApCaptcha, _Component);

  function ApCaptcha(props) {
    _classCallCheck(this, ApCaptcha);

    var _this = _possibleConstructorReturn(this, (ApCaptcha.__proto__ || Object.getPrototypeOf(ApCaptcha)).call(this, props));

    var s = _this;
    (0, _breact.autobind)(s);
    return _this;
  }

  _createClass(ApCaptcha, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-captcha', props.className),
          style: Object.assign({}, props.style) },
        _react2.default.createElement(
          'div',
          null,
          _react2.default.createElement(_apemanReactSpinner.ApSpinner, { className: 'ap-captcha-spinner',
            enabled: props.spinning,
            theme: props.spinnerTheme
          }),
          _react2.default.createElement(_apemanReactImage.ApImage, { className: 'ap-captcha-image',
            scale: 'fit',
            src: props.src,
            width: props.imageWidth,
            height: props.imageHeight,
            onLoad: s.handleImageLoad,
            onError: s.handleImageError,
            alt: ''
          })
        ),
        _react2.default.createElement(
          'div',
          null,
          _react2.default.createElement(
            'a',
            { className: 'ap-captcha-refresh-button' },
            _react2.default.createElement(
              _apemanReactTouchable.ApTouchable,
              { onTap: s.handleTap },
              _react2.default.createElement(
                'span',
                null,
                _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-captcha-refresh-icon', props.refreshIcon, {
                    'fa-spin': props.spinning
                  }) }),
                _react2.default.createElement(
                  'span',
                  null,
                  props.refreshText
                )
              )
            )
          )
        )
      );
    }

    // ------------------
    // Custom
    // ------------------

  }, {
    key: 'handleTap',
    value: function handleTap() {
      var s = this;
      var props = s.props;

      if (props.onRefresh) {
        props.onRefresh();
      }
    }
  }, {
    key: 'handleImageLoad',
    value: function handleImageLoad(ev) {
      var s = this;
      var props = s.props;

      if (props.onImageLoad) {
        props.onImageLoad(ev);
      }
    }
  }, {
    key: 'handleImageError',
    value: function handleImageError(err) {
      var s = this;
      var props = s.props;

      if (props.onImageError) {
        props.onImageError(err);
      }
    }
  }]);

  return ApCaptcha;
}(_react.Component);

Object.assign(_react.Component, {
  // --------------------
  // Specs
  // --------------------
  propTypes: {
    /** Image source url */
    src: _react.PropTypes.string,
    /** Handler for refresh */
    onRefresh: _react.PropTypes.func,
    refreshIcon: _react.PropTypes.string,
    refreshText: _react.PropTypes.string,
    imageWidth: _react.PropTypes.number,
    imageHeight: _react.PropTypes.number,
    onImageLoad: _react.PropTypes.func,
    onImageError: _react.PropTypes.func,
    spinning: _react.PropTypes.bool,
    spinnerTheme: _react.PropTypes.string
  },

  defaultProps: {
    src: null,
    onRefresh: null,
    refreshIcon: 'fa fa-refresh',
    refreshText: '',
    imageWidth: 240,
    imageHeight: 94,
    onImageLoad: null,
    onImageError: null,
    spinning: false,
    spinnerTheme: _apemanReactSpinner.ApSpinner.DEFAULT_THEME
  }

});

exports.default = ApCaptcha;
