// apeman-react-page/shim/node/ap_page_container.js
/**
 * Container element for page.
 * @class ApPageContainer
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApPageContainer */
var ApPageContainer = _react2.default.createClass({
  displayName: 'ApPageContainer',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    id: _react.PropTypes.string.isRequired
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-page-container', props.className),
        id: props.id,
        style: Object.assign({}, props.style) },
      props.children
    );
  }
});

exports.default = ApPageContainer;
