// apeman-react-page/shim/node/ap_page_style.js
/**
 * Style for ApPage.
 * @class ApPageStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApPageStyle */
var ApPageStyle = _react2.default.createClass({
  displayName: 'ApPageStyle',

  propTypes: {
    style: _react.PropTypes.object
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {}
    };
  },

  statics: {
    styleData: function styleData(conifg) {
      return {
        all: {
          '.ap-page-container': {
            display: 'block'
          },
          '.ap-page': {
            display: 'block',
            position: 'fixed',
            left: 0,
            top: 0,
            right: 0,
            bottom: 0,
            overflow: 'auto'
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApPageStyle$styleDat = ApPageStyle.styleData(props);

    var all = _ApPageStyle$styleDat.all;
    var small = _ApPageStyle$styleDat.small;
    var medium = _ApPageStyle$styleDat.medium;
    var large = _ApPageStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApPageStyle;
