// apeman-react-catalog/shim/node/ap_catalog.js
/**
 * Catalog component
 * @class ApCatalog
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApCatalog */
var ApCatalog = _react2.default.createClass({
    displayName: 'ApCatalog',


    // --------------------
    // Specs
    // --------------------

    propTypes: {},

    mixins: [],

    statics: {},

    getInitialState: function getInitialState() {
        return {};
    },
    getDefaultProps: function getDefaultProps() {
        return {};
    },
    render: function render() {
        var s = this;
        var state = s.state;
        var props = s.props;


        return _react2.default.createElement(
            'div',
            { className: (0, _classnames2.default)('ap-catalog', props.className),
                style: Object.assign({}, props.style) },
            _react2.default.createElement(
                'div',
                { className: 'ap-catalog-inner' },
                props.children
            )
        );
    }
});

exports.default = ApCatalog;
