// apeman-react-catalog/shim/node/ap_catalog_item_detail.js
/**
 * Detail of a catalog item
 * @class ApCatalogItemDetail
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApCatalogItemDetail */
var ApCatalogItemDetail = _react2.default.createClass({
  displayName: 'ApCatalogItemDetail',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    open: _react.PropTypes.bool,
    at: _react.PropTypes.oneOf(['TOP', 'CENTER', 'LEFT', 'RIGHT', 'BOTTOM'])
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      open: false,
      at: null
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    var at = String(props.at || null).toUpperCase();

    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-catalog-item-detail', {
          'ap-catalog-item-detail-open': props.open,
          'ap-catalog-item-detail-at-left': at === 'LEFT',
          'ap-catalog-item-detail-at-top': at === 'TOP',
          'ap-catalog-item-detail-at-right': at === 'RIGHT',
          'ap-catalog-item-detail-at-bottom': at === 'BOTTOM'
        }, props.className),
        style: Object.assign({}, props.style) },
      props.open ? props.children : null
    );
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApCatalogItemDetail;
