// apeman-react-catalog/shim/node/ap_catalog_style.js
/**
 * Style for ApCatalog.
 * @class ApCatalogStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApCatalogStyle */
var ApCatalogStyle = _react2.default.createClass({
  displayName: 'ApCatalogStyle',

  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR,
      transitionDuration: 500
    };
  },

  statics: {
    styleData: function styleData(config) {
      var highlightColor = config.highlightColor;
      var backgroundColor = config.backgroundColor;
      var transitionDuration = config.transitionDuration;

      return {
        all: {
          '.ap-catalog': {
            display: 'block',
            margin: '0 auto'
          },
          '.ap-catalog-inner': {
            display: 'flex',
            flexFlow: 'row wrap',
            justifyContent: 'flex-start'
          },
          '.ap-menu-inner-end': {
            flex: 'auto',
            display: 'inline-block',
            width: '0px',
            opacity: 0
          },
          '.ap-catalog-item': {
            display: 'block'
          },
          '.ap-catalog-item-summary': {
            padding: '8px 16px',
            border: '1px solid #AAA',
            margin: '-1px'
          },
          '.ap-catalog-item-summary:hover': {
            cursor: 'pointer'
          },
          '.ap-catalog-item-summary:active': {
            opacity: '0.8'
          },
          '.ap-catalog-item-detail': {
            display: 'none',
            position: 'fixed',
            left: 0,
            top: 0,
            width: '100%',
            height: '100%',
            zIndex: 8,
            background: 'white',
            padding: '16px 32px',
            boxSizing: 'border-box'
          },
          '.ap-catalog-item-detail-at-left': {
            left: '-100%'
          },
          '.ap-catalog-item-detail-at-top': {
            top: '-100%'
          },
          '.ap-catalog-item-detail-at-right': {
            left: '100%'
          },
          '.ap-catalog-item-detail-at-bottom': {
            top: '100%'
          },
          '.ap-catalog-item-detail-open': {
            display: 'block',
            transition: 'left ' + transitionDuration + 'ms, top ' + transitionDuration + 'ms'
          }

        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApCatalogStyle$style = ApCatalogStyle.styleData(props);

    var all = _ApCatalogStyle$style.all;
    var small = _ApCatalogStyle$style.small;
    var medium = _ApCatalogStyle$style.medium;
    var large = _ApCatalogStyle$style.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApCatalogStyle;
