// apeman-react-frame/shim/node/ap_frame_style.js
/**
 * Style for ApFrame.
 * @class ApFrameStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

var _acolor = require('acolor');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApFrameStyle */
var ApFrameStyle = _react2.default.createClass({
  displayName: 'ApFrameStyle',

  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
    };
  },

  statics: {
    styleData: function styleData(config) {
      var highlightColor = config.highlightColor;

      return {
        all: {
          '.ap-frame': {
            position: 'relative'
          },
          '.ap-frame-full': {
            position: 'absolute',
            left: 0,
            top: 0,
            right: 0,
            bottom: 0,
            boxSizing: 'border-box',
            overflow: 'hidden'
          },
          '.ap-frame-spinner': {
            width: '100%',
            height: '100%',
            position: 'absolute',
            textAlign: 'center',
            boxSizing: 'content-box',
            zIndex: 4,
            background: _apemanReactStyle.ApStyle.COVER_BACKGROUND,
            color: _apemanReactStyle.ApStyle.COVER_COLOR
          },
          '.ap-frame-frame': {
            width: '100%',
            height: '100%',
            boxSizing: 'content-box'
          },
          '.ap-markable-frame': {
            position: 'relative',
            overflow: 'auto'
          },
          '.ap-markable-frame-markable-item': {
            position: 'absolute',
            background: highlightColor,
            opacity: '0.75',
            boxSizing: 'border-box',
            border: '2px solid ' + (0, _acolor.mix)(highlightColor, '#000'),
            boxShadow: '4px 4px 8px rgba(0,0,0,0.5)',
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center'
          }
        },
        small: {},
        medium: {},
        large: {}
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApFrameStyle$styleDa = ApFrameStyle.styleData(props);

    var all = _ApFrameStyle$styleDa.all;
    var small = _ApFrameStyle$styleDa.small;
    var medium = _ApFrameStyle$styleDa.medium;
    var large = _ApFrameStyle$styleDa.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApFrameStyle;
