// apeman-react-frame/shim/node/ap_frame.js
/**
 * iframe components.
 * @class ApFrame
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactSpinner = require('apeman-react-spinner');

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApFrame */
var ApFrame = _react2.default.createClass({
  displayName: 'ApFrame',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Source URL string */
    src: _react.PropTypes.string,
    /** Theme of spinner */
    spinner: _react.PropTypes.string,
    /** Display with full screen */
    full: _react.PropTypes.bool,
    /** Component width */
    width: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
    /** Component height */
    height: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
    /** Ref to iframe */
    iframeRef: _react.PropTypes.func
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    var s = this;
    var props = s.props;

    return {
      spinning: !!props.src,
      err: null
    };
  },
  getDefaultProps: function getDefaultProps() {
    return {
      src: null,
      full: false,
      spinner: _apemanReactSpinner.ApSpinner.DEFAULT_THEME,
      width: 'auto',
      height: 'auto'
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;

    var className = (0, _classnames2.default)('ap-frame', {
      'ap-frame-full': props.full
    }, props.className);
    var style = Object.assign({
      width: props.width,
      height: props.height
    }, props.style);
    return _react2.default.createElement(
      'div',
      { className: className,
        style: style },
      _react2.default.createElement(_apemanReactSpinner.ApSpinner, { enabled: state.spinning,
        theme: props.spinner,
        className: 'ap-frame-spinner'
      }),
      _react2.default.createElement('iframe', { src: props.src,
        className: 'ap-frame-frame',
        frameBorder: '0',
        onLoad: s.handleLoad,
        ref: props.iframeRef
      }),
      props.children
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props;
    var state = s.state;

    return !(0, _asobj.shallowEqual)(props, nextProps) || !(0, _asobj.shallowEqual)(state, nextState);
  },
  componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
    var s = this;
    var state = s.state;

    var stopSpinning = state.spinning && !nextProps.src;
    if (stopSpinning) {
      s.setState({ spinning: false });
    }
  },


  // ------------------
  // Custom
  // ------------------

  handleLoad: function handleLoad(e) {
    var s = this;
    var props = s.props;

    s.setState({
      spinning: false
    });
    if (props.onLoad) {
      props.onLoad(e);
    }
  }

  // ------------------
  // Private
  // ------------------

});

exports.default = ApFrame;
