// apeman-react-slider/lib/ap_slider_handle.jsx
/**
 * Handle of slider component.
 * @class ApSliderHandle
 */
'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import {ApDraggable} from 'apeman-react-draggable'

/** @lends ApSliderHandle */
const ApSliderHandle = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    onMove: types.func,
    shouldMove: types.func,
    x: types.number,
    minX: types.number,
    maxX: types.number
  },

  render () {
    const s = this,
      {props} = s
    return (
      <ApDraggable onMove={props.onMove}
                   shouldMove={props.shouldMove}
                   x={props.x}
                   minX={props.minX}
                   maxX={props.maxX}
                   className={ classnames('ap-slider-handle', props.className)}
                   direction="HORIZONTAL">
        <div>
          <div className="ap-slider-handle-area"></div>
          <div className="ap-slider-handle-icon"></div>
        </div>
      </ApDraggable>
    )
  }
})

export default ApSliderHandle;
