// apeman-react-button/shim/node/ap_button.js
/**
 * Button component.
 * @class ApButton
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _keys = require('babel-runtime/core-js/object/keys');

var _keys2 = _interopRequireDefault(_keys);

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactTouchable = require('apeman-react-touchable');

var _apemanReactSpinner = require('apeman-react-spinner');

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApButton */
var ApButton = _react2.default.createClass({
  displayName: 'ApButton',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Disable button tap */
    disabled: _react.PropTypes.bool,
    /** Render with primary style */
    primary: _react.PropTypes.bool,
    /** Render with danger style */
    danger: _react.PropTypes.bool,
    /** Render with wide style */
    wide: _react.PropTypes.bool,
    /** Anchor href */
    href: _react.PropTypes.string,
    /** Document id */
    id: _react.PropTypes.string,
    /** Hide button */
    hidden: _react.PropTypes.bool,
    /** Render with simple style */
    simple: _react.PropTypes.bool,
    /** Data for touch events */
    data: _react.PropTypes.any,
    /** Show spinner */
    spinning: _react.PropTypes.bool,
    /** Theme of spinner */
    spinner: _react.PropTypes.string
  },

  mixins: [],

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      disabled: false,
      primary: false,
      danger: false,
      wide: false,
      href: 'javascript:void(0)',
      id: null,
      hidden: false,
      simple: false,
      data: null,
      spinning: false,
      spinner: _apemanReactSpinner.ApSpinner.DEFAULT_THEME
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;


    var className = (0, _classnames2.default)('ap-button', props.className, {
      'ap-button-primary': props.primary,
      'ap-button-danger': props.danger,
      'ap-button-wide': props.wide,
      'ap-button-disabled': props.disabled,
      'ap-button-simple': props.simple,
      'ap-button-spinning': props.spinning,
      'ap-button-hidden': props.hidden
    });
    var aProps = (0, _asobj.clone)(props, {
      without: ['text', 'hasTouch'].concat((0, _toConsumableArray3.default)((0, _keys2.default)(ApButton.propTypes)))
    });
    return _react2.default.createElement(
      'a',
      (0, _extends3.default)({}, aProps, {
        className: className,
        href: props.href,
        id: props.id,
        style: (0, _assign2.default)({}, props.style)
      }),
      props.children,
      props.spinning ? s._renderSpinner() : null
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props,
        state = s.state;

    return !(0, _asobj.shallowEqual)(props, nextProps) || !(0, _asobj.shallowEqual)(state, nextState);
  },


  // --------------------
  // Private
  // --------------------

  _renderSpinner: function _renderSpinner() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(_apemanReactSpinner.ApSpinner, { enabled: true,
      theme: props.spinner,
      className: 'ap-button-spinner'
    });
  }
});

exports.default = (0, _apemanReactTouchable.withTouch)(ApButton, {
  getTouchData: function getTouchData() {
    var s = this;
    var props = s.props;

    return props.data;
  }
});
