// apeman-react-button/shim/node/ap_button_group.js
/**
 * Button group component.
 * @class ApButtonGroup
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApButtonGroup */
var ApButtonGroup = _react2.default.createClass({
  displayName: 'ApButtonGroup',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Centered */
    centered: _react.PropTypes.bool
  },

  mixins: [],

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var props = s.props;


    var className = (0, _classnames2.default)('ap-button-group', {
      'ap-button-group-centered': props.centered
    }, props.className);
    return _react2.default.createElement(
      'div',
      { className: className,
        style: (0, _assign2.default)({}, props.style)
      },
      props.children
    );
  }
});

exports.default = ApButtonGroup;
