// apeman-react-body/shim/node/ap_body_style.js
/**
 * Style for ApBody.
 * @class ApBodyStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApBodyStyle */
var ApBodyStyle = _react2.default.createClass({
  displayName: 'ApBodyStyle',

  propTypes: {
    style: _react.PropTypes.object,
    color: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      color: '#555'
    };
  },

  statics: {
    styleData: function styleData(config) {
      var color = config.color;


      return {
        all: {
          '.ap-body': {
            margin: 0,
            padding: 0,
            fontFamily: 'ヒラギノ角ゴ ProN W3, Hiragino Kaku Gothic ProN, メイリオ, Meiryo, sans-serif',
            color: '' + color
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApBodyStyle$styleDat = ApBodyStyle.styleData(props);

    var all = _ApBodyStyle$styleDat.all;
    var small = _ApBodyStyle$styleDat.small;
    var medium = _ApBodyStyle$styleDat.medium;
    var large = _ApBodyStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApBodyStyle;
