// apeman-react-checkbox/shim/node/ap_checkbox.js
/**
 * Checkbox component.
 * @class ApCheckbox
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var DEFAULT_ICON = 'ion ion-android-checkbox-outline-blank';
var DEFAULT_CHECKED_ICON = 'ion ion-android-checkbox';

/** @lends ApCheckbox */

var ApCheckbox = function (_Component) {
  _inherits(ApCheckbox, _Component);

  function ApCheckbox() {
    _classCallCheck(this, ApCheckbox);

    return _possibleConstructorReturn(this, (ApCheckbox.__proto__ || Object.getPrototypeOf(ApCheckbox)).apply(this, arguments));
  }

  _createClass(ApCheckbox, [{
    key: 'render',
    value: function render() {
      var _iconClass;

      var s = this;
      var props = s.props;
      var id = props.id;
      var name = props.name;
      var value = props.value;
      var checked = props.checked;
      var onChange = props.onChange;


      var iconClass = (_iconClass = {
        'ap-checkbox-icon-checked': checked
      }, _defineProperty(_iconClass, props.checkedIcon, checked), _defineProperty(_iconClass, props.icon, !checked), _iconClass);

      var className = (0, _classnames2.default)('ap-checkbox', props.className, {
        'ap-checkbox-checked': checked
      });
      return _react2.default.createElement(
        'label',
        { className: className,
          htmlFor: id,
          id: id + '-wrap' },
        _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-checkbox-icon', iconClass) }),
        _react2.default.createElement('input', { type: 'checkbox',
          className: (0, _classnames2.default)('ap-checkbox-input'),
          id: id,
          name: name,
          value: value,
          checked: checked,
          onChange: onChange }),
        _react2.default.createElement(
          'span',
          { className: (0, _classnames2.default)('ap-checkbox-title') },
          props.title
        )
      );
    }
  }]);

  return ApCheckbox;
}(_react.Component);

Object.assign(ApCheckbox, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Document id */
    id: _react.PropTypes.string.isRequired,
    /** Name of checkbox input */
    name: _react.PropTypes.string.isRequired,
    /** Value of checkbox input */
    value: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.bool]).isRequired,
    /** Label title */
    title: _react.PropTypes.node,
    /** Checked or not */
    checked: _react.PropTypes.bool,
    /** Handle for change event */
    onChange: _react.PropTypes.func,
    /** Icon class name for normal state */
    icon: _react.PropTypes.string,
    /** Icon class name for checked state */
    checkedIcon: _react.PropTypes.string
  },

  defaultProps: {
    id: null,
    name: null,
    value: '',
    title: '',
    checked: false,
    onChange: null,
    icon: DEFAULT_ICON,
    checkedIcon: DEFAULT_CHECKED_ICON
  },

  DEFAULT_ICON: DEFAULT_ICON,
  DEFAULT_CHECKED_ICON: DEFAULT_CHECKED_ICON

});

exports.default = ApCheckbox;
