// apeman-react-checkbox/shim/node/ap_checkbox_style.js
/**
 * Style for ApCheckbox.
 * @class ApCheckboxStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApCheckboxStyle */
var ApCheckboxStyle = function (_Component) {
  _inherits(ApCheckboxStyle, _Component);

  function ApCheckboxStyle() {
    _classCallCheck(this, ApCheckboxStyle);

    return _possibleConstructorReturn(this, (ApCheckboxStyle.__proto__ || Object.getPrototypeOf(ApCheckboxStyle)).apply(this, arguments));
  }

  _createClass(ApCheckboxStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApCheckboxStyle$styl = ApCheckboxStyle.styleData(props);

      var all = _ApCheckboxStyle$styl.all;
      var small = _ApCheckboxStyle$styl.small;
      var medium = _ApCheckboxStyle$styl.medium;
      var large = _ApCheckboxStyle$styl.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApCheckboxStyle;
}(_react.Component);

Object.assign(ApCheckboxStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },

  defaultProps: {
    style: {},
    highlightColor: '#38E'
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;


    return {
      all: {
        '.ap-checkbox': {
          display: 'inline-block',
          padding: '0 4px',
          cursor: 'pointer',
          position: 'relative'
        },
        '.ap-checkbox:hover': {
          opacity: 0.9
        },
        '.ap-checkbox:active': {
          opacity: 0.75
        },
        '.ap-checkbox-input': {
          display: 'inline-block',
          padding: '0 2px',
          position: 'relative',
          top: '-2px',
          opacity: 0,
          zIndex: 8
        },
        '.ap-checkbox-title': {},
        '.ap-checkbox-icon': {
          position: 'absolute',
          left: 2,
          top: 0,
          bottom: 0,
          display: 'inline-flex',
          justifyContent: 'center',
          alignItems: 'center'
        },
        '.ap-checkbox:hover .ap-checkbox-icon': {
          color: '' + highlightColor
        },
        '.ap-checkbox:active .ap-checkbox-icon': {
          opacity: 0.5
        },
        '.ap-checkbox-icon-checked': {
          color: '' + highlightColor
        }
      },
      small: {},
      medium: {},
      large: {}
    };
  }
});

exports.default = ApCheckboxStyle;
