// apeman-react-checkbox/shim/node/ap_checkbox_group.js
/**
 * Checkbox component.
 * @class ApCheckbox
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _uuid = require('uuid');

var _uuid2 = _interopRequireDefault(_uuid);

var _ap_checkbox = require('./ap_checkbox');

var _ap_checkbox2 = _interopRequireDefault(_ap_checkbox);

var _breact = require('breact');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApCheckboxGroup */
var ApCheckboxGroup = function (_Component) {
  _inherits(ApCheckboxGroup, _Component);

  function ApCheckboxGroup(props) {
    _classCallCheck(this, ApCheckboxGroup);

    var _this = _possibleConstructorReturn(this, (ApCheckboxGroup.__proto__ || Object.getPrototypeOf(ApCheckboxGroup)).call(this, props));

    var s = _this;
    s.uuid = _uuid2.default.v4();
    (0, _breact.autobind)(s);
    return _this;
  }

  _createClass(ApCheckboxGroup, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;
      var _props$prefix = props.prefix;
      var prefix = _props$prefix === undefined ? s.uuid : _props$prefix;
      var name = props.name;
      var options = props.options;
      var checked = props.checked;
      var icon = props.icon;
      var checkedIcon = props.checkedIcon;


      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-checkbox-group', props.className) },
        Object.keys(options || {}).map(function (value) {
          return _react2.default.createElement(_ap_checkbox2.default, { key: value,
            name: name,
            value: value,
            id: prefix + '-' + value,
            checked: !!checked[value],
            title: options[value],
            icon: icon,
            checkedIcon: checkedIcon,
            onChange: s.handleChange
          });
        })
      );
    }

    // --------------------
    // Handle
    // --------------------

  }, {
    key: 'handleChange',
    value: function handleChange(e) {
      var s = this;
      var props = s.props;

      if (props.onChange) {
        props.onChange(e);
      }
    }
  }]);

  return ApCheckboxGroup;
}(_react.Component);

Object.assign(ApCheckboxGroup, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Document id prefix */
    prefix: _react.PropTypes.string,
    /** Name of checkbox input */
    name: _react.PropTypes.string.isRequired,
    /** Value and label titles */
    options: _react.PropTypes.object.isRequired,
    /** Checked state for each values */
    checked: _react.PropTypes.object.isRequired,
    /** Handle for change event */
    onChange: _react.PropTypes.func,
    /** Icon class name for normal state */
    icon: _react.PropTypes.string,
    /** Icon class name for checked state */
    checkedIcon: _react.PropTypes.string
  },

  defaultProps: {
    prefix: null,
    name: null,
    title: '',
    checked: {},
    options: {},
    onChange: null,
    icon: _ap_checkbox2.default.DEFAULT_ICON,
    checkedIcon: _ap_checkbox2.default.DEFAULT_CHECKED_ICON
  }

});

exports.default = ApCheckboxGroup;
