// apeman-react-label/lib/ap_label_style.jsx
/**
 * Style for ApLabel.
 * @class ApLabelStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApLabelStyle */
const ApLabelStyle = React.createClass({
  propTypes: {
    style: types.object
  },
  getDefaultProps () {
    return {
      style: {}
    }
  },
  statics: {
    styleData (config) {
      let {
        highlightColor,
        backgroundColor
      } = config

      return {
        all: {
          '.ap-label': {
            display: 'inline-block',
            padding: '4px 8px',
            fontSize: '14px',
            color: '#888'
          }
        },
        small: {},
        medium: {},
        large: {}
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApLabelStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApLabelStyle
