// react-flashcard-app/lib/containers/control_buttons/reset_button.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _redux = require('redux');

var _actions = require('../../actions');

var _componentsAction_button = require('../../components/action_button');

var _componentsAction_button2 = _interopRequireDefault(_componentsAction_button);

var ResetButton = function ResetButton(props) {
  return _react2['default'].createElement(_componentsAction_button2['default'], {
    disabled: props.removed.length < 1,
    onSubmit: props.reset,
    className: 'reset',
    text: props.text
  });
};

ResetButton.propTypes = {
  text: _propTypes2['default'].string,
  removed: _propTypes2['default'].arrayOf(_propTypes2['default'].string).isRequired,
  reset: _propTypes2['default'].func.isRequired
};

ResetButton.defaultProps = {
  text: 'reset'
};

var mapStateToProps = function mapStateToProps(state) {
  return {
    removed: state.app.removed,
    total: state.app.deck.length - state.app.removed.length
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return (0, _redux.bindActionCreators)({ reset: _actions.reset }, dispatch);
};

exports['default'] = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(ResetButton);
module.exports = exports['default'];
