// react-flashcard-app/lib/containers/control_buttons/undo_remove_button.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _redux = require('redux');

var _actions = require('../../actions');

var _componentsAction_button = require('../../components/action_button');

var _componentsAction_button2 = _interopRequireDefault(_componentsAction_button);

var UndoRemoveButton = function UndoRemoveButton(props) {
  return _react2['default'].createElement(_componentsAction_button2['default'], {
    disabled: props.removed.length < 1,
    onSubmit: props.undoRemove,
    className: 'undoRemove',
    text: props.text
  });
};

UndoRemoveButton.propTypes = {
  text: _propTypes2['default'].string,
  undoRemove: _propTypes2['default'].func.isRequired,
  removed: _propTypes2['default'].arrayOf(_propTypes2['default'].number).isRequired
};

UndoRemoveButton.defaultProps = {
  text: 'undo'
};

var mapStateToProps = function mapStateToProps(state) {
  return {
    removed: state.app.removed
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return (0, _redux.bindActionCreators)({ undoRemove: _actions.undoRemove }, dispatch);
};

exports['default'] = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(UndoRemoveButton);
module.exports = exports['default'];
