// apeman-react-table/lib/ap_table_header.jsx
/**
 * Table header component
 * @class ApTableHeader
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import {ApPureMixin} from 'apeman-react-mixin-pure'
import {withTouch} from 'apeman-react-touchable'
import {ApIcon} from 'apeman-react-icon'

/** @lends ApTableHeader */
const ApTableHeader = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Icon class name */
    icon: types.string,
    /** Handler for tap event */
    onTap: types.func
  },

  mixins: [
    ApPureMixin
  ],

  statics: {},

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {}
  },

  render () {
    const s = this
    let { state, props } = s

    let className = classnames('ap-table-cell', 'ap-table-header', {
      'ap-table-header-tappable': !!props.onTap
    }, props.className)
    return (
      <th className={ className }
          style={ Object.assign({}, props.style) }>
        <ApIcon className={ classnames('ap-table-header-icon', props.icon) }/>
        { props.children }
      </th>
    )
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
})

export default ApTableHeader // Dummy for doc
export default withTouch(ApTableHeader)
