// apeman-react-menu/shim/node/ap_menu_item.js
/**
 * Item for menu.
 * @class ApMenuItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _asobj = require('asobj');

var _apemanReactButton = require('apeman-react-button');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApMenuItem */
var ApMenuItem = _react2.default.createClass({
  displayName: 'ApMenuItem',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    hidden: _react.PropTypes.bool
  },

  mixins: [],

  statics: {
    /**
     * Create an item.
     * @param {string} icon - Icon class name.
     * @param {string} text - Text of item.
     * @param {function} onTap - On tap.
     * @param {object} props - Properties.
     */
    createItem: function createItem(icon, text, onTap, props) {
      return _react2.default.createElement(ApMenuItem, _extends({ icon: icon,
        text: text,
        onTap: onTap
      }, props));
    }
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      hidden: false
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;


    var className = (0, _classnames2.default)('ap-menu-item', {
      'ap-menu-item-hidden': props.hidden
    }, props.className);
    return _react2.default.createElement(
      _apemanReactButton.ApIconButton,
      _extends({}, props, {
        className: className,
        style: Object.assign({}, props.style)
      }),
      props.children
    );
  },


  // --------------------
  // Lifecycle
  // --------------------
  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props;
    var state = s.state;

    return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
  }

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApMenuItem;
