// apeman-react-menu/shim/node/ap_menu_style.js
/**
 * Style for ApMenu.
 * @class ApMenuStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApMenuStyle */
var ApMenuStyle = _react2.default.createClass({
  displayName: 'ApMenuStyle',

  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
    };
  },

  statics: {
    styleData: function styleData(config) {
      var highlightColor = config.highlightColor;
      var backgroundColor = config.backgroundColor;


      return {
        all: {
          '.ap-menu': {
            display: 'flex',
            flexFlow: 'row wrap',
            justifyContent: 'flex-start',
            maxWidth: _apemanReactStyle.ApStyle.CONTENT_WIDTH + 'px',
            margin: '0 auto'
          },
          '.ap-menu-item': {
            border: 'none',
            backgroundColor: 'transparent',
            minWidth: '86px',
            textAlign: 'center',
            color: 'inherit'
          },
          '.ap-menu-item-hidden': {
            display: 'none !important'
          },
          '.ap-menu-item:hover': {
            cursor: 'pointer',
            opacity: 0.9
          },
          '.ap-menu-item:active': {
            boxShadow: 'none',
            opacity: 0.66
          },
          '.ap-menu-end': {
            flex: 'auto',
            display: 'inline-block',
            width: '0px',
            opacity: 0
          }
        },
        small: {},
        medium: {},
        large: {}
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApMenuStyle$styleDat = ApMenuStyle.styleData(props);

    var all = _ApMenuStyle$styleDat.all;
    var small = _ApMenuStyle$styleDat.small;
    var medium = _ApMenuStyle$styleDat.medium;
    var large = _ApMenuStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApMenuStyle;
