// apeman-react-menu/shim/node/ap_menu.js
/**
 * apeman react package for menu components
 * @class ApMenu
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApMenu */
var ApMenu = _react2.default.createClass({
  displayName: 'ApMenu',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {
    SIGNOUT_ICON: 'fa fa-sign-out',
    ABOUT_ICON: 'fa fa-info-circle',
    SUPPORT: 'fa fa-life-ring',
    TAG: 'fa fa-tag',
    SETTING: 'fa fa-cog'
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var props = s.props;


    var className = (0, _classnames2.default)('ap-menu', props.className);
    return _react2.default.createElement(
      'div',
      { className: className,
        style: Object.assign({}, props.style) },
      props.children,
      _react2.default.createElement('span', { className: 'ap-menu-end' })
    );
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApMenu;
