// apeman-react-notice/shim/node/ap_notice.js
/**
 * Notice component
 * @class ApNotice
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactMixinLayout = require('apeman-react-mixin-layout');

var _apemanReactSpinner = require('apeman-react-spinner');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApNotice */
var ApNotice = _react2.default.createClass({
  displayName: 'ApNotice',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Show spinner */
    spinning: _react.PropTypes.bool,
    /** Spinner theme */
    spinner: _react.PropTypes.string
  },

  mixins: [_apemanReactMixinLayout.ApLayoutMixin],

  statics: {},

  getInitialState: function getInitialState() {
    return {
      height: 'auto'
    };
  },
  getDefaultProps: function getDefaultProps() {
    return {
      spinning: false,
      spinner: _apemanReactSpinner.ApSpinner.DEFAULT_THEME
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var layouts = s.layouts;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-notice', props.className),
        style: Object.assign({}, layouts.notice, props.style) },
      _react2.default.createElement(_apemanReactSpinner.ApSpinner, { className: 'ap-notice-spinner',
        style: layouts.spinner,
        enabled: props.spinning
      }),
      _react2.default.createElement(
        'div',
        { className: 'ap-notice-inner',
          ref: 'inner' },
        props.children
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  // --------------------
  // For ApLayoutMixin
  // --------------------

  getInitialLayouts: function getInitialLayouts() {
    return {
      spinner: null,
      notice: null
    };
  },
  calcLayouts: function calcLayouts() {
    var s = this;
    var props = s.props;

    var inner = _reactDom2.default.findDOMNode(s.refs['inner']);
    var innerRect = inner.getBoundingClientRect();
    return {
      notice: {
        height: innerRect.height
      },
      spinner: {
        height: innerRect.height
      }
    };
  }

  // ------------------
  // Private
  // ------------------

});

exports.default = ApNotice;
