// apeman-react-section/shim/node/ap_section.js
/**
 * Section component.
 * @class ApSection
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _uuid = require('uuid');

var _uuid2 = _interopRequireDefault(_uuid);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApSection */
var ApSection = _react2.default.createClass({
  displayName: 'ApSection',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    id: _react.PropTypes.string
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      id: 'ap-section-' + _uuid2.default.v4()
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var id = props.id;

    return _react2.default.createElement(
      'section',
      { className: (0, _classnames2.default)('ap-section', props.className),
        id: id,
        style: Object.assign({}, props.style) },
      props.children
    );
  }
});

exports.default = ApSection;
