// apeman-react-section/shim/node/ap_section_header_button.js
/**
 * Button for a header section.
 * @class ApSectionHeaderButton
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _asobj = require('asobj');

var _apemanReactTouchable = require('apeman-react-touchable');

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApSectionHeaderButton */
var ApSectionHeaderButton = _react2.default.createClass({
  displayName: 'ApSectionHeaderButton',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    icon: _react.PropTypes.string,
    text: _react.PropTypes.string
  },

  mixins: [],

  statics: {
    EDIT_ICON: 'fa fa-edit'
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      icon: null,
      text: null
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;


    var className = (0, _classnames2.default)('ap-section-header-button', {
      'ap-section-header-lined': props.lined
    }, props.className);
    return _react2.default.createElement(
      'a',
      { id: props.id,
        title: props.title,
        name: props.name,
        href: props.href,
        className: className,
        style: Object.assign({}, props.style)
      },
      _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)(props.icon) }),
      _react2.default.createElement(
        'span',
        { className: 'apeman-section-header-button-text' },
        props.text
      ),
      props.children
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props;
    var state = s.state;

    return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
  }

  // --------------------
  // Private
  // --------------------

});

exports.default = ApSectionHeaderButton; // Dummy export for doc parser

exports.default = (0, _apemanReactTouchable.withTouch)(ApSectionHeaderButton);
