// apeman-react-accordion/shim/node/ap_accordion_header.js
/**
 * Header for accordion.
 * @class ApAccordionHeader
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactTouchable = require('apeman-react-touchable');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApAccordionHeader */
var ApAccordionHeader = _react2.default.createClass({
  displayName: 'ApAccordionHeader',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    onToggle: _react.PropTypes.func,
    busyDuration: _react.PropTypes.number
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {
      busy: false
    };
  },
  getDefaultProps: function getDefaultProps() {
    return {
      onToggle: null,
      busyDuration: 500
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      _apemanReactTouchable.ApTouchable,
      { onTap: s.handleTap },
      _react2.default.createElement(
        'div',
        { className: 'ap-accordion-header' },
        props.children
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentWillUnmount: function componentWillUnmount() {
    var s = this;

    if (s.state.busyTimer) {
      clearTimeout(s.state.busyTimer);
    }
  },


  // ------------------
  // Helper
  // ------------------

  handleTap: function handleTap() {
    var s = this;
    var state = s.state;
    var props = s.props;

    if (state.busy) {
      return; // Prevent double tap.
    }
    s.setState({
      busy: true,
      busyTimer: setTimeout(function () {
        s.setState({
          busy: false,
          busyTimer: null
        });
      }, props.busyDuration)
    });

    if (props.onToggle) {
      props.onToggle();
    }
  }
});

exports.default = ApAccordionHeader;
