// apeman-react-accordion/shim/node/ap_accordion_style.js
/**
 * Style for ApAccordion.
 * @class ApAccordionStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApAccordionStyle */
var ApAccordionStyle = function (_Component) {
  _inherits(ApAccordionStyle, _Component);

  function ApAccordionStyle() {
    _classCallCheck(this, ApAccordionStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApAccordionStyle).apply(this, arguments));
  }

  _createClass(ApAccordionStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApAccordionStyle$sty = ApAccordionStyle.styleData(props);

      var all = _ApAccordionStyle$sty.all;
      var small = _ApAccordionStyle$sty.small;
      var medium = _ApAccordionStyle$sty.medium;
      var large = _ApAccordionStyle$sty.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApAccordionStyle;
}(_react.Component);

Object.assign(ApAccordionStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    headerHeight: _react.PropTypes.number,
    animationDuration: _react.PropTypes.number,
    borderRadius: _react.PropTypes.number,
    borderColor: _react.PropTypes.string
  },

  defaultProps: {
    style: {},
    headerHeight: 44,
    animationDuration: 300,
    borderRadius: 4,
    borderColor: '#AAA'
  },

  styleData: function styleData(config) {
    var headerHeight = config.headerHeight;
    var animationDuration = config.animationDuration;
    var borderRadius = config.borderRadius;
    var borderColor = config.borderColor;

    return {
      all: {
        '.ap-accordion': {
          position: 'relative',
          overflow: 'hidden',
          transition: 'max-height ' + animationDuration + 'ms',
          border: '1px solid ' + borderColor,
          background: 'white',
          boxSizing: 'border-box',
          borderRadius: borderRadius + 'px'
        },
        '.ap-accordion-header': {
          padding: '0 8px',
          lineHeight: headerHeight + 'px',
          height: headerHeight + 1 + 'px',
          cursor: 'pointer',
          boxSizing: 'border-box',
          margin: '-1px',
          borderRadius: '0 0 ' + borderRadius + 'px ' + borderRadius + 'px',
          border: '1px solid ' + borderColor
        },
        '.ap-accordion-header:hover': {
          opacity: 0.9
        },
        '.ap-accordion-header:active': {
          opacity: 0.66
        },
        '.ap-accordion-body': {
          padding: '8px'
        },
        '.ap-accordion-arrow': {
          display: 'inline-block',
          marginRight: '2px',
          transform: 'rotate(0deg)',
          transition: 'transform ' + animationDuration + 'ms'
        },
        '.ap-accordion-arrow-icon': {},
        '.ap-accordion-closed': {
          maxHeight: headerHeight + 'px !important'
        },
        '.ap-accordion-closed .ap-accordion-arrow': {
          transform: 'rotate(-90deg)'
        }
      }
    };
  }
});

exports.default = ApAccordionStyle;
