// apeman-react-accordion/shim/node/ap_accordion_arrow.js
/**
 * Body for accordion.
 * @class ApAccordionArrow
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApAccordionArrow */
var ApAccordionArrow = _react2.default.createClass({
  displayName: 'ApAccordionArrow',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    icon: _react.PropTypes.string
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    var s = this;
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      icon: 'fa fa-chevron-down'
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      'div',
      { className: 'ap-accordion-arrow' },
      _react2.default.createElement('i', { className: (0, _classnames2.default)('ap-accordion-arrow-icon', props.icon) })
    );
  }
});

exports.default = ApAccordionArrow;
