// apeman-react-accordion/shim/node/ap_accordion.js
/**
 * apeman react package for accordion components.
 * @class ApAccordion
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApAccordion */
var ApAccordion = _react2.default.createClass({
  displayName: 'ApAccordion',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Is open or not */
    open: _react.PropTypes.bool
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    var s = this;
    return {
      innerHeight: 1200
    };
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;

    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-accordion', {
          'ap-accordion-closed': !props.open
        }), style: { maxHeight: state.innerHeight } },
      _react2.default.createElement(
        'div',
        { className: 'ap-accordion-inner',
          ref: function ref(inner) {
            return s.registerInner(inner);
          } },
        props.children
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentWillMount: function componentWillMount() {
    var s = this;
  },
  componentDidMount: function componentDidMount() {
    var s = this;
    window.addEventListener('resize', s.handleResize);
  },
  componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
    var s = this;
    s.setState({
      innerHeight: s.getInnerHeight()
    });
  },
  componentWillUnmount: function componentWillUnmount() {
    var s = this;
    window.removeEventListener('resize', s.handleResize);
  },


  // ------------------
  // Helpers
  // ------------------

  handleResize: function handleResize() {
    var s = this;
    s.setState({
      innerHeight: s.getInnerHeight()
    });
  },
  getInnerHeight: function getInnerHeight() {
    var s = this;
    var inner = s.inner;

    return inner.offsetHeight;
  },


  // ------------------
  // Custom
  // ---------
  // ---------
  inner: null,

  registerInner: function registerInner(inner) {
    var s = this;
    s.inner = inner;
  }
});

exports.default = ApAccordion;
