// apeman-react-accordion/shim/node/ap_accordion_body.js
/**
 * Body for accordion.
 * @class ApAccordionBody
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApAccordionBody */
var ApAccordionBody = _react2.default.createClass({
  displayName: 'ApAccordionBody',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    var s = this;
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var props = s.props;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-accordion-body', props.className) },
      props.children
    );
  }
});

exports.default = ApAccordionBody;
