// apeman-react-toggle/lib/ap_toggle_style.jsx
/**
 * Style for ApToggle.
 * @class ApToggleStyle
 */

'use strict'

import React, {Component, PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApToggleStyle */
class ApToggleStyle extends Component {
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApToggleStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
}

Object.assign(ApToggleStyle, {
  propTypes: {
    style: types.object,
    highlightColor: types.string
  },
  defaultProps: {
    style: {},
    highlightColor: ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData (config) {
    let { highlightColor, backgroundColor } = config

    let borderRadius = 4
    return {
      all: {
        '.ap-toggle': {
          borderRadius: `${borderRadius}px`
        },
        '.ap-toggle-item': {
          color: `${highlightColor}`,
          border: `1px solid ${highlightColor}`,
          margin: '0 -1px 0 0',
          background: 'transparent'
        },
        '.ap-toggle-item:first-child': {
          borderRadius: `${borderRadius}px 0 0 ${borderRadius}px`
        },
        '.ap-toggle-item:last-child': {
          borderRadius: `0 ${borderRadius}px ${borderRadius}px 0`
        },
        '.ap-toggle-item-active': {
          background: `${highlightColor}`,
          color: 'white'
        }
      }
    }
  }
})

export default ApToggleStyle
