// apeman-react-toggle/lib/ap_toggle.jsx
/**
 * Toggle item
 * @class ApToggle
 */

'use strict'

import React, {Component, PropTypes as types} from 'react'
import classnames from 'classnames'
import ApToggleItem from './ap_toggle_item'

/** @lends ApToggle */
class ApToggle extends Component {
  render () {
    const s = this
    let { state, props } = s

    let { options } = props
    return (
      <div className={ classnames('ap-toggle', props.className) }
           style={ Object.assign({}, props.style) }>
        {
          Object.keys(options || {}).map((value) =>
            <ApToggleItem active={ value === props.value }
                          key={ value }
                          data={ value }
                          onTap={ props.onToggle }
            >
              { options[ value ] }
            </ApToggleItem>
          )
        }
      </div>
    )
  }
}

Object.assign(ApToggle, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    options: types.object.isRequired,
    value: types.node.isRequired,
    onToggle: types.func
  },

  defaultProps: {
    options: null,
    value: null,
    onToggle: null
  }
})

export default ApToggle
