// apeman-react-password/shim/node/ap_password_style.js
/**
 * Style for ApPassword.
 * @class ApPasswordStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApPasswordStyle */
var ApPasswordStyle = _react2.default.createClass({
  displayName: 'ApPasswordStyle',

  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string,
    maxWidth: _react.PropTypes.number
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      maxWidth: _apemanReactStyle.ApStyle.CONTENT_WIDTH,
      highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR
    };
  },

  statics: {
    styleData: function styleData(config) {
      var highlightColor = config.highlightColor;
      var maxWidth = config.maxWidth;


      return {
        all: {
          '.ap-password': {
            display: 'block',
            padding: '4px 8px',
            border: '1px solid #AAA',
            width: '100%',
            maxWidth: maxWidth + 'px',
            borderRadius: '2px',
            boxSizing: 'border-box',
            outlineColor: '' + highlightColor,
            boxShadow: '1px 1px 1px rgba(0,0,0,.05) inset'
          },
          '.ap-password-wrap': {
            position: 'relative',
            width: '100%',
            margin: '4px',
            verticalAlign: 'middle',
            maxWidth: maxWidth + 'px',
            display: 'block'
          },
          '.ap-password-wrap-empty .ap-password-toggle-button': {
            opacity: 0,
            pointerEvents: 'none'
          },
          '.ap-password-toggle-button': {
            position: 'absolute',
            top: 0,
            right: -4,
            display: 'inline-flex',
            alignItems: 'center',
            padding: '0 8px',
            bottom: 0,
            cursor: 'pointer',
            opacity: 0.8,
            transition: 'opacity 200ms'
          },
          '.ap-password-toggle-button:hover': {
            opacity: 1
          },
          '.ap-password-toggle-button:active': {
            opacity: 0.9
          },
          '.ap-password-toggle-aligner': {
            display: 'inline-block',
            verticalAlign: 'middle',
            width: '1px',
            marginRight: '-1px',
            height: '100%',
            boxSizing: 'border-box',
            opacity: 0
          },
          '.ap-password-toggle-icon': {
            display: 'inline-block',
            verticalAlign: 'middle'
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApPasswordStyle$styl = ApPasswordStyle.styleData(props);

    var all = _ApPasswordStyle$styl.all;
    var small = _ApPasswordStyle$styl.small;
    var medium = _ApPasswordStyle$styl.medium;
    var large = _ApPasswordStyle$styl.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApPasswordStyle;
