// apeman-react-password/shim/node/ap_password.js
/**
 * apeman react package for password component.
 * @class ApPassword
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_SHOW_ICON = 'ion ion-eye';
var DEFAULT_HIDE_ICON = 'ion ion-eye-disabled';

/** @lends ApPassword */
var ApPassword = _react2.default.createClass({
  displayName: 'ApPassword',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Name of text input */
    name: _react.PropTypes.string,
    /** Value of password input */
    value: _react.PropTypes.string,
    /** Placeholder of password input */
    placeholder: _react.PropTypes.string,
    /** Icon class for show button */
    showIcon: _react.PropTypes.string,
    /** Icon class for hide icon */
    hideIcon: _react.PropTypes.string
  },

  statics: {
    DEFAULT_SHOW_ICON: DEFAULT_SHOW_ICON,
    DEFAULT_HIDE_ICON: DEFAULT_HIDE_ICON
  },

  getInitialState: function getInitialState() {
    return {
      visible: false
    };
  },
  getDefaultProps: function getDefaultProps() {
    return {
      name: '',
      value: '',
      placeholder: '',
      showIcon: DEFAULT_SHOW_ICON,
      hideIcon: DEFAULT_HIDE_ICON
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var state = s.state;

    var hasVal = !!props.value;
    var type = state.visible ? 'text' : 'password';
    var icon = state.visible ? props.hideIcon : props.showIcon;
    var className = (0, _classnames2.default)('ap-password-wrap', {
      'ap-password-wrap-empty': !hasVal
    });
    return _react2.default.createElement(
      'span',
      { className: className },
      _react2.default.createElement('input', { id: props.id,
        name: props.name,
        placeholder: props.placeholder,
        onChange: props.onChange,
        className: (0, _classnames2.default)('ap-password', props.className),
        ref: function ref(input) {
          s.input = input;
        },
        value: props.value,
        type: type }),
      _react2.default.createElement(
        'a',
        { className: 'ap-password-toggle-button', onClick: s.handleToggleTap },
        _react2.default.createElement(
          'span',
          { className: 'ap-password-toggle-aligner' },
          ' '
        ),
        _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-password-toggle-icon', icon) })
      )
    );
  },


  // --------------------
  // Custom
  // --------------------

  handleToggleTap: function handleToggleTap(e) {
    var s = this;
    var state = s.state;

    s.setState({
      visible: !state.visible
    });
    _reactDom2.default.findDOMNode(s.input).focus();
  }
});

exports.default = ApPassword;
