// apeman-react-pager/shim/node/ap_pager.js
/**
 * apeman react package for pager components.
 * @class ApPager
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _numcal = require('numcal');

var _numcal2 = _interopRequireDefault(_numcal);

var _ap_pager_item = require('./ap_pager_item');

var _ap_pager_item2 = _interopRequireDefault(_ap_pager_item);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApPager */
var ApPager = function (_Component) {
  _inherits(ApPager, _Component);

  function ApPager() {
    _classCallCheck(this, ApPager);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApPager).apply(this, arguments));
  }

  _createClass(ApPager, [{
    key: 'render',


    // --------------------
    // Specs
    // --------------------
    value: function render() {
      var s = this;
      var state = s.state;
      var props = s.props;


      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-pager', props.className),
          style: Object.assign({}, props.style) },
        s.renderItems(),
        props.children
      );
    }

    // --------------------
    // Lifecycle
    // --------------------

    // ------------------
    // Private
    // ------------------

  }, {
    key: 'renderItems',
    value: function renderItems() {
      var s = this;
      var props = s.props;
      var page = props.page;
      var size = props.size;
      var total = props.total;
      var onChange = props.onChange;

      var items = [];

      var min = 0;
      var max = total - 1;
      var sizeHalf = parseInt(size / 2);

      var from = _numcal2.default.max(min, _numcal2.default.min(page - sizeHalf, max - size + 1));
      var to = _numcal2.default.min(_numcal2.default.max(min + size - 1, page + sizeHalf), max);

      var newItem = function newItem(index, title) {
        return _react2.default.createElement(_ap_pager_item2.default, { key: [index, title].join('-'),
          data: index,
          title: title,
          selected: index === page,
          onTap: function onTap(_ref) {
            var page = _ref.data;
            return onChange({ page: page });
          }
        });
      };

      for (var index = from; index <= to; index++) {
        items.push(newItem(index, String(index + 1)));
      }

      if (min < page) {
        items.unshift(newItem(page - 1, '«'));
      }
      if (page < to) {
        items.push(newItem(page + 1, '»'));
      }
      return items;
    }
  }]);

  return ApPager;
}(_react.Component);

Object.assign(ApPager, {
  propTypes: {
    /** Total page count */
    total: _react.PropTypes.number.isRequired,
    /** Current page number */
    page: _react.PropTypes.number.isRequired,
    /** Number of items to show */
    size: _react.PropTypes.number,
    /** Handler for page change */
    onChange: _react.PropTypes.func
  },
  defaultProps: function defaultProps() {
    return {
      size: 5,
      page: -1,
      total: -1
    };
  }
});

exports.default = ApPager;
