// apeman-react-pager/shim/node/ap_pager_style.js
/**
 * Style for ApPager.
 * @class ApPagerStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

var _acolor = require('acolor');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApPagerStyle */
var ApPagerStyle = function (_Component) {
  _inherits(ApPagerStyle, _Component);

  function ApPagerStyle() {
    _classCallCheck(this, ApPagerStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApPagerStyle).apply(this, arguments));
  }

  _createClass(ApPagerStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApPagerStyle$styleDa = ApPagerStyle.styleData(props);

      var all = _ApPagerStyle$styleDa.all;
      var small = _ApPagerStyle$styleDa.small;
      var medium = _ApPagerStyle$styleDa.medium;
      var large = _ApPagerStyle$styleDa.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApPagerStyle;
}(_react.Component);

Object.assign(ApPagerStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  defaultProps: {
    style: {},
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;
    var backgroundColor = config.backgroundColor;


    var size = 24;

    return {
      all: {
        '.ap-pager': {},
        '.ap-pager-item': {
          display: 'inline-block',
          textAlign: 'center',
          fontSize: '14px',
          marginRight: '-1px',
          color: '' + highlightColor,
          border: '1px solid ' + highlightColor,
          background: '' + backgroundColor
        },
        '.ap-pager-item-inner': {
          display: 'inline-block',
          padding: '4px',
          width: size + 'px',
          height: size + 'px',
          lineHeight: size + 'px'
        },
        '.ap-pager-item:hover': {
          textDecoration: 'underline',
          cursor: 'pointer',
          background: '' + (0, _acolor.mix)(highlightColor, 'white')
        },
        '.ap-pager-item:active': {
          opacity: 0.9
        },
        '.ap-pager-item.ap-pager-item-selected, .ap-pager-item.ap-pager-item-selected:hover, .ap-pager-item.ap-pager-item-selected:active': {
          cursor: 'default',
          opacity: 1,
          background: '' + highlightColor,
          color: 'white'
        }
      }

    };
  }
});

exports.default = ApPagerStyle;
