// apeman-react-pager/shim/node/ap_pager_item.js
/**
 * apeman react package for pager components.
 * @class ApPagerItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _asobj = require('asobj');

var _apemanReactTouchable = require('apeman-react-touchable');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApPagerItem */
var ApPagerItem = function (_Component) {
  _inherits(ApPagerItem, _Component);

  function ApPagerItem() {
    _classCallCheck(this, ApPagerItem);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApPagerItem).apply(this, arguments));
  }

  _createClass(ApPagerItem, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;


      var className = (0, _classnames2.default)('ap-pager-item', {
        'ap-pager-item-selected': props.selected
      }, props.className);
      return _react2.default.createElement(
        'a',
        { className: className,
          style: Object.assign({}, props.style)
        },
        _react2.default.createElement(
          'span',
          { className: 'ap-pager-item-inner' },
          props.title,
          props.children
        )
      );
    }
  }, {
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      var s = this;
      var props = s.props;
      var state = s.state;

      return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
    }
  }]);

  return ApPagerItem;
}(_react.Component);

Object.assign(ApPagerItem, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Text */
    title: _react.PropTypes.string,
    /** Selected or not */
    selected: _react.PropTypes.bool,
    /** Data for event */
    data: _react.PropTypes.any
  },

  defaultProps: {
    selected: false
  }

});

exports.default = ApPagerItem; // Dummy export for doc parser

exports.default = (0, _apemanReactTouchable.withTouch)(ApPagerItem, {
  getTouchData: function getTouchData() {
    var s = this;
    var props = s.props;

    return props.data;
  }
});
