// apeman-react-links/lib/ap_links_style.jsx
/**
 * Style for ApLinks.
 * @class ApLinksStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApLinksStyle */
const ApLinksStyle = React.createClass({
  propTypes: {
    style: types.object
  },
  getDefaultProps () {
    return {}
  },
  statics: {
    styleData (config) {
      return {
        all: {
          '.ap-links': {},
          '.ap-links-item': {},
          '.ap-links-item a': {
            display: 'inline-block',
            padding: '2px 8px'
          },
          '.ap-links-plain': {
            listStyle: 'none',
            padding: 0,
            margin: 0
          },
          '.ap-links-plain a': {
            textDecoration: 'underline',
            color: '#AAA'
          },
          '.ap-links-righted': {
            textAlign: 'right'
          }
        }
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApLinksStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApLinksStyle
