// apeman-react-html/shim/node/ap_html_style.js
/**
 * Style for ApHtml.
 * @class ApHtmlStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApHtmlStyle */
var ApHtmlStyle = _react2.default.createClass({
  displayName: 'ApHtmlStyle',

  propTypes: {
    style: _react.PropTypes.object
  },

  getDefaultProps: function getDefaultProps() {
    return {};
  },

  statics: {
    styleData: function styleData(config) {
      return {
        all: {
          '.ap-html': {
            padding: 0,
            margin: 0
          }
        }

      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApHtmlStyle$styleDat = ApHtmlStyle.styleData(props);

    var all = _ApHtmlStyle$styleDat.all;
    var small = _ApHtmlStyle$styleDat.small;
    var medium = _ApHtmlStyle$styleDat.medium;
    var large = _ApHtmlStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApHtmlStyle;
