// apeman-react-dropdown/shim/node/ap_dropdown.js
/**
 * Dropdown component
 * @class ApDropdown
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactSpinner = require('apeman-react-spinner');

var _apemanReactTouchable = require('apeman-react-touchable');

var _apemanReactMixinLayout = require('apeman-react-mixin-layout');

var _apemanReactButton = require('apeman-react-button');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApDropdown */
var ApDropdown = _react2.default.createClass({
  displayName: 'ApDropdown',


  // --------------------
  // Specs
  // --------------------
  propTypes: {
    /** Icon class name for open state */
    openIcon: _react.PropTypes.string,
    /** Icon class name for closed state */
    closedIcon: _react.PropTypes.string,
    /** Open or not */
    open: _react.PropTypes.bool.isRequired,
    /** Handler for tap event */
    onTap: _react.PropTypes.func.isRequired,
    /** Dropdown title */
    title: _react.PropTypes.node
  },

  mixins: [_apemanReactMixinLayout.ApLayoutMixin],

  getInitialState: function getInitialState() {
    return {
      open: false
    };
  },
  getDefaultProps: function getDefaultProps() {
    return {
      openIcon: 'ion ion-chevron-up',
      closedIcon: 'ion ion-chevron-down'
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;
    var layouts = s.layouts;

    var className = (0, _classnames2.default)('ap-dropdown', {
      'ap-dropdown-open': props.open,
      'ap-dropdown-closed': !props.open
    }, props.className);
    return _react2.default.createElement(
      'div',
      { className: className,
        style: (0, _assign2.default)({}, props.style) },
      props.spinner,
      _react2.default.createElement(
        'div',
        { className: 'ap-dropdown-button-wrap' },
        _react2.default.createElement(
          _apemanReactButton.ApButton,
          { className: 'ap-dropdown-button', onTap: s.handleButtonTap },
          props.title,
          _react2.default.createElement('i', { className: (0, _classnames2.default)('ap-dropdown-icon', props.open ? props.openIcon : props.closedIcon) })
        )
      ),
      _react2.default.createElement(
        'div',
        { className: 'ap-dropdown-content', style: layouts.content },
        _react2.default.createElement(
          'div',
          { className: 'ap-dropdown-content-inner',
            ref: function ref(inner) {
              return s.registerContentInner(inner);
            } },
          props.children
        )
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentWillMount: function componentWillMount() {
    var s = this;
  },
  componentDidMount: function componentDidMount() {
    var s = this;
    s.layout();
  },


  // ------------------
  // For ApLayoutMixin
  // ------------------

  getInitialLayouts: function getInitialLayouts() {
    return {
      content: null
    };
  },
  calcLayouts: function calcLayouts() {
    var s = this;
    var contentInner = s.contentInner;

    var contentRect = contentInner.getBoundingClientRect();
    return {
      content: {
        height: contentRect.height
      }
    };
  },


  // ------------------
  // Custom
  // ------------------

  contentInner: null,

  registerContentInner: function registerContentInner(contentInner) {
    var s = this;
    s.contentInner = contentInner;
  },
  handleButtonTap: function handleButtonTap(e) {
    var s = this;
    var props = s.props;

    props.onTap(e);
  }
});

exports.default = (0, _apemanReactTouchable.withOutside)((0, _apemanReactSpinner.withSpin)(ApDropdown));
