// apeman-react-slider/shim/node/ap_slider_handle.js
/**
 * Handle of slider component.
 * @class ApSliderHandle
 */
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactDraggable = require('apeman-react-draggable');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApSliderHandle */
var ApSliderHandle = _react2.default.createClass({
  displayName: 'ApSliderHandle',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    onMove: _react.PropTypes.func,
    shouldMove: _react.PropTypes.func,
    x: _react.PropTypes.number,
    minX: _react.PropTypes.number,
    maxX: _react.PropTypes.number
  },

  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      _apemanReactDraggable.ApDraggable,
      { onMove: props.onMove,
        shouldMove: props.shouldMove,
        x: props.x,
        minX: props.minX,
        maxX: props.maxX,
        className: (0, _classnames2.default)('ap-slider-handle', props.className),
        direction: 'HORIZONTAL' },
      _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement('div', { className: 'ap-slider-handle-area' }),
        _react2.default.createElement('div', { className: 'ap-slider-handle-icon' })
      )
    );
  }
});

exports.default = ApSliderHandle;
