// apeman-react-header/shim/node/ap_header_dropdown_item.js
/**
 * Dropdown item for header
 * @class ApHeaderDropdownItem
 */
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactDropdown = require('apeman-react-dropdown');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApHeaderDropdownItem */
var ApHeaderDropdownItem = _react2.default.createClass({
  displayName: 'ApHeaderDropdownItem',


  // --------------------
  // Specs
  // --------------------

  statics: {
    /**
     * Create
     * @param {string} label - Label oth the item.
     * @param {function}  onTap - Handle for on tap
     * @param {object} props - Property values.
     * @returns {object} - ApHeaderTabItem element.
     */
    createItem: function createItem(label, onTap, props) {
      return _react2.default.createElement(
        ApHeaderDropdownItem,
        (0, _extends3.default)({}, props, {
          onTap: onTap
        }),
        label
      );
    }
  },

  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      _apemanReactDropdown.ApDropdownItem,
      (0, _extends3.default)({}, props, {
        className: (0, _classnames2.default)('ap-header-dropdown-item', props.className)
      }),
      props.children
    );
  }
});

exports.default = ApHeaderDropdownItem;
