// apeman-react-header/shim/node/ap_header_tab.js
/**
 * Tab for header.
 * @class ApHeaderTab
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactTab = require('apeman-react-tab');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApHeaderTab */
var ApHeaderTab = _react2.default.createClass({
  displayName: 'ApHeaderTab',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      'nav',
      { className: 'ap-header-tab-wrap' },
      _react2.default.createElement(
        _apemanReactTab.ApTab,
        { className: (0, _classnames2.default)('ap-header-tab', props.className) },
        props.children
      )
    );
  }
});

exports.default = ApHeaderTab;
