// apeman-react-header/shim/node/ap_header_logo.js
/**
 * Logo for header.
 * @class ApHeaderLogo
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApHeaderLogo */
var ApHeaderLogo = _react2.default.createClass({
  displayName: 'ApHeaderLogo',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    href: _react.PropTypes.string
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      href: '/'
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      'h1',
      { className: (0, _classnames2.default)('ap-header-logo', props.className) },
      _react2.default.createElement(
        'a',
        { className: 'ap-header-logo-link', href: props.href },
        props.children
      )
    );
  }
});

exports.default = ApHeaderLogo;
