// apeman-react-header/shim/node/ap_header.js
/**
 * apeman react package for header components.
 * @class ApHeader
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactTouchable = require('apeman-react-touchable');

var _apemanReactMixinResize = require('apeman-react-mixin-resize');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _events = require('./events');

var _events2 = _interopRequireDefault(_events);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var addResizeHandler = _events2.default.addResizeHandler;
var removeResizeHandler = _events2.default.removeResizeHandler;

/** @lends ApHeader */

var ApHeader = _react2.default.createClass({
  displayName: 'ApHeader',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [_apemanReactMixinResize.ApResizeMixin],

  getInitialState: function getInitialState() {
    return {
      headerHeight: 'auto'
    };
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    return _react2.default.createElement(
      'div',
      { className: 'ap-header-space',
        style: { minHeight: state.headerHeight } },
      _react2.default.createElement(
        'header',
        { className: (0, _classnames2.default)('ap-header', props.className),
          style: (0, _assign2.default)({}, props.style),
          ref: function ref(header) {
            s.header = header;
          } },
        _react2.default.createElement('div', { className: 'ap-header-back' }),
        props.children
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentDidMount: function componentDidMount() {
    var s = this;
    s.resizeHeader();
    addResizeHandler(s.resizeHeader);
  },
  componentWillUnmount: function componentWillUnmount() {
    var s = this;
    removeResizeHandler(s.resizeHeader);
  },
  windowDidResize: function windowDidResize() {
    var s = this;
    s.resizeHeader();
  },


  // ------------------
  // Custom
  // ------------------

  resizeHeader: function resizeHeader() {
    var s = this;
    var header = s.header;

    if (!header) {
      return;
    }

    var _header$getBoundingCl = header.getBoundingClientRect();

    var height = _header$getBoundingCl.height;

    s.setState({
      headerHeight: height
    });
  }

  // ------------------
  // Private
  // ------------------

});

var ApHeaderWrap = (0, _apemanReactTouchable.withOutside)(ApHeader);
ApHeaderWrap.events = _events2.default;

exports.default = ApHeaderWrap;
