// apeman-react-header/shim/node/ap_header_menu_item.js
/**
 * menu for header.
 * @class ApHeaderMenuItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactButton = require('apeman-react-button');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApHeaderMenuItem */
var ApHeaderMenuItem = _react2.default.createClass({
  displayName: 'ApHeaderMenuItem',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  statics: {
    /**
     * Create an props.
     * @param {string} label - Label oth the item.
     * @param {function} onTap - Callback for on tap.
     * @param {object} props - Property values.
     * @returns {object} - ApHeaderMenu element.
     */
    createItem: function createItem(label, onTap, props) {
      return _react2.default.createElement(
        ApHeaderMenuItem,
        (0, _extends3.default)({ onTap: onTap }, props),
        label
      );
    },

    /**
     * Create a primary button item.
     * @param {string} label - Label oth the item.
     * @param {function} onTap - Callback for on tap.
     * @param {object} props - Property values.
     * @returns {object} - ApHeaderMenu element.
     */
    createPrimaryItem: function createPrimaryItem(label, onTap, props) {
      return ApHeaderMenuItem.createItem(label, onTap, (0, _assign2.default)({}, props, { primary: true }));
    }
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      _apemanReactButton.ApButton,
      (0, _extends3.default)({}, props, {
        className: (0, _classnames2.default)('ap-header-menu-item', props.className)
      }),
      props.children
    );
  }
});

exports.default = ApHeaderMenuItem;
