// apeman-react-header/shim/node/ap_header_tab_item.js
/**
 * Tab item for header.
 * @class ApHeaderTabItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactTab = require('apeman-react-tab');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApHeaderTabItem */
var ApHeaderTabItem = function (_Component) {
  (0, _inherits3.default)(ApHeaderTabItem, _Component);

  function ApHeaderTabItem() {
    (0, _classCallCheck3.default)(this, ApHeaderTabItem);
    return (0, _possibleConstructorReturn3.default)(this, (ApHeaderTabItem.__proto__ || (0, _getPrototypeOf2.default)(ApHeaderTabItem)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApHeaderTabItem, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var className = (0, _classnames2.default)('ap-header-tab-item', props.className, {
        'ap-header-tab-item-selected': !!props.selected
      });
      return _react2.default.createElement(
        _apemanReactTab.ApTabItem,
        (0, _extends3.default)({}, props, {
          className: className
        }),
        props.children
      );
    }
  }]);
  return ApHeaderTabItem;
}(_react.Component);

(0, _assign2.default)(ApHeaderTabItem, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    selected: _react.PropTypes.bool
  },

  /**
   * Create
   * @param {string} label - Label oth the item.
   * @param {string} href - Tab href
   * @param {object} props - Property values.
   * @returns {object} - ApHeaderTabItem element.
   */
  createItem: function createItem(label, href, props) {
    return _react2.default.createElement(
      ApHeaderTabItem,
      (0, _extends3.default)({ href: href
      }, props),
      label
    );
  },


  defaultProps: {
    selected: false
  }

});

exports.default = ApHeaderTabItem;
