// apeman-react-toast/shim/node/ap_error_toast.js
/**
 * Error toast.
 * @class ApErrorToast
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ap_toast = require('./ap_toast');

var _ap_toast2 = _interopRequireDefault(_ap_toast);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApErrorToast */
var ApErrorToast = _react2.default.createClass({
  displayName: 'ApErrorToast',


  // --------------------
  // Specs
  // --------------------

  propTypes: _ap_toast2.default.propTypes,

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      messages: [],
      icon: 'fa fa-check-circle',
      dismiss: null,
      level: 'error'
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;


    return _react2.default.createElement(_ap_toast2.default, _extends({}, props, { className: 'ap-error-toast' }));
  }
});

exports.default = ApErrorToast;
