// apeman-react-toast/shim/node/ap_toast_group.js
/**
 * Toast group.
 * @class ApToastGroup
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _asobj = require('asobj');

var _ap_toast = require('./ap_toast');

var _ap_toast2 = _interopRequireDefault(_ap_toast);

var _ap_info_toast = require('./ap_info_toast');

var _ap_info_toast2 = _interopRequireDefault(_ap_info_toast);

var _ap_warn_toast = require('./ap_warn_toast');

var _ap_warn_toast2 = _interopRequireDefault(_ap_warn_toast);

var _ap_error_toast = require('./ap_error_toast');

var _ap_error_toast2 = _interopRequireDefault(_ap_error_toast);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var levels = ['default', 'info', 'error', 'warn'];

/** @lends ApToastGroup */
var ApToastGroup = _react2.default.createClass({
  displayName: 'ApToastGroup',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Default messages */
    default: _react.PropTypes.arrayOf(_react.PropTypes.string),
    /** Info messages */
    info: _react.PropTypes.arrayOf(_react.PropTypes.string),
    /** Error messages */
    error: _react.PropTypes.arrayOf(_react.PropTypes.string),
    /** Warn messages */
    warn: _react.PropTypes.arrayOf(_react.PropTypes.string),
    /** Dismiss message */
    dismiss: _react.PropTypes.func
  },

  mixins: [],

  statics: {
    levels: levels
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      info: [],
      error: [],
      warn: []
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;
    var default_ = props.default;
    var className = props.className;
    var info = props.info;
    var error = props.error;
    var warn = props.warn;
    var _dismiss = props.dismiss;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-toast-group', {}, className) },
      _react2.default.createElement(_ap_toast2.default, { messages: default_, dismiss: function dismiss(message) {
          return _dismiss(message, 'default');
        } }),
      _react2.default.createElement(_ap_info_toast2.default, { messages: info, dismiss: function dismiss(message) {
          return _dismiss(message, 'info');
        } }),
      _react2.default.createElement(_ap_warn_toast2.default, { messages: warn, dismiss: function dismiss(message) {
          return _dismiss(message, 'warn');
        } }),
      _react2.default.createElement(_ap_error_toast2.default, { messages: error, dismiss: function dismiss(message) {
          return _dismiss(message, 'error');
        } }),
      props.children
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentDidMount: function componentDidMount() {
    var s = this;
    var props = s.props;
  },
  componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
    var s = this;
    var props = s.props;
  },
  componentWillUnmount: function componentWillUnmount() {
    var s = this;
    var props = s.props;
  },
  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props;
    var state = s.state;
    var _iteratorNormalCompletion = true;
    var _didIteratorError = false;
    var _iteratorError = undefined;

    try {
      for (var _iterator = levels[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
        var level = _step.value;

        var messageChanged = (0, _asobj.deepEqual)(props[level], nextProps[level]);
        if (messageChanged) {
          return true;
        }
      }
    } catch (err) {
      _didIteratorError = true;
      _iteratorError = err;
    } finally {
      try {
        if (!_iteratorNormalCompletion && _iterator.return) {
          _iterator.return();
        }
      } finally {
        if (_didIteratorError) {
          throw _iteratorError;
        }
      }
    }

    return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
  },


  // --------------------
  // Custom
  // --------------------

  handleDismiss: function handleDismiss(e) {
    var s = this;
    var state = s.state;
    var name = e.name;
    var dismissed = e.dismissed;

    if (state[name] === dismissed) {
      s.setState({ name: null });
    }
  },


  /**
   * Bind toaster events.
   * @param toaster
   */
  bindToaster: function bindToaster(toaster) {
    var s = this;
    toaster.addListener('toast', s.handleToast);
  },


  /**
   * Unbind toaster events.
   * @param toaster
   */
  unbindToaster: function unbindToaster(toaster) {
    var s = this;
    toaster.removeListener('toast', s.handleToast);
  },


  /**
   * Handle toaster events.
   * @param data
   */
  handleToast: function handleToast(data) {
    var s = this;
    var state = { _date: new Date() };
    Toaster.levels.forEach(function (level) {
      state[level] = data.level === level ? data.message : null;
    });
    s.setState(state);
  }
});

exports.default = ApToastGroup;
