// apeman-react-toast/shim/node/ap_toast.js
/**
 * Toast component
 * @class ApToast
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _asobj = require('asobj');

var _ap_toast_item = require('./ap_toast_item');

var _ap_toast_item2 = _interopRequireDefault(_ap_toast_item);

var _reactAddonsCssTransitionGroup = require('react-addons-css-transition-group');

var _reactAddonsCssTransitionGroup2 = _interopRequireDefault(_reactAddonsCssTransitionGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApToast */
var ApToast = _react2.default.createClass({
  displayName: 'ApToast',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Messages to show */
    messages: _react.PropTypes.arrayOf(_react.PropTypes.string),
    /** Icon to show */
    icon: _react.PropTypes.string,
    /** Dismiss an item */
    dismiss: _react.PropTypes.func,
    /** Level of toast */
    level: _react.PropTypes.string
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      transitionTimeout: 100,
      level: 'default',
      messages: [],
      icon: null,
      dismiss: null
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var messages = props.messages;
    var icon = props.icon;
    var dismiss = props.dismiss;
    var className = props.className;
    var style = props.style;
    var level = props.level;
    var transitionTimeout = props.transitionTimeout;


    var children = messages.filter(Boolean).map(function (message) {
      return _react2.default.createElement(_ap_toast_item2.default, { key: message,
        text: message,
        icon: icon,
        onTap: function onTap() {
          return dismiss(message, level);
        }
      });
    });
    if (children.length === 0) {
      return null;
    }
    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-toast', className, {}),
        style: Object.assign({}, style) },
      _react2.default.createElement(
        _reactAddonsCssTransitionGroup2.default,
        { component: 'div',
          className: 'ap-toast-inner',
          transitionName: 'ap-toast-transition',
          transitionEnterTimeout: transitionTimeout,
          transitionLeaveTimeout: transitionTimeout
        },
        children
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentWillMount: function componentWillMount() {
    var s = this;
  },
  componentDidMount: function componentDidMount() {
    var s = this;
    var props = s.props;
  },
  componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
    var s = this;
  },
  componentWillUnmount: function componentWillUnmount() {
    var s = this;
  },
  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props;
    var state = s.state;

    var messageChanged = !(0, _asobj.deepEqual)(props.messages, nextProps.messages);
    if (messageChanged) {
      return true;
    }
    return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
  }

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApToast;
