// apeman-react-toast/shim/node/ap_toast_item.js
/**
 * Toast item component
 * @class ApToastItem
 */
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _asobj = require('asobj');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

var _apemanReactTouchable = require('apeman-react-touchable');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApToastItem */
var ApToastItem = _react2.default.createClass({
  displayName: 'ApToastItem',

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    text: _react.PropTypes.string,
    icon: _react.PropTypes.string
  },

  mixins: [],

  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      'div',
      { className: 'ap-toast-item' },
      _react2.default.createElement(
        'span',
        { className: 'ap-toast-text' },
        _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-toast-item-icon', props.icon) }),
        props.text
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props;
    var state = s.state;

    return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
  }
});

exports.default = ApToastItem; // Dummy for doc

exports.default = (0, _apemanReactTouchable.withTouch)(ApToastItem);
