// react-flashcard-app/lib/FlashcardApp.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _containersApp = require('./containers/app');

var _containersApp2 = _interopRequireDefault(_containersApp);

var _configureStore = require('./configureStore');

var _configureStore2 = _interopRequireDefault(_configureStore);

var store = (0, _configureStore2['default'])();

var FlashcardApp = function FlashcardApp(props) {
  return _react2['default'].createElement(
    _reactRedux.Provider,
    { store: store },
    _react2['default'].createElement(_containersApp2['default'], { data: props.data, options: props.options })
  );
};

FlashcardApp.propTypes = {
  data: _propTypes2['default'].object.isRequired,
  options: _propTypes2['default'].object
};

FlashcardApp.defaultProps = {
  options: {
    topControlBar: ['decrement', 'undoRemove', 'remove', 'increment'],
    bottomControlBar: ['flip', 'reset', 'shuffle', 'revert'],
    buttonTexts: {}
  }
};

exports['default'] = FlashcardApp;
module.exports = exports['default'];
