// apeman-react-info/shim/node/ap_info_item.js
/**
 * Item of info.
 * @class ApInfoItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactMixinPure = require('apeman-react-mixin-pure');

var _apemanReactTouchable = require('apeman-react-touchable');

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApInfoItem */
var ApInfoItem = _react2.default.createClass({
  displayName: 'ApInfoItem',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    label: _react.PropTypes.string,
    value: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number, _react.PropTypes.node]),
    unit: _react.PropTypes.string,
    icon: _react.PropTypes.string,
    link: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.func]),
    linkText: _react.PropTypes.string
  },

  mixins: [_apemanReactMixinPure.ApPureMixin],

  statics: {
    OPEN_ICON: 'fa fa-angle-down',
    CLOSE_ICON: 'fa fa-angle-right',
    NEXT_ICON: 'fa fa-angle-right',
    PREV_ICON: 'fa fa-angle-left'
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      icon: null,
      label: null,
      value: null,
      link: null,
      linkText: null,
      unit: null
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;

    var className = (0, _classnames2.default)('ap-info-item', {
      'ap-info-item-tappable': !!props.React
    }, props.className);
    var label = props.label;
    var value = props.value;
    var unit = props.unit;
    var link = props.link;
    var linkText = props.linkText;
    var icon = props.icon;

    return _react2.default.createElement(
      'div',
      { className: className,
        style: (0, _assign2.default)({}, props.style) },
      label ? s.renderLabel(label) : null,
      props.children,
      _react2.default.createElement(
        'div',
        { className: 'ap-info-item-value' },
        link && linkText ? s.renderValueText(value) : null,
        link ? s.renderValueLink(linkText || value, link) : s.renderValueText(value),
        unit ? _react2.default.createElement(
          'small',
          { className: 'ap-info-item-unit' },
          unit
        ) : null,
        icon ? s.renderIcon(icon) : null
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
  renderLabel: function renderLabel(label) {
    var s = this;
    return _react2.default.createElement(
      'div',
      { className: 'ap-info-item-label' },
      label
    );
  },
  renderValueText: function renderValueText(text) {
    return _react2.default.createElement(
      'span',
      { className: 'ap-info-item-value-text' },
      text
    );
  },
  renderValueLink: function renderValueLink(text, link) {
    var s = this;
    if (typeof link === 'function') {
      return _react2.default.createElement(
        'a',
        { className: 'ap-info-item-link', href: 'javascript:void(0)',
          onClick: link },
        text
      );
    } else {
      return _react2.default.createElement(
        'a',
        { className: 'ap-info-item-link', href: link },
        text
      );
    }
  },
  renderIcon: function renderIcon(icon) {
    var s = this;
    return _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)(icon, 'ap-info-item-icon') });
  }
});

exports.default = (0, _apemanReactTouchable.withTouch)(ApInfoItem);
