// apeman-react-info/shim/node/ap_info.js
/**
 * Infor component.
 * @class ApInfo
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApInfo */
var ApInfo = _react2.default.createClass({
  displayName: 'ApInfo',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    pullLeft: _react.PropTypes.bool,
    pullRight: _react.PropTypes.bool
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-info', {
          'ap-info-left': props.pullLeft,
          'ap-info-right': props.pullRight
        }, props.className),
        style: (0, _assign2.default)({}, props.style) },
      props.children
    );
  }
  // --------------------
  // Lifecycle
  // --------------------
  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApInfo;
