// apeman-react-info/shim/node/ap_info_collapsible_item.js
/**
 * Info item with collapsible feature
 * @class ApInfoCollapsibleItem
 */
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactMixinPure = require('apeman-react-mixin-pure');

var _apemanReactMixinLayout = require('apeman-react-mixin-layout');

var _ap_info_item = require('./ap_info_item');

var _ap_info_item2 = _interopRequireDefault(_ap_info_item);

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var OPEN_ICON = _ap_info_item2.default.OPEN_ICON;
var CLOSE_ICON = _ap_info_item2.default.CLOSE_ICON;

/** @lends ApInfoCollapsibleItem */

var ApInfoCollapsibleItem = _react2.default.createClass({
  displayName: 'ApInfoCollapsibleItem',

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Label text */
    label: _react.PropTypes.string,
    /** Value text */
    value: _react.PropTypes.string,
    /** Unit suffix */
    unit: _react.PropTypes.string,
    /** Callback for toggle */
    onToggle: _react.PropTypes.func
  },

  mixins: [_apemanReactMixinPure.ApPureMixin, _apemanReactMixinLayout.ApLayoutMixin],

  getInitialState: function getInitialState() {
    return {
      open: false
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;
    var layouts = s.layouts;

    var className = (0, _classnames2.default)('ap-info-collapsible-item', props.className);
    var icon = state.open ? OPEN_ICON : CLOSE_ICON;
    return _react2.default.createElement(
      'div',
      { className: className },
      _react2.default.createElement(
        'div',
        { className: 'ap-info-collapsible-header' },
        _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-info-item-icon', 'ap-info-collapsible-icon', icon) }),
        _react2.default.createElement(_ap_info_item2.default, { label: props.label,
          unit: props.unit,
          value: props.value,
          onTap: s.toggleCollapsing
        })
      ),
      _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-info-collapsible-wrap', {
            'ap-info-collapsible-wrap-closed': !state.open
          }), style: layouts.wrap },
        _react2.default.createElement(
          'div',
          { className: 'ap-info-collapsible-body',
            ref: function ref(body) {
              s.body = body;
            }
          },
          props.children
        )
      )
    );
  },


  // --------------------
  // For ApLayoutMixin
  // --------------------

  getInitialLayouts: function getInitialLayouts() {
    return {
      wrap: {
        height: 'auto'
      }
    };
  },
  calcLayouts: function calcLayouts() {
    var s = this;
    var body = s.body;

    var bodyRect = body && body.getBoundingClientRect();
    return {
      wrap: {
        height: bodyRect && bodyRect.height
      }
    };
  },


  // --------------------
  // Custom
  // --------------------
  toggleCollapsing: function toggleCollapsing(e) {
    var s = this;
    var props = s.props;
    var state = s.state;

    s.setState({
      open: !state.open
    });
    if (props.onToggle) {
      props.onToggle(e);
    }
  }
});

exports.default = ApInfoCollapsibleItem;
