// apeman-react-catalog/lib/ap_catalog_item_summary.jsx
/**
 * Summary of a catalog item
 * @class ApCatalogItemSummary
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import {withTouch} from 'apeman-react-touchable'

/** @lends ApCatalogItemSummary */
const ApCatalogItemSummary = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  statics: {},

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {}
  },

  render () {
    const s = this
    let { state, props } = s

    return (
      <div className={ classnames('ap-catalog-item-summary', props.className)}
           style={Object.assign({}, props.style) }>
        { props.children }
      </div>
    )
  },

  // --------------------
  // Lifecycle
  // --------------------

  componentWillMount () {
    const s = this
  },

  componentDidMount () {
    const s = this
  },

  componentWillReceiveProps (nextProps) {
    const s = this
  },

  shouldComponentUpdate(nextProps, nextState) {
    const s = this
    return true
  },

  componentWillUpdate(nextProps, nextState) {
    const s = this
  },

  componentDidUpdate(prevProps, prevState) {
    const s = this
  },

  componentWillUnmount () {
    const s = this
  }

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
})

export default ApCatalogItemSummary // Dummy for doc parser
export default withTouch(ApCatalogItemSummary)
