// apeman-react-catalog/lib/ap_catalog.jsx
/**
 * Catalog component
 * @class ApCatalog
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'

/** @lends ApCatalog */
const ApCatalog = React.createClass({


    // --------------------
    // Specs
    // --------------------

    propTypes: {},

    mixins: [],

    statics: {},

    getInitialState () {
        return {}
    },

    getDefaultProps () {
        return {}
    },

    render () {
        const s = this
        let {state, props} = s

        return (
            <div className={ classnames('ap-catalog', props.className)}
                 style={Object.assign({}, props.style) }>
                <div className="ap-catalog-inner">
                    { props.children }
                </div>
            </div>
        )
    },


    // --------------------
    // Lifecycle
    // --------------------


    // ------------------
    // Custom
    // ------------------

    // ------------------
    // Private
    // ------------------
})

export default ApCatalog;
