// apeman-react-catalog/lib/ap_catalog_item_detail.jsx
/**
 * Detail of a catalog item
 * @class ApCatalogItemDetail
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'

/** @lends ApCatalogItemDetail */
const ApCatalogItemDetail = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    open: types.bool,
    at: types.oneOf([
      'TOP',
      'CENTER',
      'LEFT',
      'RIGHT',
      'BOTTOM'
    ])
  },

  mixins: [],

  statics: {},

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {
      open: false,
      at: null
    }
  },

  render () {
    const s = this
    let { state, props } = s

    let at = String(props.at || null).toUpperCase()

    return (
      <div className={ classnames('ap-catalog-item-detail', {
                'ap-catalog-item-detail-open': props.open,
                'ap-catalog-item-detail-at-left': at === 'LEFT',
                'ap-catalog-item-detail-at-top': at === 'TOP',
                'ap-catalog-item-detail-at-right': at === 'RIGHT',
                'ap-catalog-item-detail-at-bottom': at === 'BOTTOM'
            }, props.className)}
           style={Object.assign({}, props.style) }>
        {props.open ? props.children : null}
      </div>
    )
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
})

export default ApCatalogItemDetail
