// apeman-react-page/lib/ap_page_style.jsx
/**
 * Style for ApPage.
 * @class ApPageStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApPageStyle */
const ApPageStyle = React.createClass({
  propTypes: {
    style: types.object
  },
  getDefaultProps () {
    return {
      style: {}
    }
  },
  statics: {
    styleData(conifg) {
      return {
        all: {
          '.ap-page-container': {
            display: 'block'
          },
          '.ap-page': {
            display: 'block',
            position: 'fixed',
            left: 0,
            top: 0,
            right: 0,
            bottom: 0,
            overflow: 'auto'
          }
        }
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApPageStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApPageStyle
