// apeman-react-article/shim/node/ap_article.js
/**
 * apeman react package for article components.
 * @class ApArticle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApArticle */
var ApArticle = _react2.default.createClass({
  displayName: 'ApArticle',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    return _react2.default.createElement(
      'article',
      { className: (0, _classnames2.default)('ap-article', props.className),
        style: Object.assign({}, props.style) },
      props.children
    );
  }
});

exports.default = ApArticle;
