// apeman-react-article/shim/node/ap_article_style.js
/**
 * Style for ApArticle.
 * @class ApArticleStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApArticleStyle */
var ApArticleStyle = function (_Component) {
  _inherits(ApArticleStyle, _Component);

  function ApArticleStyle() {
    _classCallCheck(this, ApArticleStyle);

    return _possibleConstructorReturn(this, (ApArticleStyle.__proto__ || Object.getPrototypeOf(ApArticleStyle)).apply(this, arguments));
  }

  _createClass(ApArticleStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApArticleStyle$style = ApArticleStyle.styleData(props);

      var all = _ApArticleStyle$style.all;
      var small = _ApArticleStyle$style.small;
      var medium = _ApArticleStyle$style.medium;
      var large = _ApArticleStyle$style.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApArticleStyle;
}(_react.Component);

Object.assign(ApArticleStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  defaultProps: {
    style: {},
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;
    var backgroundColor = config.backgroundColor;

    return {
      all: {
        '.ap-article': {
          padding: '16px 0'
        },
        '.ap-selectable-article-content': {
          display: 'none'
        },
        '.ap-selectable-article-header': {
          display: 'flex',
          alignItems: 'center'
        },
        '.ap-selectable-article-header .ap-label': {
          padding: '4px 8px 4px 0'
        },
        '.ap-selectable-article-body': {
          padding: '8px 0',
          boxSizing: 'border-box'
        },
        '.ap-selectable-article-alt': {
          display: 'block',
          textAlign: 'center',
          opacity: 0.3,
          padding: '16px 8px'
        }
      }
    };
  }
});

exports.default = ApArticleStyle;
