// apeman-react-article/shim/node/ap_selectable_article.js
/**
 * Article with selections
 * @class ApSelectableArticle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactStyle = require('apeman-react-style');

var _apemanReactSelect = require('apeman-react-select');

var _apemanReactLabel = require('apeman-react-label');

var _uuid = require('uuid');

var _uuid2 = _interopRequireDefault(_uuid);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/** @lends ApSelectableArticle */
var ApSelectableArticle = _react2.default.createClass({
  displayName: 'ApSelectableArticle',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Label */
    label: _react.PropTypes.string,
    /** Article id */
    id: _react.PropTypes.string,
    /** Name of select */
    name: _react.PropTypes.string,
    /** Value of select */
    value: _react.PropTypes.string,
    /** Placeholder of select */
    placeholder: _react.PropTypes.string,
    /** Select options */
    options: _react.PropTypes.object,
    /** Alt messages */
    alt: _react.PropTypes.string
  },

  mixins: [],

  statics: {
    Content: _react2.default.createClass({
      displayName: 'Content',

      propTypes: {
        /** Select option value */
        contentFor: _react.PropTypes.string.isRequired
      },
      render: function render() {
        var s = this;
        var props = s.props;

        var className = (0, _classnames2.default)('ap-selectable-article-content', 'ap-selectable-article-content-' + props.contentFor);

        return _react2.default.createElement(
          'div',
          { className: className },
          props.children
        );
      }
    })
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;
    var label = props.label;
    var value = props.value;
    var alt = props.alt;
    var _props$id = props.id;
    var id = _props$id === undefined ? 'ap-selectable-article-' + s.uuid : _props$id;

    var select = ['id', 'name', 'placeholder', 'options', 'value', 'onChange', 'multiple'].reduce(function (select, name) {
      return Object.assign(select, _defineProperty({}, name, props[name]));
    }, {});
    select.id = id + '-select';
    return _react2.default.createElement(
      'article',
      { className: (0, _classnames2.default)('ap-selectable-article', props.className),
        id: id,
        style: Object.assign({}, props.style) },
      _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { className: 'ap-selectable-article-dynamic-style'
        },
        '#' + id + ' .ap-selectable-article-content-' + value + ' { display: block; }'
      ),
      _react2.default.createElement(
        'div',
        { className: 'ap-selectable-article-header' },
        _react2.default.createElement(
          _apemanReactLabel.ApLabel,
          { htmlFor: select.id },
          label ? label : null
        ),
        _react2.default.createElement(_apemanReactSelect.ApSelect, select)
      ),
      _react2.default.createElement(
        'div',
        { className: 'ap-selectable-article-body' },
        value ? null : _react2.default.createElement(
          'div',
          { className: 'ap-selectable-article-alt' },
          alt
        ),
        props.children
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentDidMount: function componentDidMount() {
    var s = this;
    s.uuid = _uuid2.default.v4();
  }

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApSelectableArticle;
