// apeman-react-main/shim/node/ap_main_style.js
/**
 * Style for ApMain.
 * @class ApMainStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApMainStyle */
var ApMainStyle = _react2.default.createClass({
  displayName: 'ApMainStyle',

  propTypes: {
    style: _react.PropTypes.object
  },

  getDefaultProps: function getDefaultProps() {
    return {
      style: {}
    };
  },

  statics: {
    styleData: function styleData(config) {
      return {
        all: {}
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApMainStyle$styleDat = ApMainStyle.styleData(props);

    var all = _ApMainStyle$styleDat.all;
    var small = _ApMainStyle$styleDat.small;
    var medium = _ApMainStyle$styleDat.medium;
    var large = _ApMainStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApMainStyle;
