// apeman-react-tab/shim/node/ap_tab_content.js
/**
 * Content of tab
 * @class ApTabContent
 */
'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApTabContent */
var ApTabContent = _react2.default.createClass({
  displayName: 'ApTabContent',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** The tab is selected or not */
    selected: _react.PropTypes.bool
  },

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var className = (0, _classnames2.default)('ap-tab-content', props.className, {
      'ap-tab-content-selected': !!props.selected
    });
    var style = Object.assign({}, props.style);
    return _react2.default.createElement(
      'div',
      { className: className,
        style: style
      },
      props.children
    );
  }
});

module.exports = ApTabContent;
