// apeman-react-tab/shim/node/ap_tab_style.js
/**
 * Style for ApTab.
 * @class ApTabStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApTabStyle */
var ApTabStyle = function (_Component) {
  _inherits(ApTabStyle, _Component);

  function ApTabStyle() {
    _classCallCheck(this, ApTabStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApTabStyle).apply(this, arguments));
  }

  _createClass(ApTabStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApTabStyle$styleData = ApTabStyle.styleData(props);

      var all = _ApTabStyle$styleData.all;
      var small = _ApTabStyle$styleData.small;
      var medium = _ApTabStyle$styleData.medium;
      var large = _ApTabStyle$styleData.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApTabStyle;
}(_react.Component);

Object.assign(ApTabStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string,
    backgroundColor: _react.PropTypes.string,
    borderColor: _react.PropTypes.string
  },

  defaultProps: {
    style: {},
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR,
    borderColor: '#888'
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;
    var borderColor = config.borderColor;
    var backgroundColor = config.backgroundColor;

    return {
      all: {
        '.ap-tab': {
          display: 'flex'
        },
        '.ap-tab-item': {
          display: 'inline-block',
          flex: 1,
          textAlign: 'center',
          padding: '8px',
          color: '#555',
          border: '1px solid ' + borderColor,
          marginRight: '-1px',
          position: 'relative',
          textDecoration: 'none'
        },
        '.ap-tab-item:hover': {
          boxSizing: 'border-box',
          cursor: 'pointer',
          textDecoration: 'none'
        },
        '.ap-tab-item-selected': {
          color: '' + highlightColor,
          borderColor: '' + highlightColor,
          zIndex: 2
        },
        '.ap-tab-item:active': {
          color: '' + highlightColor,
          borderColor: '' + highlightColor
        },
        '.ap-tab-group': {
          display: 'block',
          padding: '4px 0 0',
          border: '1px solid ' + borderColor,
          background: '#EEE'
        },
        '.ap-tab-content': {
          display: 'none',
          borderTop: '1px solid ' + borderColor,
          marginTop: '-1px',
          position: 'relative',
          zIndex: 2,
          padding: '4px 8px',
          background: backgroundColor
        },
        '.ap-tab-content-selected': {
          display: 'block'
        },
        '.ap-tab-group .ap-tab': {
          margin: '4px 8px 0'
        },
        '.ap-tab-group .ap-tab-item': {
          padding: '4px 0 2px',
          borderBottom: 'none',
          position: 'relative',
          zIndex: 1,
          color: '#888',
          borderColor: 'transparent'
        },
        '.ap-tab-group .ap-tab-item-selected': {
          zIndex: 3,
          background: backgroundColor,
          color: 'inherit',
          borderColor: borderColor
        }
      }
    };
  }
});

exports.default = ApTabStyle;
