// react-flashcard-app/lib/actions/index.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
var initApp = function initApp(data) {
  return {
    type: 'INIT_APP',
    data: data
  };
};

exports.initApp = initApp;
var increment = function increment() {
  return {
    type: 'INCREMENT'
  };
};

exports.increment = increment;
var decrement = function decrement() {
  return {
    type: 'DECREMENT'
  };
};

exports.decrement = decrement;
var flip = function flip() {
  return {
    type: 'FLIP'
  };
};

exports.flip = flip;
var revert = function revert() {
  return {
    type: 'REVERT'
  };
};

exports.revert = revert;
var remove = function remove() {
  return {
    type: 'REMOVE_CARD'
  };
};

exports.remove = remove;
var undoRemove = function undoRemove() {
  return {
    type: 'UNDO_REMOVE'
  };
};

exports.undoRemove = undoRemove;
var reset = function reset() {
  return {
    type: 'RESET'
  };
};

exports.reset = reset;
var shuffle = function shuffle() {
  return {
    type: 'SHUFFLE'
  };
};
exports.shuffle = shuffle;
