// react-flashcard-app/lib/components/bottom_control_bar.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _control_buttons = require('./control_buttons');

var _control_buttons2 = _interopRequireDefault(_control_buttons);

var BottomControlBar = function BottomControlBar(_ref) {
  var options = _ref.options;
  return _react2['default'].createElement(
    'div',
    {
      id: 'bottom-control-bar',
      className: 'btn-group btn-group-justified',
      role: 'group',
      'aria-label': '...'
    },
    options.buttons.map(function (button, index) {
      return (0, _control_buttons2['default'])(button, options.buttonTexts[button], index);
    })
  );
};

BottomControlBar.propTypes = {
  options: _propTypes2['default'].shape({
    buttons: _propTypes2['default'].arrayOf(_propTypes2['default'].string),
    buttonTexts: _propTypes2['default'].objectOf(_propTypes2['default'].string)
  })
};

BottomControlBar.defaultProps = {
  options: {}
};

exports['default'] = BottomControlBar;
module.exports = exports['default'];
