// react-flashcard-app/src/containers/deck.js
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { flip } from '../actions';
import Card from './card';

export const Deck = (props) => {
  if (props.total === 0) {
    return (
      <div className='deck'>
        <div className='card' data-is-current='true'>
          <div className='card-face'>
            <div className='card-text'>
              no cards left in this deck
            </div>
          </div>
        </div>
      </div>
    );
  }
  return (
    <div onClick={props.flip} className='deck' title='click to flip'>
      {props.cards.map((card, index) => (
        <Card
          isCurrent={index === props.cardIndex}
          key={card.id}
          values={card.values}
        />
      ))}
    </div>
  );
};

Deck.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  flip: PropTypes.func.isRequired,
  total: PropTypes.number.isRequired,
  cardIndex: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  cardIndex: state.app.cardIndex,
  cards: state.app.deck.cards,
  removed: state.app.removed,
  total: state.app.deck.length - state.app.removed.length,
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({ flip }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Deck);
