// apeman-react-links/shim/node/ap_links_style.js
/**
 * Style for ApLinks.
 * @class ApLinksStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApLinksStyle */
var ApLinksStyle = _react2.default.createClass({
  displayName: 'ApLinksStyle',

  propTypes: {
    style: _react.PropTypes.object
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },

  statics: {
    styleData: function styleData(config) {
      return {
        all: {
          '.ap-links': {},
          '.ap-links-item': {},
          '.ap-links-item a': {
            display: 'inline-block',
            padding: '2px 8px'
          },
          '.ap-links-plain': {
            listStyle: 'none',
            padding: 0,
            margin: 0
          },
          '.ap-links-plain a': {
            textDecoration: 'underline',
            color: '#AAA'
          },
          '.ap-links-righted': {
            textAlign: 'right'
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApLinksStyle$styleDa = ApLinksStyle.styleData(props);

    var all = _ApLinksStyle$styleDa.all;
    var small = _ApLinksStyle$styleDa.small;
    var medium = _ApLinksStyle$styleDa.medium;
    var large = _ApLinksStyle$styleDa.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApLinksStyle;
