// apeman-react-links/shim/node/ap_links.js
/**
 * apeman react package for link list component.
 * @class ApLinks
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApLinks */
var ApLinks = _react2.default.createClass({
  displayName: 'ApLinks',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    links: _react.PropTypes.oneOfType([_react.PropTypes.object, _react.PropTypes.func]),
    plain: _react.PropTypes.bool,
    righted: _react.PropTypes.bool
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      links: {},
      plain: false,
      righted: false

    };
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var links = props.links || {};
    return _react2.default.createElement(
      'ul',
      { className: (0, _classnames2.default)('ap-links', {
          'ap-links-plain': props.plain,
          'ap-links-righted': props.righted
        }, props.className) },
      Object.keys(links).map(function (key) {
        return _react2.default.createElement(
          'li',
          { className: 'ap-links-item',
            key: 'ap-links-item-' + key },
          s._renderAnchor(key, links[key])
        );
      })
    );
  },
  _renderAnchor: function _renderAnchor(text, link) {
    var s = this;
    var href = null;
    var onClick = null;
    if (typeof link === 'function') {
      return _react2.default.createElement(
        'a',
        { href: 'javascript:void(0)',
          onClick: link },
        text
      );
    } else {
      return _react2.default.createElement(
        'a',
        { href: link },
        text
      );
    }
  }
});

exports.default = ApLinks;
