// apeman-react-table/shim/node/ap_table_header.js
/**
 * Table header component
 * @class ApTableHeader
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactMixinPure = require('apeman-react-mixin-pure');

var _apemanReactTouchable = require('apeman-react-touchable');

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApTableHeader */
var ApTableHeader = _react2.default.createClass({
  displayName: 'ApTableHeader',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Icon class name */
    icon: _react.PropTypes.string,
    /** Handler for tap event */
    onTap: _react.PropTypes.func
  },

  mixins: [_apemanReactMixinPure.ApPureMixin],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    var className = (0, _classnames2.default)('ap-table-cell', 'ap-table-header', {
      'ap-table-header-tappable': !!props.onTap
    }, props.className);
    return _react2.default.createElement(
      'th',
      { className: className,
        style: Object.assign({}, props.style) },
      _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-table-header-icon', props.icon) }),
      props.children
    );
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApTableHeader; // Dummy for doc

exports.default = (0, _apemanReactTouchable.withTouch)(ApTableHeader);
