// apeman-react-table/shim/node/ap_table_style.js
/**
 * Style for ApTable.
 * @class ApTableStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApTableStyle */
var ApTableStyle = _react2.default.createClass({
  displayName: 'ApTableStyle',

  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR,
      borderColor: '#CCC'
    };
  },

  statics: {
    styleData: function styleData(config) {
      var highlightColor = config.highlightColor;
      var backgroundColor = config.backgroundColor;
      var borderColor = config.borderColor;

      return {
        all: {
          '.ap-table': {
            borderCollapse: 'collapse',
            width: '100%'
          },
          '.ap-table-cell': {
            padding: '4px 8px',
            border: '1px solid ' + borderColor
          },
          '.ap-table-cell:first-child': {
            borderLeft: 'none'
          },
          '.ap-table-cell:last-child': {
            borderRight: 'none'
          },
          '.ap-table-head .ap-table-row:first-child .ap-table-cell': {
            borderTop: 'none'
          },
          '.ap-table-body:first-child .ap-table-cell': {
            borderTop: 'none'
          },
          '.ap-table-body:last-child .ap-table-cell': {
            borderBottom: 'none'
          },
          '.ap-table-foot:last-child .ap-table-cell': {
            borderBottom: 'none'
          },
          '.ap-table-header-tappable': {
            cursor: 'pointer'
          },
          '.ap-table-header-icon': {
            float: 'right'
          }
        }

      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApTableStyle$styleDa = ApTableStyle.styleData(props);

    var all = _ApTableStyle$styleDa.all;
    var small = _ApTableStyle$styleDa.small;
    var medium = _ApTableStyle$styleDa.medium;
    var large = _ApTableStyle$styleDa.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApTableStyle;
