// apeman-react-table/shim/node/ap_table_body.js
/**
 * Table body component
 * @class ApTableBody
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApTableBody */
var ApTableBody = _react2.default.createClass({
  displayName: 'ApTableBody',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    return _react2.default.createElement(
      'tbody',
      { className: (0, _classnames2.default)('ap-table-body', props.className),
        style: Object.assign({}, props.style) },
      props.children
    );
  }
});

exports.default = ApTableBody;
