// apeman-react-tab/lib/ap_tab_style.jsx
/**
 * Style for ApTab.
 * @class ApTabStyle
 */

'use strict'

import React, {Component, PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApTabStyle */
class ApTabStyle extends Component {
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApTabStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
}

Object.assign(ApTabStyle, {
  propTypes: {
    style: types.object,
    highlightColor: types.string,
    backgroundColor: types.string,
    borderColor: types.string
  },

  defaultProps: {
    style: {},
    highlightColor: ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: ApStyle.DEFAULT_BACKGROUND_COLOR,
    borderColor: '#888'
  },
  styleData (config) {
    let { highlightColor, borderColor, backgroundColor } = config
    return {
      all: {
        '.ap-tab': {
          display: 'flex'
        },
        '.ap-tab-item': {
          display: 'inline-block',
          flex: 1,
          textAlign: 'center',
          padding: '8px',
          color: '#555',
          border: `1px solid ${borderColor}`,
          marginRight: '-1px',
          position: 'relative',
          textDecoration: 'none'
        },
        '.ap-tab-item:hover': {
          boxSizing: 'border-box',
          cursor: 'pointer',
          textDecoration: 'none'
        },
        '.ap-tab-item-selected': {
          color: `${highlightColor}`,
          borderColor: `${highlightColor}`,
          zIndex: 2
        },
        '.ap-tab-item:active': {
          color: `${highlightColor}`,
          borderColor: `${highlightColor}`
        },
        '.ap-tab-group': {
          display: 'block',
          padding: '4px 0 0',
          border: `1px solid ${borderColor}`,
          background: '#EEE'
        },
        '.ap-tab-content': {
          display: 'none',
          borderTop: `1px solid ${borderColor}`,
          marginTop: '-1px',
          position: 'relative',
          zIndex: 2,
          padding: '4px 8px',
          background: backgroundColor
        },
        '.ap-tab-content-selected': {
          display: 'block'
        },
        '.ap-tab-group .ap-tab': {
          margin: '4px 8px 0'
        },
        '.ap-tab-group .ap-tab-item': {
          padding: '4px 0 2px',
          borderBottom: 'none',
          position: 'relative',
          zIndex: 1,
          color: '#888',
          borderColor: 'transparent'
        },
        '.ap-tab-group .ap-tab-item-selected': {
          zIndex: 3,
          background: backgroundColor,
          color: 'inherit',
          borderColor: borderColor
        }
      }
    }
  }
})

export default ApTabStyle
