// apeman-react-info/lib/ap_info.jsx
/**
 * Infor component.
 * @class ApInfo
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'

/** @lends ApInfo */
const ApInfo = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    pullLeft: types.bool,
    pullRight: types.bool
  },

  mixins: [],

  statics: {},

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {}
  },

  render () {
    const s = this
    let { state, props } = s

    return (
      <div className={ classnames('ap-info', {
                'ap-info-left': props.pullLeft,
                'ap-info-right': props.pullRight
            }, props.className) }
           style={ Object.assign({}, props.style) }>
        { props.children }
      </div>
    )
  }
  // --------------------
  // Lifecycle
  // --------------------
  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
})

export default ApInfo
