// apeman-react-draggable/lib/ap_draggable_style.jsx
/**
 * Style for ApDraggable.
 * @augments ApStyle
 * @class ApDraggableStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

const DEFAULT_TRANSITION_DURATION = 80

/** @lends ApDraggableStyle */
const ApDraggableStyle = React.createClass({
  propTypes: {
    style: types.object
  },
  getDefaultProps () {
    return {
      style: {}
    }
  },
  statics: {
    styleData (config) {
      return {
        all: {
          '.ap-draggable': {
            cursor: '-webkit-grab',
            transition: `transform ${DEFAULT_TRANSITION_DURATION}ms`
          },
          '.ap-draggable:active': {
            cursor: '-webkit-grabbing'
          },
          '.ap-draggable.ap-draggable-dragging': {
            boxShadow: '1px 1px 2px rgba(0,0,0,0.25)'
          }
        }
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApDraggableStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApDraggableStyle
