// apeman-react-pager/lib/ap_pager_style.jsx
/**
 * Style for ApPager.
 * @class ApPagerStyle
 */

'use strict'

import React, {Component, PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'
import {mix} from 'acolor'

/** @lends ApPagerStyle */
class ApPagerStyle extends Component {
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApPagerStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
}

Object.assign(ApPagerStyle, {
  propTypes: {
    style: types.object,
    highlightColor: types.string
  },
  defaultProps: {
    style: {},
    highlightColor: ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData (config) {
    let { highlightColor, backgroundColor } = config

    let size = 24

    return {
      all: {
        '.ap-pager': {},
        '.ap-pager-item': {
          display: 'inline-block',
          textAlign: 'center',
          fontSize: '14px',
          marginRight: '-1px',
          color: `${highlightColor}`,
          border: `1px solid ${highlightColor}`,
          background: `${backgroundColor}`
        },
        '.ap-pager-item-inner': {
          display: 'inline-block',
          padding: '4px',
          width: `${size}px`,
          height: `${size}px`,
          lineHeight: `${size}px`
        },
        '.ap-pager-item:hover': {
          textDecoration: 'underline',
          cursor: 'pointer',
          background: `${mix(highlightColor, 'white')}`
        },
        '.ap-pager-item:active': {
          opacity: 0.9
        },
        '.ap-pager-item.ap-pager-item-selected, .ap-pager-item.ap-pager-item-selected:hover, .ap-pager-item.ap-pager-item-selected:active': {
          cursor: 'default',
          opacity: 1,
          background: `${highlightColor}`,
          color: 'white'
        }
      }

    }
  }
})

export default ApPagerStyle
