// apeman-react-container/shim/node/ap_container.js
/**
 * apeman react package for container component.
 * @class ApContainer
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApContainer */
var ApContainer = _react2.default.createClass({
  displayName: 'ApContainer',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Has relative position */
    relative: _react.PropTypes.bool,
    /** With centered style */
    centered: _react.PropTypes.bool
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      relative: false
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;

    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-container', {
          'ap-container-relative': !!props.relative,
          'ap-container-centered': !!props.centered
        }, props.className) },
      props.children
    );
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApContainer;
