// apeman-react-container/shim/node/ap_container_style.js
/**
 * Style for ApContainer.
 * @class ApContainerStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApContainerStyle */
var ApContainerStyle = _react2.default.createClass({
  displayName: 'ApContainerStyle',

  propTypes: {
    style: _react.PropTypes.object,
    width: _react.PropTypes.number
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      width: _apemanReactStyle.ApStyle.CONTAINER_WIDTH
    };
  },

  statics: {
    styleData: function styleData(config) {
      var width = config.width;

      return {
        all: {
          '.ap-container': {
            maxWidth: width + 'px',
            margin: '0 auto',
            display: 'block',
            padding: 0
          },
          '.ap-container-relative': {
            position: 'relative'
          },
          '.ap-container-centered': {
            margin: '0 auto',
            textAlign: 'center'
          }
        },
        small: {},
        medium: {},
        large: {}
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApContainerStyle$sty = ApContainerStyle.styleData(props);

    var all = _ApContainerStyle$sty.all;
    var small = _ApContainerStyle$sty.small;
    var medium = _ApContainerStyle$sty.medium;
    var large = _ApContainerStyle$sty.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApContainerStyle;
