// apeman-react-note/shim/node/ap_note_style.js
/**
 * Style for ApNote.
 * @class ApNoteStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApNoteStyle */
var ApNoteStyle = _react2.default.createClass({
  displayName: 'ApNoteStyle',

  propTypes: {
    style: _react.PropTypes.object,
    color: _react.PropTypes.string,
    maxWidth: _react.PropTypes.number
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      color: '#666',
      maxWidth: _apemanReactStyle.ApStyle.CONTENT_WIDTH
    };
  },

  statics: {
    styleData: function styleData(config) {
      var color = config.color;
      var maxWidth = config.maxWidth;

      return {
        all: {
          '.ap-note': {
            fontSize: '12px',
            color: '' + color,
            maxWidth: maxWidth + 'px'
          },
          '.ap-note a:link, .ap-note a:visited': {
            textDecoration: 'underline',
            color: '' + color,
            cursor: 'pointer'
          },
          '.ap-note a:link:active': {
            opacity: 0.5
          }

        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApNoteStyle$styleDat = ApNoteStyle.styleData(props);

    var all = _ApNoteStyle$styleDat.all;
    var small = _ApNoteStyle$styleDat.small;
    var medium = _ApNoteStyle$styleDat.medium;
    var large = _ApNoteStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApNoteStyle;
