// apeman-react-note/shim/node/ap_note.js
/**
 * Note components.
 * @class ApNote
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApNote */
var ApNote = _react2.default.createClass({
  displayName: 'ApNote',


  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  render: function render() {
    var s = this;
    var props = s.props;


    return _react2.default.createElement(
      'p',
      { className: (0, _classnames2.default)('ap-note', props.className),
        style: Object.assign({}, props.style) },
      props.children
    );
  }
});

exports.default = ApNote;
