// react-flashcard-app/lib/containers/app.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; desc = parent = undefined; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var _redux = require('redux');

var _deck = require('./deck');

var _deck2 = _interopRequireDefault(_deck);

var _componentsTop_control_bar = require('../components/top_control_bar');

var _componentsTop_control_bar2 = _interopRequireDefault(_componentsTop_control_bar);

var _componentsBottom_control_bar = require('../components/bottom_control_bar');

var _componentsBottom_control_bar2 = _interopRequireDefault(_componentsBottom_control_bar);

var _actions = require('../actions');

var _utilsData = require('../utils/data');

require('../../css/flashcards.css');

var App = (function (_Component) {
  _inherits(App, _Component);

  function App() {
    _classCallCheck(this, App);

    _get(Object.getPrototypeOf(App.prototype), 'constructor', this).apply(this, arguments);
  }

  _createClass(App, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      this.props.initApp((0, _utilsData.initialState)(this.props.data));
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      this.props.initApp((0, _utilsData.initialState)(this.props.data));
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2['default'].createElement(
        'div',
        { className: 'flashcard-app' },
        _react2['default'].createElement(
          'div',
          { className: 'row top-row' },
          _react2['default'].createElement(
            'div',
            { className: 'col-md-12' },
            _react2['default'].createElement(_componentsTop_control_bar2['default'], {
              options: {
                buttons: this.props.options.topControlBar,
                buttonTexts: this.props.options.buttonTexts
              }
            })
          )
        ),
        _react2['default'].createElement(
          'div',
          { className: 'row middle-row' },
          _react2['default'].createElement(
            'div',
            { className: 'col-md-12' },
            _react2['default'].createElement(_deck2['default'], null)
          )
        ),
        _react2['default'].createElement(
          'div',
          { className: 'row bottom-row' },
          _react2['default'].createElement(
            'div',
            { className: 'col-md-12' },
            _react2['default'].createElement(_componentsBottom_control_bar2['default'], {
              options: {
                buttons: this.props.options.bottomControlBar,
                buttonTexts: this.props.options.buttonTexts
              }
            })
          )
        )
      );
    }
  }]);

  return App;
})(_react.Component);

App.propTypes = {
  initApp: _propTypes2['default'].func.isRequired,
  data: _propTypes2['default'].shape({
    id: _propTypes2['default'].number,
    name: _propTypes2['default'].string,
    cards: _propTypes2['default'].arrayOf(_propTypes2['default'].string)
  }).isRequired,
  options: _propTypes2['default'].shape({
    topControlBar: _propTypes2['default'].arrayOf(_propTypes2['default'].string),
    bottomControlBar: _propTypes2['default'].arrayOf(_propTypes2['default'].string),
    buttonTexts: _propTypes2['default'].objectOf(_propTypes2['default'].string)
  })
};

App.defaultProps = {
  options: {}
};

var mapStateToProps = function mapStateToProps() {
  return {};
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return (0, _redux.bindActionCreators)({ initApp: _actions.initApp }, dispatch);
};

exports['default'] = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(App);
module.exports = exports['default'];
