// react-flashcard-app/lib/containers/card_face.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactRedux = require('react-redux');

var CardFace = function CardFace(props) {
  return _react2['default'].createElement(
    'div',
    { className: props.className },
    _react2['default'].createElement(
      'div',
      { className: 'card-text' },
      props.text
    ),
    _react2['default'].createElement(
      'div',
      { className: 'info' },
      props.deck.name + ' ' + props.removed + '/' + (props.deck.length - props.removed)
    )
  );
};

CardFace.propTypes = {
  className: _propTypes2['default'].string.isRequired,
  text: _propTypes2['default'].string.isRequired,
  deck: _propTypes2['default'].object.isRequired,
  removed: _propTypes2['default'].number.isRequired
};

var mapStateToProps = function mapStateToProps(state) {
  return {
    deck: state.app.deck,
    removed: state.app.removed.length
  };
};

exports['default'] = (0, _reactRedux.connect)(mapStateToProps)(CardFace);
module.exports = exports['default'];
