// apeman-react-tag/lib/ap_tag_style.jsx
/**
 * Style for ApTag.
 * @class ApTagStyle
 */

'use strict'

import React, { Component, PropTypes as types } from 'react'
import { ApStyle } from 'apeman-react-style'

/** @lends ApTagStyle */
class ApTagStyle extends Component {
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApTagStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
}

Object.assign(ApTagStyle, {
  propTypes: {
    style: types.object,
    height: types.number
  },
  defaultProps: {
    style: {},
    height: 16
  },
  styleData (config) {
    let { height } = config
    let dotSize = 4
    let backgroundColor = '#888'
    return {
      all: {
        '.ap-tag': {
          display: 'inline-block',
          fontSize: '12px',
          padding: '0px 4px 0 0',
          margin: '0 8px 0 4px',
          textAlign: 'center',
          minWidth: '18px',
          lineHeight: `${height}px`,
          color: 'white',
          backgroundColor: `${backgroundColor}`,
          boxSizing: 'border-box',
          position: 'relative'
        },
        '.ap-tag-corner': {
          position: 'absolute',
          display: 'inline-block',
          width: '1px',
          height: 0,
          top: 0,
          left: `${-height / 2}px`,
          borderStyle: 'solid',
          boxSizing: 'border-box',
          borderWidth: `${height / 2}px ${height / 2}px ${height / 2}px 0`,
          borderColor: `transparent ${backgroundColor} transparent transparent`
        },
        '.ap-tag-dot': {
          display: 'inline-block',
          background: 'white',
          width: `${dotSize}px`,
          height: `${dotSize}px`,
          borderRadius: '50%',
          verticalAlign: 'middle',
          position: 'relative',
          top: '-1px',
          left: '-2px'
        },
        '.ap-tag-text': {
          display: 'inline-block',
          margin: '0 2px 0 0'
        }
      }
    }
  }
})

export default ApTagStyle
