// apeman-react-text/shim/node/ap_text_style.js
/**
 * Style for ApText.
 * @class ApTextStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

var _acolor = require('acolor');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApTextStyle */
var ApTextStyle = function (_Component) {
  (0, _inherits3.default)(ApTextStyle, _Component);

  function ApTextStyle() {
    (0, _classCallCheck3.default)(this, ApTextStyle);
    return (0, _possibleConstructorReturn3.default)(this, (ApTextStyle.__proto__ || (0, _getPrototypeOf2.default)(ApTextStyle)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApTextStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApTextStyle$styleDat = ApTextStyle.styleData(props),
          all = _ApTextStyle$styleDat.all,
          small = _ApTextStyle$styleDat.small,
          medium = _ApTextStyle$styleDat.medium,
          large = _ApTextStyle$styleDat.large;

      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: (0, _assign2.default)(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);
  return ApTextStyle;
}(_react.Component);

(0, _assign2.default)(ApTextStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string,
    maxWidth: _react.PropTypes.number
  },

  defaultProps: {
    style: {},
    maxWidth: _apemanReactStyle.ApStyle.CONTENT_WIDTH,
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR
  },

  styleData: function styleData(config) {
    var highlightColor = config.highlightColor,
        maxWidth = config.maxWidth;

    return {
      all: {
        '.ap-text': {
          display: 'block',
          padding: '4px 8px',
          border: '1px solid #AAA',
          width: '100%',
          maxWidth: maxWidth + 'px',
          borderRadius: '2px',
          boxSizing: 'border-box',
          outlineColor: '' + highlightColor,
          boxShadow: '1px 1px 1px rgba(0,0,0,.05) inset'
        },
        '.ap-text-wrap': {
          position: 'relative',
          width: '100%',
          padding: '4px',
          margin: 0,
          verticalAlign: 'middle',
          boxSizing: 'border-box',
          maxWidth: maxWidth + 'px',
          display: 'block'
        },
        '.ap-text-multiple': {
          overflow: 'auto'
        },
        '.ap-text-candidate-list': {
          position: 'absolute',
          left: 0,
          right: '1px',
          top: '100%',
          zIndex: 4,
          padding: '4px 0',
          margin: '0 1px',
          boxShadow: '1px 1px 2px rgba(0,0,0,0.33)',
          background: 'white',
          boxSizing: 'border-box'
        },
        '.ap-text-candidate-list-item': {
          display: 'block',
          padding: 0,
          margin: 0
        },
        '.ap-text-candidate-list-item a': {
          display: 'block',
          padding: '4px 8px'
        },
        '.ap-text-candidate-list-item a:hover': {
          cursor: 'pointer',
          background: '#FAFAFA'
        },
        '.ap-text-candidate-list-item a:active': {
          background: '#F5F5F5'
        },
        '.ap-text-candidate-list-item-selected a': {
          background: (0, _acolor.alpha)(highlightColor, 0.33)
        },
        '.ap-text-candidate-list-item-selected a:hover': {
          background: (0, _acolor.alpha)(highlightColor, 0.5)
        },
        '.ap-text-candidate-list-item-selected a:active': {
          background: (0, _acolor.alpha)(highlightColor, 0.2)
        }
      }
    };
  }
});
exports.default = ApTextStyle;
