// apeman-react-list/shim/node/ap_list_footer.js
/**
 * List footer component.
 * @class ApListFooter
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactSpinner = require('apeman-react-spinner');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApListFooter */
var ApListFooter = _react2.default.createClass({
  displayName: 'ApListFooter',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Show spinner */
    spinning: _react.PropTypes.bool,
    /** Spinner theme */
    spinner: _react.PropTypes.string
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      spinning: false,
      spinner: _apemanReactSpinner.ApSpinner.DEFAULT_THEME
    };
  },
  render: function render() {
    var s = this;
    var state = s.state,
        props = s.props;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-list-footer', props.className),
        style: (0, _assign2.default)({}, props.style) },
      _react2.default.createElement(_apemanReactSpinner.ApSpinner, { enabled: props.spinning,
        className: 'ap-list-footer-spinner'
      }),
      props.children
    );
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

});

exports.default = ApListFooter;
