// apeman-react-list/shim/node/ap_list_more_button.js
/**
 * @class ApListMoreButton
 */
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _asobj = require('asobj');

var _apemanReactTouchable = require('apeman-react-touchable');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ApListMoreButton = _react2.default.createClass({
  displayName: 'ApListMoreButton',

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    text: _react.PropTypes.string,
    enabled: _react.PropTypes.bool
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      text: 'Load More',
      enabled: false
    };
  },
  render: function render() {
    var s = this;
    var state = s.state,
        props = s.props;

    var className = (0, _classnames2.default)('ap-list-more-button', {
      'ap-list-more-button-enabled': props.enabled
    }, props.className);
    return _react2.default.createElement(
      'a',
      { className: className,
        style: (0, _assign2.default)({}, props.style) },
      props.text,
      props.children
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props,
        state = s.state;

    return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
  }
});

exports.default = (0, _apemanReactTouchable.withTouch)(ApListMoreButton);
