// apeman-react-list/shim/node/ap_list_item_image.js
/**
 * Image for list item.
 * @class ApListItemImage
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactImage = require('apeman-react-image');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApListItemImage */
var ApListItemImage = _react2.default.createClass({
  displayName: 'ApListItemImage',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    src: _react.PropTypes.string,
    alt: _react.PropTypes.string,
    scale: _react.PropTypes.oneOf(['fit', 'fill', 'none']),
    width: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
    height: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string])
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      src: null,
      alt: null,
      width: null,
      height: null,
      scale: 'fill'
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;


    return _react2.default.createElement(
      'span',
      { className: (0, _classnames2.default)('ap-list-item-image', props.className) },
      _react2.default.createElement(_apemanReactImage.ApImage, { src: props.src,
        alt: props.alt,
        scale: props.scale,
        width: props.width,
        height: props.height
      })
    );
  }
});

exports.default = ApListItemImage;
