// apeman-react-list/shim/node/ap_list_item.js
/**
 * List item component.
 * @class ApListItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ApListItem = undefined;

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactTouchable = require('apeman-react-touchable');

var _ap_list_item_icon = require('./ap_list_item_icon');

var _ap_list_item_icon2 = _interopRequireDefault(_ap_list_item_icon);

var _ap_list_item_image = require('./ap_list_item_image');

var _ap_list_item_image2 = _interopRequireDefault(_ap_list_item_image);

var _ap_list_item_text = require('./ap_list_item_text');

var _ap_list_item_text2 = _interopRequireDefault(_ap_list_item_text);

var _ap_list_item_arrow_icon = require('./ap_list_item_arrow_icon');

var _ap_list_item_arrow_icon2 = _interopRequireDefault(_ap_list_item_arrow_icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var sizeString = function sizeString(size) {
  return isNaN(Number(size)) ? size : size + 'px';
};

/** @lends ApListItem */

var ApListItem = function (_Component) {
  (0, _inherits3.default)(ApListItem, _Component);

  function ApListItem() {
    (0, _classCallCheck3.default)(this, ApListItem);
    return (0, _possibleConstructorReturn3.default)(this, (ApListItem.__proto__ || (0, _getPrototypeOf2.default)(ApListItem)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApListItem, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;


      var className = (0, _classnames2.default)('ap-list-item', props.className, {
        'ap-list-item-tappable': props.hasTouch
      });

      return _react2.default.createElement(
        'li',
        { className: className
        },
        _react2.default.createElement(
          'a',
          { className: 'ap-list-item-inner',
            href: props.href,
            onClick: function onClick(e) {
              return e.preventDefault();
            }
          },
          _react2.default.createElement(
            'span',
            { className: 'ap-list-item-aligner' },
            '\xA0'
          ),
          props.imgSrc ? _react2.default.createElement(_ap_list_item_image2.default, { className: 'ap-list-item-sumbnail-image',
            src: props.imgSrc,
            alt: props.imgAlt,
            height: props.imgHeight,
            width: props.imgWidth }) : null,
          props.title ? _react2.default.createElement(
            _ap_list_item_text2.default,
            { className: 'ap-list-item-title-wrap' },
            _react2.default.createElement(
              'span',
              { className: 'ap-list-item-title' },
              props.title
            ),
            _react2.default.createElement(
              'span',
              { className: 'ap-list-item-sub-title' },
              props.subTitle
            )
          ) : null,
          props.children,
          props.icon ? _react2.default.createElement(_ap_list_item_icon2.default, { className: props.icon }) : null,
          props.disclosure ? _react2.default.createElement(_ap_list_item_arrow_icon2.default, { className: 'ap-list-item-disclosure-icon' }) : null,
          props.alt ? _react2.default.createElement(
            'div',
            { className: 'ap-list-item-alt' },
            props.alt
          ) : null
        )
      );
    }
  }]);
  return ApListItem;
}(_react.Component);

(0, _assign2.default)(ApListItem, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    disclosure: _react.PropTypes.bool,
    imgSrc: _react.PropTypes.string,
    imgAlt: _react.PropTypes.string,
    imgWidth: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
    imgHeight: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
    /** Title of item */
    title: _react.PropTypes.node,
    /** Sub title of item */
    subTitle: _react.PropTypes.node,
    /** Custom icon class */
    icon: _react.PropTypes.string,
    /** Data for events */
    data: _react.PropTypes.object,
    /** Alt text */
    alt: _react.PropTypes.string
  },

  defaultProps: {
    disclosure: false,
    imgSrc: null,
    imgAlt: null,
    imgWidth: 72,
    imgHeight: 48,
    title: null,
    subTitle: null,
    data: null
  }
});

exports.ApListItem = ApListItem;
exports.default = (0, _apemanReactTouchable.withTouch)(ApListItem, {
  getTouchData: function getTouchData() {
    var s = this;
    var props = s.props;

    return props.data;
  }
});
