// apeman-react-view/lib/ap_view_sidebar.jsx
/**
 * Sidebar of views
 * @class ApViewSidebar
 */
'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import {ApTouchable} from 'apeman-react-touchable'
import ReactCSSTransitionGroup from 'react-addons-css-transition-group'

/** @lends ApViewSidebar */
const ApViewSidebar = React.createClass({
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Shows sidebar */
    enabled: types.bool,
    /** Handler for close */
    onClose: types.func,
    /** Duration for transition */
    transitionTimeout: types.number
  },

  getDefaultProps () {
    return {
      transitionTimeout: 400,
      enabled: false
    }
  },

  render () {
    const s = this
    let { props } = s

    let { enabled, transitionTimeout } = props
    let className = classnames('ap-view-sidebar', props.className, {
      'ap-view-sidebar-enabled': enabled
    })
    return (
      <div className={ className }
           style={ Object.assign({}, props.style) }>
        <ReactCSSTransitionGroup transitionName='ap-view-sidebar-transition'
                                 transitionEnterTimeout={ transitionTimeout }
                                 transitionLeaveTimeout={ transitionTimeout }
        >
          {
            enabled ? (
              <div className='ap-view-sidebar-inner' key='inner'>
                <div className='ap-view-sidebar-content'>
                  { props.children }
                </div>
                <ApTouchable className='ap-view-sidebar-closer'
                             onTap={ props.onClose }
                >
                </ApTouchable>
              </div>
            ) : null
          }
        </ReactCSSTransitionGroup>
      </div>
    )
  }
})

export default ApViewSidebar
