// apeman-react-view/lib/ap_view_body.jsx
/**
 * Body for view.
 * @class ApViewBody
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import {ApContainer} from 'apeman-react-container'

/** @lends ApViewBody */
const ApViewBody = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {},

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {}
  },

  render () {
    const s = this
    let { props } = s

    return (
      <div className={ classnames('ap-view-body', props.className) }
           style={ Object.assign({}, props.style) }>
        <ApContainer className='ap-view-body-inner'>
          { props.children }
        </ApContainer>
      </div>
    )
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
})

export default ApViewBody
