// apeman-react-form/shim/node/ap_form_style.js
/**
 * Style for ApForm.
 * @class ApFormStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApFormStyle */
var ApFormStyle = _react2.default.createClass({
  displayName: 'ApFormStyle',

  propTypes: {
    style: _react.PropTypes.object,
    maxWidth: _react.PropTypes.number
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      errorColor: '#E11',
      errorBackgroundColor: '#FFF0F0'
    };
  },

  statics: {
    styleData: function styleData(config) {
      var errorColor = config.errorColor;
      var errorBackgroundColor = config.errorBackgroundColor;

      return {
        all: {
          '.ap-form': {
            position: 'relative',
            margin: 0
          },
          '.ap-form-centered': {
            margin: '8px auto',
            maxWidth: _apemanReactStyle.ApStyle.CONTENT_WIDTH
          },
          '.ap-form-dummy-input': {
            visibility: 'hidden',
            zIndex: -999,
            display: 'none',
            width: 0,
            height: 0,
            opacity: 0
          },
          '.ap-form-error-list-empty': {
            display: 'none !important'
          },
          '.ap-form-error-list': {
            fontSize: 'smaller',
            paddingLeft: '16px',
            color: errorColor
          },
          '.ap-form-error-list-item': {}
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApFormStyle$styleDat = ApFormStyle.styleData(props);

    var all = _ApFormStyle$styleDat.all;
    var small = _ApFormStyle$styleDat.small;
    var medium = _ApFormStyle$styleDat.medium;
    var large = _ApFormStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApFormStyle;
