// apeman-react-form/shim/node/ap_form.js
/**
 * Form component
 * @class ApForm
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ApForm = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactSpinner = require('apeman-react-spinner');

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApForm */
var ApForm = function (_Component) {
  _inherits(ApForm, _Component);

  function ApForm() {
    _classCallCheck(this, ApForm);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApForm).apply(this, arguments));
  }

  _createClass(ApForm, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;
      var spinner = props.spinner;

      var formProps = (0, _asobj.clone)(props, {
        without: ['centered', 'spinning', 'spinnerTheme', 'spinner', 'l', 'idOf', 'errorColor', 'errorList', 'errorStyle']
      });
      var className = (0, _classnames2.default)('ap-form', props.className, {
        'ap-form-centered': props.centered
      });
      return _react2.default.createElement(
        'form',
        _extends({}, formProps, {
          className: className }),
        _react2.default.createElement('input', { type: 'text',
          name: 'ap_form_dummy',
          disabled: true,
          value: undefined,
          placeholder: 'This is a dummy input to prevent submit on enter',
          className: 'ap-form-dummy-input' }),
        spinner,
        props.children
      );
    }
  }]);

  return ApForm;
}(_react.Component);

Object.assign(ApForm, {
  // --------------------
  // Specs
  // --------------------
  propTypes: {
    id: _react.PropTypes.string.isRequired,
    centered: _react.PropTypes.bool
  },

  defaultProps: {
    centered: false
  }

});

exports.ApForm = ApForm;
exports.default = (0, _apemanReactSpinner.withSpin)(ApForm);
