// apeman-react-card/lib/ap_card.jsx
/**
 * apeman react package for card components
 * @class ApCard
 */

'use strict'

import React, { PropTypes as types, Component } from 'react'
import classnames from 'classnames'
import { ApImage } from 'apeman-react-image'
import { withTouch } from 'apeman-react-touchable'
import { clone } from 'asobj'

/** @lends ApCard */
const ApCard = React.createClass({

  // --------------------
  // Specs
  // --------------------
  render () {
    const s = this
    let { props } = s
    let {
      title,
      children
    } = props
    let divProps = clone(props, {
      without: [ 'title', 'children', 'onTap', 'text', 'hasTouch' ]
    })
    return (
      <div { ...divProps }
           className={ classnames('ap-card', {
             'ap-card-tappable': props.hasTouch
           }, props.className) }
           style={ Object.assign({}, props.style) }>
        <div className='ap-card-inner'>
          <ApCard.Header title={ title }/>
          <ApCard.Body children={ children }/>
        </div>
      </div>
    )
  },
  propTypes: {
    title: types.node
  }
})

Object.assign(ApCard, {
  Header ({ title }) {
    if (!title) {
      return null
    }
    return (
      <div className='ap-card-header'>
        <div className='ap-card-title'>{ title } </div>
      </div>
    )
  },

  Body ({ children }) {
    return (
      <div className='ap-card-body'>
        { children }
      </div>
    )
  }
})

export default withTouch(ApCard)
