// apeman-react-card/lib/ap_card_group.jsx
/**
 * Group components
 * @class ApCardGroup
 */

'use strict'

import React, { Component, PropTypes as types } from 'react'
import classnames from 'classnames'
import {clone} from 'asobj'

/** @lends ApCardGroup */
class ApCardGroup extends Component {
  render () {
    const s = this
    let { state, props } = s
    let divProps = clone(props, {
      without: ['alt', 'empty']
    })
    return (
      <div { ...divProps }
           className={ classnames('ap-card-group', {
             'ap-card-group-empty': !!props.empty
           }, props.className) }
           style={ Object.assign({}, props.style) }>
        { props.children }
        { props.empty ? <ApCardGroup.Alt text={ props.alt }/> : null }
      </div>
    )
  }
}

Object.assign(ApCardGroup, {

  propTypes: {
    /** Mark as empty */
    empty: types.bool,
    /** Alt text to show when empty */
    alt: types.string
  },

  defaultProps: {
    empty: false
  },

  Alt ({ text }) {
    return (
      <div className="ap-card-group-alt">{ text }</div>
    )
  }

})

export default ApCardGroup
