// apeman-react-card/lib/ap_card_style.jsx
/**
 * Style for ApCard.
 * @class ApCardStyle
 */

'use strict'

import React, { PropTypes as types, Component } from 'react'
import { ApStyle } from 'apeman-react-style'

/** @lends ApCardStyle */
class ApCardStyle extends Component {
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApCardStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
}

Object.assign(ApCardStyle, {
  propTypes: {
    style: types.object,
    highlightColor: types.string
  },
  defaultProps: {
    style: {},
    highlightColor: ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData (config) {
    let { highlightColor, backgroundColor } = config
    return {
      all: {
        '.ap-card': {
          margin: '0',
          boxSizing: 'border-box',
          maxWidth: `${ApStyle.CONTENT_WIDTH}px`,
          padding: '8px'
        },
        '.ap-card-inner': {
          display: 'flex',
          padding: '8px',
          borderRadius: '2px',
          boxSizing: 'border-box',
          boxShadow: '2px 2px 4px rgba(0,0,0,0.33)',
          backgroundColor: backgroundColor,
          border: '1px solid #D4D4D4',
          flexWrap: 'wrap',
          width: '100%'
        },
        '.ap-card-tappable:hover .ap-card-inner': {
          cursor: 'pointer'
        },
        '.ap-card-tappable:active .ap-card-inner': {
          boxShadow: 'none'
        },
        '.ap-card-group': {
          display: 'flex',
          flexWrap: 'wrap',
          justifyContent: 'flex-start'
        },

        '.ap-card-group-empty .ap-card-group-alt': {
          display: 'block'
        },
        '.ap-card-group-alt': {
          display: 'none',
          textAlign: 'center',
          padding: '32px 8px',
          color: '#CCC'
        },
        '.ap-card-group .ap-card': {
          width: '50%'
        }
      },
      small: {
        '.ap-card-group .ap-card': {
          width: '33%'
        }
      }
    }
  }
})

export default ApCardStyle
