// apeman-react-radio/shim/node/ap_radio_style.js
/**
 * Style for ApRadio.
 * @class ApRadioStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApRadioStyle */
var ApRadioStyle = function (_Component) {
  _inherits(ApRadioStyle, _Component);

  function ApRadioStyle() {
    _classCallCheck(this, ApRadioStyle);

    return _possibleConstructorReturn(this, (ApRadioStyle.__proto__ || Object.getPrototypeOf(ApRadioStyle)).apply(this, arguments));
  }

  _createClass(ApRadioStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApRadioStyle$styleDa = ApRadioStyle.styleData(props);

      var all = _ApRadioStyle$styleDa.all;
      var small = _ApRadioStyle$styleDa.small;
      var medium = _ApRadioStyle$styleDa.medium;
      var large = _ApRadioStyle$styleDa.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApRadioStyle;
}(_react.Component);

Object.assign(_react.Component, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  defaultProps: {
    scope: false,
    style: {},
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;

    return {
      all: {
        '.ap-radio': {
          display: 'inline-block',
          padding: '0 4px',
          cursor: 'pointer',
          position: 'relative'
        },
        '.ap-radio:hover': {
          opacity: 0.9
        },
        '.ap-radio:active': {
          opacity: 0.75
        },
        '.ap-radio-input': {
          display: 'inline-block',
          padding: '0 2px',
          position: 'relative',
          top: '-2px',
          opacity: 0,
          zIndex: 8
        },
        '.ap-radio-title': {
          display: 'inline-block'
        },
        '.ap-radio-icon': {
          position: 'absolute',
          left: 2,
          top: 0,
          bottom: 0,
          display: 'inline-flex',
          justifyContent: 'center',
          alignItems: 'center'
        },
        '.ap-radio:hover .ap-radio-icon': {
          color: '' + highlightColor
        },
        '.ap-radio:active .ap-radio-icon': {
          opacity: 0.5
        },
        '.ap-radio-icon-checked': {
          color: '' + highlightColor
        }
      }
    };
  }
});

exports.default = ApRadioStyle;
