// apeman-react-radio/shim/node/ap_radio_group.js
/**
 * Radio component.
 * @class ApRadio
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _uuid = require('uuid');

var _uuid2 = _interopRequireDefault(_uuid);

var _ap_radio = require('./ap_radio');

var _ap_radio2 = _interopRequireDefault(_ap_radio);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApRadioGroup */
var ApRadioGroup = _react2.default.createClass({
  displayName: 'ApRadioGroup',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Document id prefix */
    prefix: _react.PropTypes.string,
    /** Name of Radio input */
    name: _react.PropTypes.string.isRequired,
    /** Value and label titles */
    options: _react.PropTypes.object.isRequired,
    /** Checked state for each values */
    checked: _react.PropTypes.object.isRequired,
    /** Handle for change event */
    onChange: _react.PropTypes.func,
    /** Icon class name for normal state */
    icon: _react.PropTypes.string,
    /** Icon class name for checked state */
    checkedIcon: _react.PropTypes.string
  },

  getDefaultProps: function getDefaultProps() {
    return {
      prefix: _uuid2.default.v4(),
      name: null,
      title: '',
      checked: {},
      options: {},
      onChange: null,
      icon: _ap_radio2.default.DEFAULT_ICON,
      checkedIcon: _ap_radio2.default.DEFAULT_CHECKED_ICON
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var prefix = props.prefix;
    var name = props.name;
    var options = props.options;
    var checked = props.checked;
    var icon = props.icon;
    var checkedIcon = props.checkedIcon;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-Radio-group', props.className) },
      Object.keys(options || {}).map(function (value) {
        return _react2.default.createElement(_ap_radio2.default, { key: value,
          name: name,
          value: value,
          id: prefix + '-' + value,
          checked: !!checked[value],
          title: options[value],
          icon: icon,
          checkedIcon: checkedIcon,
          onChange: s.handleChange
        });
      })
    );
  },


  // --------------------
  // Handle
  // --------------------

  handleChange: function handleChange(e) {
    var s = this;
    var props = s.props;

    if (props.onChange) {
      props.onChange(e);
    }
  }
});

exports.default = ApRadioGroup;
