// apeman-react-radio/shim/node/ap_radio.js
/**
 * Radio button component.
 * @class ApRadio
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var DEFAULT_ICON = 'ion ion-ios-circle-outline';
var DEFAULT_CHECKED_ICON = 'ion ion-ios-circle-filled';

/** @lends ApRadio */
var ApRadio = _react2.default.createClass({
  displayName: 'ApRadio',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Element id of the radio box */
    id: _react.PropTypes.string.isRequired,
    /** Name of the radio box */
    name: _react.PropTypes.string.isRequired,
    /** value of the radio box */
    value: _react.PropTypes.string.isRequired,
    /** Title text */
    title: _react.PropTypes.string,
    /** Checked or not */
    checked: _react.PropTypes.bool,
    /** Handler for change event */
    onChange: _react.PropTypes.func,
    /** Class name for icon */
    icon: _react.PropTypes.string,
    /** Class name for checked icon */
    checkedIcon: _react.PropTypes.string
  },

  statics: {
    DEFAULT_ICON: DEFAULT_ICON,
    DEFAULT_CHECKED_ICON: DEFAULT_CHECKED_ICON
  },

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      id: null,
      name: null,
      value: null,
      title: '',
      checked: false,
      onChange: null,
      icon: DEFAULT_ICON,
      checkedIcon: DEFAULT_CHECKED_ICON
    };
  },
  render: function render() {
    var _iconClass;

    var s = this;
    var props = s.props;
    var id = props.id;
    var name = props.name;
    var value = props.value;
    var checked = props.checked;
    var onChange = props.onChange;


    var iconClass = (_iconClass = {
      'ap-radio-icon-checked': checked
    }, _defineProperty(_iconClass, props.checkedIcon, checked), _defineProperty(_iconClass, props.icon, !checked), _iconClass);

    var className = (0, _classnames2.default)('ap-radio', {
      'ap-checkbox-checked': checked
    });
    return _react2.default.createElement(
      'label',
      { className: className,
        htmlFor: id,
        id: id + '-wrap' },
      _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-radio-icon', iconClass) }),
      _react2.default.createElement('input', { type: 'radio',
        className: (0, _classnames2.default)('ap-radio-input'),
        id: id,
        name: name,
        value: value,
        checked: checked,
        onChange: onChange }),
      _react2.default.createElement(
        'span',
        { className: (0, _classnames2.default)('ap-radio-title') },
        props.title
      )
    );
  }
});

exports.default = ApRadio;
