// apeman-react-field/shim/node/ap_field.js
/**
 * Form field.
 * @class ApField
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApField */
var ApField = function (_Component) {
  (0, _inherits3.default)(ApField, _Component);

  function ApField() {
    (0, _classCallCheck3.default)(this, ApField);
    return (0, _possibleConstructorReturn3.default)(this, (ApField.__proto__ || (0, _getPrototypeOf2.default)(ApField)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApField, [{
    key: 'render',

    // --------------------
    // Specs
    // --------------------
    value: function render() {
      var s = this;
      var props = s.props;


      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-field', props.className, {
            'ap-field-align-center': props.center
          }),
          style: (0, _assign2.default)({}, props.style) },
        _react2.default.createElement(
          'div',
          { className: 'ap-field-inner' },
          props.children
        )
      );
    }
  }]);
  return ApField;
}(_react.Component);

(0, _assign2.default)(ApField, {
  // --------------------
  // Specs
  // --------------------
  propTypes: {
    center: _react.PropTypes.bool
  },

  defaultProps: {
    center: false
  }
});

exports.default = ApField;
