// apeman-react-field/shim/node/ap_field_style.js
/**
 * Style for ApField.
 * @class ApFieldStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApFieldStyle */
var ApFieldStyle = function (_Component) {
  (0, _inherits3.default)(ApFieldStyle, _Component);

  function ApFieldStyle() {
    (0, _classCallCheck3.default)(this, ApFieldStyle);
    return (0, _possibleConstructorReturn3.default)(this, (ApFieldStyle.__proto__ || (0, _getPrototypeOf2.default)(ApFieldStyle)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApFieldStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApFieldStyle$styleDa = ApFieldStyle.styleData(props);

      var all = _ApFieldStyle$styleDa.all;
      var small = _ApFieldStyle$styleDa.small;
      var medium = _ApFieldStyle$styleDa.medium;
      var large = _ApFieldStyle$styleDa.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: (0, _assign2.default)(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);
  return ApFieldStyle;
}(_react.Component);

(0, _assign2.default)(ApFieldStyle, {
  propTypes: {
    style: _react.PropTypes.object
  },
  defaultProps: {
    style: {}
  },
  styleData: function styleData(config) {
    var hPadding = 8;
    return {
      all: {
        '.ap-field': {
          display: 'block',
          margin: '4px 0'
        },
        '.ap-field-inner': {
          display: 'block',
          maxWidth: '480px'
        },
        '.ap-field-label': {
          display: 'block',
          fontSize: '14px',
          minWidth: 96,
          padding: 0,
          margin: 0,
          textAlign: 'left',
          boxSizing: 'border-box'
        },
        '.ap-field-label-inner': {
          display: 'block',
          fontSize: '12px',
          padding: '4px ' + hPadding + 'px 0',
          color: '#888',
          boxSizing: 'border-box'
        },
        '.ap-field-value': {
          display: 'block',
          width: '100%',
          padding: '0 ' + hPadding + 'px',
          boxSizing: 'border-box'
        },
        '.ap-field-value .ap-text-wrap, .ap-field-value .ap-password-wrap': {
          margin: 0,
          padding: 0
        },
        '.ap-field-value .ap-text, .ap-field-value .ap-password': {
          width: '100%',
          boxSizing: 'border-box',
          margin: '2px 0'
        },
        '.ap-field-set': {
          borderStyle: 'solid',
          borderRadius: '2px'
        },
        '.ap-field-set-plain': {
          border: 'none',
          padding: 0,
          margin: 0
        },
        '.ap-field-legend': {
          padding: '0 4px'
        },
        '.ap-field-align-center .ap-field-inner': {
          textAlign: 'center',
          display: 'block'
        }
      },
      small: {
        '.ap-field-inner': {
          display: 'flex',
          alignItems: 'center'
        },
        '.ap-field-label': {
          display: 'inline-block',
          padding: 0,
          margin: 0
        },
        '.ap-field-label-inner': {
          display: 'inline-block',
          minWidth: '48px',
          fontSize: 'inherit',
          padding: '8px ' + hPadding + 'px'
        },
        '.ap-field-value': {
          verticalAlign: 'middle',
          padding: '4px 8px'
        }
      }
    };
  }
});

exports.default = ApFieldStyle;
