// apeman-react-field/shim/node/ap_field_legend.js
/**
 * Legend of form field.
 * @class ApFieldLegend
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApFieldLegend */
var ApFieldLegend = function (_Component) {
  (0, _inherits3.default)(ApFieldLegend, _Component);

  function ApFieldLegend() {
    (0, _classCallCheck3.default)(this, ApFieldLegend);
    return (0, _possibleConstructorReturn3.default)(this, (ApFieldLegend.__proto__ || (0, _getPrototypeOf2.default)(ApFieldLegend)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApFieldLegend, [{
    key: 'render',

    // --------------------
    // Specs
    // --------------------
    value: function render() {
      var s = this;
      var props = s.props;


      return _react2.default.createElement(
        'legend',
        (0, _extends3.default)({}, props, {
          className: (0, _classnames2.default)('ap-field-legend', props.className) }),
        props.children
      );
    }

    // --------------------
    // Lifecycle
    // --------------------

  }, {
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      var s = this;
      var props = s.props;
      var state = s.state;

      return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
    }
  }]);
  return ApFieldLegend;
}(_react.Component);

(0, _assign2.default)(ApFieldLegend, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  defaultProps: {}
});

exports.default = ApFieldLegend;
