// apeman-react-radio/lib/ap_radio.jsx
/**
 * Radio button component.
 * @class ApRadio
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import {ApIcon} from 'apeman-react-icon'

const DEFAULT_ICON = 'ion ion-ios-circle-outline'
const DEFAULT_CHECKED_ICON = 'ion ion-ios-circle-filled'

/** @lends ApRadio */
const ApRadio = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Element id of the radio box */
    id: types.string.isRequired,
    /** Name of the radio box */
    name: types.string.isRequired,
    /** value of the radio box */
    value: types.string.isRequired,
    /** Title text */
    title: types.string,
    /** Checked or not */
    checked: types.bool,
    /** Handler for change event */
    onChange: types.func,
    /** Class name for icon */
    icon: types.string,
    /** Class name for checked icon */
    checkedIcon: types.string
  },

  statics: {
    DEFAULT_ICON,
    DEFAULT_CHECKED_ICON
  },

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {
      id: null,
      name: null,
      value: null,
      title: '',
      checked: false,
      onChange: null,
      icon: DEFAULT_ICON,
      checkedIcon: DEFAULT_CHECKED_ICON
    }
  },

  render () {
    const s = this
    let { props } = s

    let {
      id,
      name,
      value,
      checked,
      onChange
    } = props

    let iconClass = {
      'ap-radio-icon-checked': checked,
      [ props.checkedIcon ]: checked,
      [ props.icon ]: !checked
    }

    let className = classnames('ap-radio', {
      'ap-checkbox-checked': checked
    })
    return (
      <label className={ className }
             htmlFor={ id }
             id={`${id}-wrap`}>
        <ApIcon className={ classnames('ap-radio-icon', iconClass) }/>
        <input type="radio"
               className={ classnames('ap-radio-input') }
               id={ id }
               name={ name }
               value={ value }
               checked={ checked }
               onChange={ onChange }/>
        <span className={ classnames('ap-radio-title') }>{ props.title }</span>
      </label>
    )
  }
})

export default ApRadio
