// apeman-react-field/lib/ap_field_style.jsx
/**
 * Style for ApField.
 * @class ApFieldStyle
 */

'use strict'

import React, { Component, PropTypes as types } from 'react'
import { ApStyle } from 'apeman-react-style'

/** @lends ApFieldStyle */
class ApFieldStyle extends Component {
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApFieldStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
}

Object.assign(ApFieldStyle, {
  propTypes: {
    style: types.object
  },
  defaultProps: {
    style: {}
  },
  styleData (config) {
    let hPadding = 8
    return {
      all: {
        '.ap-field': {
          display: 'block',
          margin: '4px 0'
        },
        '.ap-field-inner': {
          display: 'block',
          maxWidth: '480px'
        },
        '.ap-field-label': {
          display: 'block',
          fontSize: '14px',
          minWidth: 96,
          padding: 0,
          margin: 0,
          textAlign: 'left',
          boxSizing: 'border-box'
        },
        '.ap-field-label-inner': {
          display: 'block',
          fontSize: '12px',
          padding: `4px ${hPadding}px 0`,
          color: '#888',
          boxSizing: 'border-box'
        },
        '.ap-field-value': {
          display: 'block',
          width: '100%',
          padding: `0 ${hPadding}px`,
          boxSizing: 'border-box'
        },
        '.ap-field-value .ap-text-wrap, .ap-field-value .ap-password-wrap': {
          margin: 0,
          padding: 0
        },
        '.ap-field-value .ap-text, .ap-field-value .ap-password': {
          width: '100%',
          boxSizing: 'border-box',
          margin: '2px 0'
        },
        '.ap-field-set': {
          borderStyle: 'solid',
          borderRadius: '2px'
        },
        '.ap-field-set-plain': {
          border: 'none',
          padding: 0,
          margin: 0
        },
        '.ap-field-legend': {
          padding: '0 4px'
        },
        '.ap-field-align-center .ap-field-inner': {
          textAlign: 'center',
          display: 'block'
        }
      },
      small: {
        '.ap-field-inner': {
          display: 'flex',
          alignItems: 'center'
        },
        '.ap-field-label': {
          display: 'inline-block',
          padding: 0,
          margin: 0
        },
        '.ap-field-label-inner': {
          display: 'inline-block',
          minWidth: '48px',
          fontSize: 'inherit',
          padding: `8px ${hPadding}px`
        },
        '.ap-field-value': {
          verticalAlign: 'middle',
          padding: '4px 8px'
        }
      }
    }
  }
})

export default ApFieldStyle
