// apeman-react-field/lib/ap_field_label.jsx
/**
 * Label of form field.
 * @class ApFieldLabel
 */

'use strict'

import React, {Component, PropTypes as types} from 'react'
import classnames from 'classnames'
import {shallowEqual} from 'asobj'
import {ApLabel} from 'apeman-react-label'

/** @lends ApFieldLabel */
class ApFieldLabel extends Component {
  // --------------------
  // Specs
  // --------------------
  render () {
    const s = this
    let { props } = s

    return (
      <ApLabel { ...props }
               className={ classnames('ap-field-label', props.className) }
               htmlFor={ props.htmlFor }>
                <span className='ap-field-label-inner'>
                    { props.children }
                </span>
      </ApLabel>
    )
  }

  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate (nextProps, nextState) {
    const s = this
    let { props, state } = s
    return !shallowEqual(props, nextProps) || shallowEqual(state, nextState)
  }
}

Object.assign(ApFieldLabel, {
  // --------------------
  // Specs
  // --------------------
  propTypes: {
    htmlFor: types.string
  },

  defaultProps: {
    htmlFor: null
  }
})

export default ApFieldLabel
