// apeman-react-aside/lib/ap_aside_style.jsx
/**
 * Style for ApAside.
 * @class ApAsideStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApAsideStyle */
const ApAsideStyle = React.createClass({
  propTypes: {
    style: types.object,
    highlightColor: types.string
  },
  getDefaultProps () {
    return {
      style: {},
      highlightColor: ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: ApStyle.DEFAULT_BACKGROUND_COLOR
    }
  },
  statics: {
    styleData (config) {
      let { highlightColor, backgroundColor } = config
      return {
        all: {
          '.ap-aside': {
            display: 'block',
            width: '100%',
            boxSizing: 'border-box',
            padding: '4px 8px',
            border: '1px solid #F0F0F0',
            background: backgroundColor
          },
          '.aside-container': {
            position: 'relative'
          }
        },
        small: {
          '.ap-aside': {
            position: 'absolute',
            right: 0,
            top: 0,
            width: 260,
            zIndex: 4
          }
        }
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApAsideStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApAsideStyle
