// apeman-react-body/lib/ap_body_style.jsx
/**
 * Style for ApBody.
 * @class ApBodyStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApBodyStyle */
const ApBodyStyle = React.createClass({
  propTypes: {
    style: types.object,
    color: types.string
  },
  getDefaultProps () {
    return {
      style: {},
      color: '#555'
    }
  },
  statics: {
    styleData (config) {
      let { color } = config

      return {
        all: {
          '.ap-body': {
            margin: 0,
            padding: 0,
            fontFamily: 'ヒラギノ角ゴ ProN W3, Hiragino Kaku Gothic ProN, メイリオ, Meiryo, sans-serif',
            color: `${color}`
          }
        }
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApBodyStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApBodyStyle
