// apeman-react-view/shim/node/ap_view_footer.js
/**
 * Footer for view.
 * @class ApViewFooter
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactMixinLayout = require('apeman-react-mixin-layout');

var _apemanReactContainer = require('apeman-react-container');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApViewFooter */
var ApViewFooter = _react2.default.createClass({
  displayName: 'ApViewFooter',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Fixed to bottom */
    fixed: _react.PropTypes.bool
  },

  mixins: [_apemanReactMixinLayout.ApLayoutMixin],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      fixed: false
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var layouts = s.layouts;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-view-footer', {
          'ap-view-footer-fixed': props.fixed
        }, props.className),
        style: (0, _assign2.default)({}, props.style) },
      _react2.default.createElement('div', { className: 'ap-view-footer-space',
        ref: 'state',
        style: layouts.space
      }),
      _react2.default.createElement(
        'div',
        { className: 'ap-view-footer-content',
          ref: 'content',
          style: layouts.content
        },
        _react2.default.createElement(
          _apemanReactContainer.ApContainer,
          { className: 'ap-view-footer-inner' },
          props.children
        )
      )
    );
  },

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // For ApLayoutMixin
  // ------------------

  getInitialLayouts: function getInitialLayouts() {
    var s = this;
    return {
      space: {
        height: 'auto'
      },
      content: {}
    };
  },
  calcLayouts: function calcLayouts() {
    var s = this;

    var content = _reactDom2.default.findDOMNode(s.refs['content']);
    var contentRect = content.getBoundingClientRect();
    return {
      space: {
        height: contentRect.height
      },
      content: {}
    };
  }

  // ------------------
  // Private
  // ------------------

});

exports.default = ApViewFooter;
