// apeman-react-view/shim/node/ap_view_header.js
/**
 * Header for view.
 * @class ApViewHeader
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ap_view_header_button = require('./ap_view_header_button');

var _ap_view_header_button2 = _interopRequireDefault(_ap_view_header_button);

var _ap_view_header_title = require('./ap_view_header_title');

var _ap_view_header_title2 = _interopRequireDefault(_ap_view_header_title);

var _apemanReactContainer = require('apeman-react-container');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApViewHeader */
var ApViewHeader = function (_Component) {
  (0, _inherits3.default)(ApViewHeader, _Component);

  function ApViewHeader() {
    (0, _classCallCheck3.default)(this, ApViewHeader);
    return (0, _possibleConstructorReturn3.default)(this, (ApViewHeader.__proto__ || (0, _getPrototypeOf2.default)(ApViewHeader)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApViewHeader, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;


      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-view-header', props.className),
          style: (0, _assign2.default)({}, props.style) },
        _react2.default.createElement(
          _apemanReactContainer.ApContainer,
          { className: 'ap-view-header-inner' },
          _react2.default.createElement(_ap_view_header_button2.default, { icon: props.leftIcon,
            text: props.leftText,
            onTap: props.onLeftTap,
            className: 'ape-view-header-button-left'
          }),
          _react2.default.createElement(
            'div',
            { className: 'ap-view-header-content' },
            _react2.default.createElement(_ap_view_header_title2.default, { icon: props.titleIcon,
              text: props.titleText,
              onTap: props.onTap
            }),
            props.children
          ),
          _react2.default.createElement(_ap_view_header_button2.default, { icon: props.rightIcon,
            text: props.rightText,
            onTap: props.onRightTap,
            className: 'ape-view-header-button-right'
          })
        )
      );
    }
  }]);
  return ApViewHeader;
}(_react.Component);

(0, _assign2.default)(ApViewHeader, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Handler of left button tap */
    onLeftTap: _react.PropTypes.func,
    /** Text of left button */
    leftText: _react.PropTypes.string,
    /** Icon of left button */
    leftIcon: _react.PropTypes.string,
    /** Text of title */
    titleText: _react.PropTypes.string,
    /** Title of icon */
    titleIcon: _react.PropTypes.string,
    /** Text of right button */
    rightText: _react.PropTypes.string,
    /** Icon of right button */
    rightIcon: _react.PropTypes.string,
    /** Handler of right button tap */
    onRightTap: _react.PropTypes.func
  },

  defaultProps: {
    leftText: null,
    leftIcon: null,
    onLeftTap: null,
    titleText: null,
    titleIcon: null,
    rightText: null,
    rightIcon: null,
    onRightTap: null
  },

  // From http://ionicons.com/
  BACK_ICON: 'ion ion-chevron-left',
  NEXT_ICON: 'ion ion-chevron-right',
  ADD_ICON: 'ion ion-plus-round',
  UP_ICON: 'ion ion-chevron-up',
  CLOSE_ICON: 'ion ion-android-close',
  DOWN_ICON: 'ion ion-chevron-down',
  SHARE_ICON: 'ion ion-share',
  COMPOSE_ICON: 'ion ion-compose',
  MENU_ICON: 'ion ion-navicon-round',

  /**
   * Get icon for view pop.
   * @param {string} way - View push way.
   * @returns {string} - Pop icon for a way
   */
  popIcon: function popIcon(way) {
    switch (String(way).toUpperCase()) {
      case 'PUSH_LEFT':
        return ApViewHeader.BACK_ICON;
      default:
        return ApViewHeader.CLOSE_ICON;
    }
  }
});

exports.default = ApViewHeader;
