// apeman-react-view/shim/node/ap_view_sidebar.js
/**
 * Sidebar of views
 * @class ApViewSidebar
 */
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactTouchable = require('apeman-react-touchable');

var _reactAddonsCssTransitionGroup = require('react-addons-css-transition-group');

var _reactAddonsCssTransitionGroup2 = _interopRequireDefault(_reactAddonsCssTransitionGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApViewSidebar */
var ApViewSidebar = _react2.default.createClass({
  displayName: 'ApViewSidebar',

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Shows sidebar */
    enabled: _react.PropTypes.bool,
    /** Handler for close */
    onClose: _react.PropTypes.func,
    /** Duration for transition */
    transitionTimeout: _react.PropTypes.number
  },

  getDefaultProps: function getDefaultProps() {
    return {
      transitionTimeout: 400,
      enabled: false
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var enabled = props.enabled;
    var transitionTimeout = props.transitionTimeout;

    var className = (0, _classnames2.default)('ap-view-sidebar', props.className, {
      'ap-view-sidebar-enabled': enabled
    });
    return _react2.default.createElement(
      'div',
      { className: className,
        style: (0, _assign2.default)({}, props.style) },
      _react2.default.createElement(
        _reactAddonsCssTransitionGroup2.default,
        { transitionName: 'ap-view-sidebar-transition',
          transitionEnterTimeout: transitionTimeout,
          transitionLeaveTimeout: transitionTimeout
        },
        enabled ? _react2.default.createElement(
          'div',
          { className: 'ap-view-sidebar-inner', key: 'inner' },
          _react2.default.createElement(
            'div',
            { className: 'ap-view-sidebar-content' },
            props.children
          ),
          _react2.default.createElement(_apemanReactTouchable.ApTouchable, { className: 'ap-view-sidebar-closer',
            onTap: props.onClose
          })
        ) : null
      )
    );
  }
});

exports.default = ApViewSidebar;
