// apeman-react-view/shim/node/ap_view.js
/**
 * View component.
 * @class ApView
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _stringcase = require('stringcase');

var _asobj = require('asobj');

var _apemanReactSpinner = require('apeman-react-spinner');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var noop = function noop() {
  return undefined;
};

/** @lends ApView */

var ApView = function (_Component) {
  (0, _inherits3.default)(ApView, _Component);

  function ApView() {
    (0, _classCallCheck3.default)(this, ApView);
    return (0, _possibleConstructorReturn3.default)(this, (ApView.__proto__ || (0, _getPrototypeOf2.default)(ApView)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApView, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;
      var way = props.way;

      var className = (0, _classnames2.default)('ap-view', 'ap-view-way-' + (0, _stringcase.spinalcase)(way), {
        'ap-view-wide': !!props.wide,
        'ap-view-headerless': !!props.headerless
      }, props.className);
      var divProps = (0, _asobj.clone)(props, {
        without: ['wide', 'headerless', 'way', 'spinner', 'onMount', 'onUnmount']
      });
      return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, divProps, {
          className: className,
          style: (0, _assign2.default)({}, props.style)
        }),
        _react2.default.createElement(
          'div',
          { className: 'ap-view-inner' },
          props.children,
          props.spinner
        )
      );
    }
  }, {
    key: 'componentDidMount',
    value: function componentDidMount() {
      var s = this;
      var props = s.props;
      var _props$onMount = props.onMount;
      var onMount = _props$onMount === undefined ? noop : _props$onMount;


      onMount();
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      var s = this;
      var props = s.props;
      var _props$onUnmount = props.onUnmount;
      var onUnmount = _props$onUnmount === undefined ? noop : _props$onUnmount;


      onUnmount();
    }
  }]);
  return ApView;
}(_react.Component);

(0, _assign2.default)(ApView, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Wide style */
    wide: _react.PropTypes.bool,
    /** Hide header */
    headerless: _react.PropTypes.bool,
    way: _react.PropTypes.oneOf(['NONE', 'PUSH_LEFT', 'PULL_UP', 'FADE_IN']),
    /** Handle for on mount */
    onMount: _react.PropTypes.func,

    /** Handle for on unmount */
    onUnmount: _react.PropTypes.func
  },

  defaultProps: {
    wide: false,
    way: 'PUSH_LEFT'
  }

});

exports.default = (0, _apemanReactSpinner.withSpin)(ApView);
