// apeman-react-view/shim/node/ap_view_header_button.js
/**
 * Button for header.
 * @class ApViewHeaderButton
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

var _apemanReactTouchable = require('apeman-react-touchable');

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApViewHeaderButton */
var ApViewHeaderButton = _react2.default.createClass({
  displayName: 'ApViewHeaderButton',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Text of button */
    text: _react.PropTypes.string,
    /** Text of icon */
    icon: _react.PropTypes.string
  },

  getDefaultProps: function getDefaultProps() {
    return {
      text: null,
      icon: null
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var text = props.text;
    var icon = props.icon;


    var isEmpty = !(text || icon);
    if (isEmpty) {
      return _react2.default.createElement('span', { className: 'ap-view-header-button-alt' });
    }

    return _react2.default.createElement(
      'a',
      { className: (0, _classnames2.default)('ap-view-header-button', props.className),
        style: (0, _assign2.default)({}, props.style) },
      _react2.default.createElement(
        'span',
        { className: 'ap-view-header-button-inner' },
        _react2.default.createElement(
          'span',
          { className: 'ap-view-header-button-aligner' },
          '&ngbsp;'
        ),
        _react2.default.createElement(_apemanReactIcon.ApIcon, { className: (0, _classnames2.default)('ap-view-header-ion', icon) }),
        _react2.default.createElement(
          'span',
          { className: 'ap-view-header-button-text' },
          text
        ),
        props.children
      )
    );
  },
  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    var props = s.props;
    var state = s.state;

    return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
  }

  // ------------------
  // Custom
  // ------------------

});

exports.default = (0, _apemanReactTouchable.withTouch)(ApViewHeaderButton);
