// apeman-react-view/shim/node/ap_view_stack.js
/**
 * Stack for views.
 * @class ApViewStack
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _reactAddonsCssTransitionGroup = require('react-addons-css-transition-group');

var _reactAddonsCssTransitionGroup2 = _interopRequireDefault(_reactAddonsCssTransitionGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApViewStack */
var ApViewStack = function (_Component) {
  (0, _inherits3.default)(ApViewStack, _Component);

  function ApViewStack() {
    (0, _classCallCheck3.default)(this, ApViewStack);
    return (0, _possibleConstructorReturn3.default)(this, (ApViewStack.__proto__ || (0, _getPrototypeOf2.default)(ApViewStack)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApViewStack, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;
      var transitionTimeout = props.transitionTimeout;
      var fixed = props.fixed;

      var className = (0, _classnames2.default)('ap-view-stack', {
        'ap-view-stack-empty': _react2.default.Children.count(props.children) === 0,
        'ap-view-stack-fixed': fixed
      }, props.className);
      return _react2.default.createElement(
        'div',
        { className: className,
          style: (0, _assign2.default)({}, props.style) },
        _react2.default.createElement(
          _reactAddonsCssTransitionGroup2.default,
          { component: 'div',
            className: 'ap-view-stack-inner',
            transitionName: 'ap-view-stack-transition',
            transitionEnterTimeout: transitionTimeout,
            transitionLeaveTimeout: transitionTimeout
          },

          // Reverse children so that last view coming first
          _react2.default.Children.toArray(props.children).reverse()
        )
      );
    }
  }]);
  return ApViewStack;
}(_react.Component);

(0, _assign2.default)(ApViewStack, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Duration for transition */
    transitionTimeout: _react.PropTypes.number,
    fixed: _react.PropTypes.bool
  },

  defaultProps: {
    transitionTimeout: 400,
    fixed: false
  }
});

exports.default = ApViewStack;
