// apeman-react-view/shim/node/ap_view_style.js
/**
 * Style for ApView.
 * @class ApViewStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApViewStyle */
var ApViewStyle = function (_Component) {
  (0, _inherits3.default)(ApViewStyle, _Component);

  function ApViewStyle() {
    (0, _classCallCheck3.default)(this, ApViewStyle);
    return (0, _possibleConstructorReturn3.default)(this, (ApViewStyle.__proto__ || (0, _getPrototypeOf2.default)(ApViewStyle)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApViewStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApViewStyle$styleDat = ApViewStyle.styleData(props);

      var all = _ApViewStyle$styleDat.all;
      var small = _ApViewStyle$styleDat.small;
      var medium = _ApViewStyle$styleDat.medium;
      var large = _ApViewStyle$styleDat.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: (0, _assign2.default)(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);
  return ApViewStyle;
}(_react.Component);

(0, _assign2.default)(ApViewStyle, {

  propTypes: {
    style: _react.PropTypes.object,
    sidePadding: _react.PropTypes.number,
    headerHeight: _react.PropTypes.number,
    highlightColor: _react.PropTypes.string,
    backgroundColor: _react.PropTypes.string
  },

  defaultProps: {
    style: {},
    sidePadding: 8,
    headerHeight: 44,
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR,
    footerBackgroundColor: _apemanReactStyle.ApStyle.COVER_BACKGROUND
  },
  styleData: function styleData(config) {
    var headerHeight = config.headerHeight;
    var backgroundColor = config.backgroundColor;
    var sidePadding = config.sidePadding;
    var footerBackgroundColor = config.footerBackgroundColor;

    var sidebarRate = 0.8;
    var sideTransitionTimeout = 400;
    return {
      all: {
        '.ap-view': {
          position: 'relative',
          overflow: 'auto'
        },
        '.ap-view-wide .ap-view-body-inner.ap-container': {
          maxWidth: 'none'
        },
        '.ap-view-headerless .ap-view-body': {
          paddingTop: 0
        },
        '.ap-view-headerless .ap-view-header': {
          display: 'none'
        },
        '.ap-view-inner': {
          position: 'absolute',
          left: 0,
          top: 0,
          right: 0,
          bottom: 0
        },
        '.ap-view-header': {
          position: 'absolute',
          left: 0,
          top: 0,
          right: 0,
          height: headerHeight + 'px',
          display: 'block',
          lineHeight: headerHeight + 'px',
          padding: 0,
          margin: 0,
          zIndex: 1,
          boxSizing: 'border-box'
        },
        '.ap-view-header .ap-container': {
          borderBottom: '1px solid #F5F5F5'
        },
        '.ap-view-header-inner': {
          display: 'flex',
          padding: 0,
          margin: '0 auto'
        },
        '.ap-view-header-content': {
          width: '100%',
          flex: 1
        },
        '.ap-view-body': {
          padding: headerHeight + 8 + 'px ' + sidePadding + 'px 8px',
          margin: 0,
          display: 'block',
          backgroundColor: '' + backgroundColor
        },
        '.ap-view-body-inner': {
          margin: '0 auto'
        },
        '.ap-view-header-button': {
          display: 'inline-block',
          cursor: 'pointer',
          textAlign: 'center',
          lineHeight: headerHeight + 'px'
        },
        '.ap-view-header-button:hover': {
          opacity: 0.9
        },
        '.ap-view-header-button:active': {
          opacity: 0.66
        },
        '.ap-view-header-button-alt': {
          display: 'inline-block',
          minWidth: 44
        },
        '.ap-view-header-button-inner': {
          display: 'inline-block',
          padding: '0 4px',
          minWidth: '44px'
        },
        '.ap-view-header-button-aligner': {
          display: 'inline-block',
          width: '1px',
          overflow: 'hidden',
          opacity: 0,
          color: 'transparent',
          marginRight: '-1px',
          lineHeight: headerHeight + 'px',
          height: headerHeight + 'px',
          verticalAlign: 'middle'
        },
        '.ap-view-header-ion': {
          display: 'inline-block',
          lineHeight: headerHeight + 'px',
          verticalAlign: 'middle'
        },
        '.ap-view-header-button-text': {
          lineHeight: headerHeight + 'px',
          verticalAlign: 'middle'
        },
        '.ap-view-header-title': {
          margin: '0 auto',
          display: 'block',
          verticalAlign: 'middle',
          fontWeight: 'normal',
          fontSize: '14px',
          textAlign: 'center'
        },
        '.ap-view-header-title-inner': {
          display: 'inline-flex',
          height: headerHeight + 'px',
          alignItems: 'center'
        },
        '.ap-view-footer': {
          position: 'relative',
          display: 'block',
          width: '100%'
        },
        '.ap-view-footer-space': {
          display: 'none'
        },
        '.ap-view-footer-fixed': {},
        '.ap-view-footer-content': {
          boxSizing: 'border-box'
        },
        '.ap-view-footer-inner': {
          padding: sidePadding + 'px ' + sidePadding + 'px',
          background: '' + footerBackgroundColor,
          borderStyle: 'solid',
          borderWidth: '1px 1px 0',
          borderColor: '#F0F0F0',
          boxSizing: 'border-box'
        },
        '.ap-view-footer-fixed .ap-view-footer-content': {
          position: 'fixed',
          left: 0,
          right: 0,
          bottom: 0,
          zIndex: 4
        },
        '.ap-view-footer-fixed .ap-view-footer-space': {
          display: 'block'
        },
        '.ap-view-sidebar': {
          overflow: 'visible',
          height: 0,
          width: 0,
          position: 'static'
        },
        '.ap-view-sidebar-enabled': {},
        '.ap-view-sidebar-inner': {
          position: 'absolute',
          left: 0,
          top: 0,
          bottom: 0,
          right: 0,
          zIndex: 8,
          display: 'flex',
          justifyContent: 'flex-start',
          background: 'rgba(0,0,0,0.3)',
          boxSizing: 'border-box'
        },
        '.ap-view-sidebar-content': {
          background: backgroundColor,
          width: 100 * sidebarRate + '%',
          height: '100%',
          position: 'relative',
          overflowY: 'auto',
          borderRight: '1px solid #F0F0F0',
          display: 'block',
          boxShadow: '1px 1px 2px rgba(0, 0, 0, 0.1)'
        },
        '.ap-view-sidebar-closer': {
          width: 100 * (1 - sidebarRate) + '%',
          height: '100%',
          display: 'block'
        },
        '.ap-view-sidebar-transition-enter': {
          background: 'rgba(0,0,0,0)'
        },
        '.ap-view-sidebar-transition-enter .ap-view-sidebar-content': {
          left: -100 * sidebarRate + '%'
        },
        '.ap-view-sidebar-transition-enter.ap-view-sidebar-transition-enter-active': {
          background: 'rgba(0,0,0,0.3)',
          transition: 'background ' + sideTransitionTimeout + 'ms'
        },
        '.ap-view-sidebar-transition-enter.ap-view-sidebar-transition-enter-active .ap-view-sidebar-content': {
          left: '0',
          transition: 'left ' + sideTransitionTimeout + 'ms'
        },
        '.ap-view-sidebar-transition-leave': {
          background: 'rgba(0,0,0,0.3)'
        },
        '.ap-view-sidebar-transition-leave .ap-view-sidebar-content': {
          left: '0'
        },
        '.ap-view-sidebar-transition-leave.ap-view-sidebar-transition-leave-active': {
          background: 'rgba(0,0,0,0)',
          transition: 'background ' + sideTransitionTimeout + 'ms'
        },
        '.ap-view-sidebar-transition-leave.ap-view-sidebar-transition-leave-active .ap-view-sidebar-content': {
          left: -100 * sidebarRate + '%',
          transition: 'left ' + sideTransitionTimeout + 'ms'
        }
      }
    };
  }
});

exports.default = ApViewStyle;
