// apeman-react-view/shim/node/ap_view_header_title.js
/**
 * Button for header.
 * @class ApViewHeaderTitle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactIcon = require('apeman-react-icon');

var _apemanReactTouchable = require('apeman-react-touchable');

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApViewHeaderTitle */
var ApViewHeaderTitle = function (_Component) {
  (0, _inherits3.default)(ApViewHeaderTitle, _Component);

  function ApViewHeaderTitle() {
    (0, _classCallCheck3.default)(this, ApViewHeaderTitle);
    return (0, _possibleConstructorReturn3.default)(this, (ApViewHeaderTitle.__proto__ || (0, _getPrototypeOf2.default)(ApViewHeaderTitle)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApViewHeaderTitle, [{
    key: 'shouldComponentUpdate',
    value: function shouldComponentUpdate(nextProps, nextState) {
      var s = this;
      var props = s.props;
      var state = s.state;

      return !(0, _asobj.shallowEqual)(props, nextProps) || (0, _asobj.shallowEqual)(state, nextState);
    }
  }, {
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;
      var text = props.text;
      var icon = props.icon;


      var isEmpty = !(text || icon);
      if (isEmpty) {
        return null;
      }

      return _react2.default.createElement(
        'h3',
        { className: (0, _classnames2.default)('ap-view-header-title', props.className),
          style: (0, _assign2.default)({}, props.style) },
        _react2.default.createElement(
          'span',
          { className: 'ap-view-header-title-inner' },
          _react2.default.createElement(_apemanReactIcon.ApIcon, { className: icon }),
          _react2.default.createElement(
            'span',
            { className: 'ap-view-header-title-text' },
            text
          ),
          props.children
        )
      );
    }
  }]);
  return ApViewHeaderTitle;
}(_react.Component);

(0, _assign2.default)(ApViewHeaderTitle, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Text of the title */
    text: _react.PropTypes.string,
    /** Icon of the title */
    icon: _react.PropTypes.string
  },

  defaultProps: {
    text: null,
    icon: null
  }

});

exports.default = (0, _apemanReactTouchable.withTouch)(ApViewHeaderTitle);
