// apeman-react-aside/shim/node/ap_aside_style.js
/**
 * Style for ApAside.
 * @class ApAsideStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApAsideStyle */
var ApAsideStyle = _react2.default.createClass({
  displayName: 'ApAsideStyle',

  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
    };
  },

  statics: {
    styleData: function styleData(config) {
      var highlightColor = config.highlightColor;
      var backgroundColor = config.backgroundColor;

      return {
        all: {
          '.ap-aside': {
            display: 'block',
            width: '100%',
            boxSizing: 'border-box',
            padding: '4px 8px',
            border: '1px solid #F0F0F0',
            background: backgroundColor
          },
          '.aside-container': {
            position: 'relative'
          }
        },
        small: {
          '.ap-aside': {
            position: 'absolute',
            right: 0,
            top: 0,
            width: 260,
            zIndex: 4
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApAsideStyle$styleDa = ApAsideStyle.styleData(props);

    var all = _ApAsideStyle$styleDa.all;
    var small = _ApAsideStyle$styleDa.small;
    var medium = _ApAsideStyle$styleDa.medium;
    var large = _ApAsideStyle$styleDa.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApAsideStyle;
