// apeman-react-note/lib/ap_note_style.jsx
/**
 * Style for ApNote.
 * @class ApNoteStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApNoteStyle */
const ApNoteStyle = React.createClass({
  propTypes: {
    style: types.object,
    color: types.string,
    maxWidth: types.number
  },
  getDefaultProps () {
    return {
      style: {},
      color: '#666',
      maxWidth: ApStyle.CONTENT_WIDTH
    }
  },
  statics: {
    styleData (config) {
      let { color, maxWidth } = config
      return {
        all: {
          '.ap-note': {
            fontSize: '12px',
            color: `${color}`,
            maxWidth: `${maxWidth}px`
          },
          '.ap-note a:link, .ap-note a:visited': {
            textDecoration: 'underline',
            color: `${color}`,
            cursor: 'pointer'
          },
          '.ap-note a:link:active': {
            opacity: 0.5
          }

        }
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApNoteStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApNoteStyle
