// apeman-react-card/shim/node/ap_card_group.js
/**
 * Group components
 * @class ApCardGroup
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApCardGroup */
var ApCardGroup = function (_Component) {
  (0, _inherits3.default)(ApCardGroup, _Component);

  function ApCardGroup() {
    (0, _classCallCheck3.default)(this, ApCardGroup);
    return (0, _possibleConstructorReturn3.default)(this, (ApCardGroup.__proto__ || (0, _getPrototypeOf2.default)(ApCardGroup)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApCardGroup, [{
    key: 'render',
    value: function render() {
      var s = this;
      var state = s.state;
      var props = s.props;

      var divProps = (0, _asobj.clone)(props, {
        without: ['alt', 'empty']
      });
      return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, divProps, {
          className: (0, _classnames2.default)('ap-card-group', {
            'ap-card-group-empty': !!props.empty
          }, props.className),
          style: (0, _assign2.default)({}, props.style) }),
        props.children,
        props.empty ? _react2.default.createElement(ApCardGroup.Alt, { text: props.alt }) : null
      );
    }
  }]);
  return ApCardGroup;
}(_react.Component);

(0, _assign2.default)(ApCardGroup, {

  propTypes: {
    /** Mark as empty */
    empty: _react.PropTypes.bool,
    /** Alt text to show when empty */
    alt: _react.PropTypes.string
  },

  defaultProps: {
    empty: false
  },

  Alt: function Alt(_ref) {
    var text = _ref.text;

    return _react2.default.createElement(
      'div',
      { className: 'ap-card-group-alt' },
      text
    );
  }
});

exports.default = ApCardGroup;
