// apeman-react-card/shim/node/ap_card_style.js
/**
 * Style for ApCard.
 * @class ApCardStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApCardStyle */
var ApCardStyle = function (_Component) {
  (0, _inherits3.default)(ApCardStyle, _Component);

  function ApCardStyle() {
    (0, _classCallCheck3.default)(this, ApCardStyle);
    return (0, _possibleConstructorReturn3.default)(this, (ApCardStyle.__proto__ || (0, _getPrototypeOf2.default)(ApCardStyle)).apply(this, arguments));
  }

  (0, _createClass3.default)(ApCardStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApCardStyle$styleDat = ApCardStyle.styleData(props);

      var all = _ApCardStyle$styleDat.all;
      var small = _ApCardStyle$styleDat.small;
      var medium = _ApCardStyle$styleDat.medium;
      var large = _ApCardStyle$styleDat.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: (0, _assign2.default)(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);
  return ApCardStyle;
}(_react.Component);

(0, _assign2.default)(ApCardStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  defaultProps: {
    style: {},
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;
    var backgroundColor = config.backgroundColor;

    return {
      all: {
        '.ap-card': {
          margin: '0',
          boxSizing: 'border-box',
          maxWidth: _apemanReactStyle.ApStyle.CONTENT_WIDTH + 'px',
          padding: '8px'
        },
        '.ap-card-inner': {
          display: 'flex',
          padding: '8px',
          borderRadius: '2px',
          boxSizing: 'border-box',
          boxShadow: '2px 2px 4px rgba(0,0,0,0.33)',
          backgroundColor: backgroundColor,
          border: '1px solid #D4D4D4',
          flexWrap: 'wrap',
          width: '100%'
        },
        '.ap-card-tappable:hover .ap-card-inner': {
          cursor: 'pointer'
        },
        '.ap-card-tappable:active .ap-card-inner': {
          boxShadow: 'none'
        },
        '.ap-card-group': {
          display: 'flex',
          flexWrap: 'wrap',
          justifyContent: 'flex-start'
        },

        '.ap-card-group-empty .ap-card-group-alt': {
          display: 'block'
        },
        '.ap-card-group-alt': {
          display: 'none',
          textAlign: 'center',
          padding: '32px 8px',
          color: '#CCC'
        },
        '.ap-card-group .ap-card': {
          width: '50%'
        }
      },
      small: {
        '.ap-card-group .ap-card': {
          width: '33%'
        }
      }
    };
  }
});

exports.default = ApCardStyle;
