// apeman-react-card/shim/node/ap_card.js
/**
 * apeman react package for card components
 * @class ApCard
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactImage = require('apeman-react-image');

var _apemanReactTouchable = require('apeman-react-touchable');

var _asobj = require('asobj');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApCard */
var ApCard = _react2.default.createClass({
  displayName: 'ApCard',


  // --------------------
  // Specs
  // --------------------
  render: function render() {
    var s = this;
    var props = s.props;
    var title = props.title;
    var children = props.children;

    var divProps = (0, _asobj.clone)(props, {
      without: ['title', 'children', 'onTap', 'text', 'hasTouch']
    });
    return _react2.default.createElement(
      'div',
      (0, _extends3.default)({}, divProps, {
        className: (0, _classnames2.default)('ap-card', {
          'ap-card-tappable': props.hasTouch
        }, props.className),
        style: (0, _assign2.default)({}, props.style) }),
      _react2.default.createElement(
        'div',
        { className: 'ap-card-inner' },
        _react2.default.createElement(ApCard.Header, { title: title }),
        _react2.default.createElement(ApCard.Body, { children: children })
      )
    );
  },

  propTypes: {
    title: _react.PropTypes.node
  }
});

(0, _assign2.default)(ApCard, {
  Header: function Header(_ref) {
    var title = _ref.title;

    if (!title) {
      return null;
    }
    return _react2.default.createElement(
      'div',
      { className: 'ap-card-header' },
      _react2.default.createElement(
        'div',
        { className: 'ap-card-title' },
        title,
        ' '
      )
    );
  },
  Body: function Body(_ref2) {
    var children = _ref2.children;

    return _react2.default.createElement(
      'div',
      { className: 'ap-card-body' },
      children
    );
  }
});

exports.default = (0, _apemanReactTouchable.withTouch)(ApCard);
