// apeman-react-toggle/shim/node/ap_toggle.js
/**
 * Toggle item
 * @class ApToggle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ap_toggle_item = require('./ap_toggle_item');

var _ap_toggle_item2 = _interopRequireDefault(_ap_toggle_item);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApToggle */
var ApToggle = function (_Component) {
  _inherits(ApToggle, _Component);

  function ApToggle() {
    _classCallCheck(this, ApToggle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApToggle).apply(this, arguments));
  }

  _createClass(ApToggle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var state = s.state;
      var props = s.props;
      var options = props.options;

      return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)('ap-toggle', props.className),
          style: Object.assign({}, props.style) },
        Object.keys(options || {}).map(function (value) {
          return _react2.default.createElement(
            _ap_toggle_item2.default,
            { active: value === props.value,
              key: value,
              data: value,
              onTap: props.onToggle
            },
            options[value]
          );
        })
      );
    }
  }]);

  return ApToggle;
}(_react.Component);

Object.assign(ApToggle, {
  // --------------------
  // Specs
  // --------------------

  propTypes: {
    options: _react.PropTypes.object.isRequired,
    value: _react.PropTypes.node.isRequired,
    onToggle: _react.PropTypes.func
  },

  defaultProps: {
    options: null,
    value: null,
    onToggle: null
  }
});

exports.default = ApToggle;
