// apeman-react-toggle/shim/node/ap_toggle_style.js
/**
 * Style for ApToggle.
 * @class ApToggleStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApToggleStyle */
var ApToggleStyle = function (_Component) {
  _inherits(ApToggleStyle, _Component);

  function ApToggleStyle() {
    _classCallCheck(this, ApToggleStyle);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(ApToggleStyle).apply(this, arguments));
  }

  _createClass(ApToggleStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApToggleStyle$styleD = ApToggleStyle.styleData(props);

      var all = _ApToggleStyle$styleD.all;
      var small = _ApToggleStyle$styleD.small;
      var medium = _ApToggleStyle$styleD.medium;
      var large = _ApToggleStyle$styleD.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApToggleStyle;
}(_react.Component);

Object.assign(ApToggleStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  defaultProps: {
    style: {},
    highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
    backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
  },
  styleData: function styleData(config) {
    var highlightColor = config.highlightColor;
    var backgroundColor = config.backgroundColor;


    var borderRadius = 4;
    return {
      all: {
        '.ap-toggle': {
          borderRadius: borderRadius + 'px'
        },
        '.ap-toggle-item': {
          color: '' + highlightColor,
          border: '1px solid ' + highlightColor,
          margin: '0 -1px 0 0',
          background: 'transparent'
        },
        '.ap-toggle-item:first-child': {
          borderRadius: borderRadius + 'px 0 0 ' + borderRadius + 'px'
        },
        '.ap-toggle-item:last-child': {
          borderRadius: '0 ' + borderRadius + 'px ' + borderRadius + 'px 0'
        },
        '.ap-toggle-item-active': {
          background: '' + highlightColor,
          color: 'white'
        }
      }
    };
  }
});

exports.default = ApToggleStyle;
