// apeman-react-draggable/shim/node/ap_draggable_style.js
/**
 * Style for ApDraggable.
 * @augments ApStyle
 * @class ApDraggableStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DEFAULT_TRANSITION_DURATION = 80;

/** @lends ApDraggableStyle */
var ApDraggableStyle = _react2.default.createClass({
  displayName: 'ApDraggableStyle',

  propTypes: {
    style: _react.PropTypes.object
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {}
    };
  },

  statics: {
    styleData: function styleData(config) {
      return {
        all: {
          '.ap-draggable': {
            cursor: '-webkit-grab',
            transition: 'transform ' + DEFAULT_TRANSITION_DURATION + 'ms'
          },
          '.ap-draggable:active': {
            cursor: '-webkit-grabbing'
          },
          '.ap-draggable.ap-draggable-dragging': {
            boxShadow: '1px 1px 2px rgba(0,0,0,0.25)'
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApDraggableStyle$sty = ApDraggableStyle.styleData(props);

    var all = _ApDraggableStyle$sty.all;
    var small = _ApDraggableStyle$sty.small;
    var medium = _ApDraggableStyle$sty.medium;
    var large = _ApDraggableStyle$sty.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApDraggableStyle;
