// apeman-react-container/lib/ap_container.jsx
/**
 * apeman react package for container component.
 * @class ApContainer
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'

/** @lends ApContainer */
const ApContainer = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Has relative position */
    relative: types.bool,
    /** With centered style */
    centered: types.bool
  },

  mixins: [],

  statics: {},

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {
      relative: false
    }
  },

  render () {
    const s = this
    let { props } = s
    return (
      <div className={ classnames('ap-container', {
                'ap-container-relative': !!props.relative,
                'ap-container-centered': !!props.centered
            }, props.className) }>
        { props.children }
      </div>
    )
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
})

export default ApContainer
