// apeman-react-container/lib/ap_container_style.jsx
/**
 * Style for ApContainer.
 * @class ApContainerStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApContainerStyle */
const ApContainerStyle = React.createClass({
  propTypes: {
    style: types.object,
    width: types.number
  },
  getDefaultProps () {
    return {
      style: {},
      width: ApStyle.CONTAINER_WIDTH
    }
  },
  statics: {
    styleData (config) {
      let { width } = config
      return {
        all: {
          '.ap-container': {
            maxWidth: `${width}px`,
            margin: '0 auto',
            display: 'block',
            padding: 0
          },
          '.ap-container-relative': {
            position: 'relative'
          },
          '.ap-container-centered': {
            margin: '0 auto',
            textAlign: 'center'
          }
        },
        small: {},
        medium: {},
        large: {}
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApContainerStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApContainerStyle
