// apeman-react-chat/shim/node/ap_chat_style.js
/**
 * Style for ApChat.
 * @class ApChatStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApChatStyle */
var ApChatStyle = _react2.default.createClass({
  displayName: 'ApChatStyle',

  propTypes: {
    style: _react.PropTypes.object,
    highlightColor: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      highlightColor: _apemanReactStyle.ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: _apemanReactStyle.ApStyle.DEFAULT_BACKGROUND_COLOR
    };
  },

  statics: {
    styleData: function styleData(config) {
      var highlightColor = config.highlightColor;
      var backgroundColor = config.backgroundColor;

      var arrowSize = 6;

      var arrowPosition = arrowSize / 2 * -1 + 1.2;
      return {
        all: {
          '.ap-chat': {},
          '.ap-chat-item': {
            display: 'flex',
            justifyContent: 'flex-start',
            flexDirection: 'row'
          },
          '.ap-chat-item-align-left': {},
          '.ap-chat-item-align-right': {
            flexDirection: 'row-reverse'
          },
          '.ap-chat-item-balloon': {
            display: 'block',
            maxWidth: _apemanReactStyle.ApStyle.CONTENT_WIDTH,
            fontSize: 'smaller',
            margin: '8px 0',
            position: 'relative'
          },
          '.ap-chat-item-balloon-content': {
            position: 'relative',
            zIndex: 2,
            background: backgroundColor,
            borderRadius: '16px',
            padding: '4px 8px',
            display: 'block',
            width: '100%',
            boxSizing: 'border-box'
          },
          '.ap-chat-item-balloon-arrow': {
            position: 'absolute',
            boxSizing: 'border-box',
            width: arrowSize,
            height: arrowSize,
            overflow: 'hidden',
            display: 'inline-block',
            transform: 'skewY(30deg)',
            transformOrigin: '50% 50%',
            borderRadius: '0 0 0 100%',
            top: 4,
            background: backgroundColor,
            left: arrowPosition,
            zIndex: 2
          },
          '.ap-chat-item-align-right .ap-chat-item-balloon-arrow': {
            borderRadius: '0 0 100% 0',
            right: arrowPosition,
            transform: 'skewY(-30deg)',
            left: 'auto'
          },
          '.ap-chat-item-info': {
            fontSize: 'x-small',
            boxSizing: 'border-box',
            padding: '8px 4px',
            opacity: 0.5,
            display: 'inline-flex',
            flexDirection: 'column',
            justifyContent: 'flex-start'
          },
          '.ap-chat-item-align-right .ap-chat-item-info': {
            alignItems: 'flex-end'
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApChatStyle$styleDat = ApChatStyle.styleData(props);

    var all = _ApChatStyle$styleDat.all;
    var small = _ApChatStyle$styleDat.small;
    var medium = _ApChatStyle$styleDat.medium;
    var large = _ApChatStyle$styleDat.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApChatStyle;
