// apeman-react-chat/shim/node/ap_chat_item.js
/**
 * Chat item
 * @class ApChatItem
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function formatDate(date) {
  if (!date) {
    return '';
  }
  return String(date);
}

/** @lends ApChatItem */
var ApChatItem = _react2.default.createClass({
  displayName: 'ApChatItem',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    /** Balloon text */
    text: _react.PropTypes.string,
    /** Align direction */
    align: _react.PropTypes.oneOf(['left', 'right']),
    /** Info texts */
    info: _react.PropTypes.oneOfType([_react.PropTypes.array, _react.PropTypes.string])
  },

  mixins: [],

  statics: {},

  getInitialState: function getInitialState() {
    return {};
  },
  getDefaultProps: function getDefaultProps() {
    return {
      text: null,
      align: 'left',
      info: null
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('ap-chat-item', 'ap-chat-item-align-' + props.align, {}, props.className),
        style: Object.assign({}, props.style) },
      s._renderBalloon([].concat(props.text)),
      s._renderInfo(props.info),
      props.children
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentWillMount: function componentWillMount() {
    var s = this;
  },
  componentDidMount: function componentDidMount() {
    var s = this;
  },
  componentWillReceiveProps: function componentWillReceiveProps(nextProps) {
    var s = this;
  },
  shouldComponentUpdate: function shouldComponentUpdate(nextProps, nextState) {
    var s = this;
    return true;
  },
  componentWillUpdate: function componentWillUpdate(nextProps, nextState) {
    var s = this;
  },
  componentDidUpdate: function componentDidUpdate(prevProps, prevState) {
    var s = this;
  },
  componentWillUnmount: function componentWillUnmount() {
    var s = this;
  },


  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------

  _renderInfo: function _renderInfo(info) {
    var s = this;
    if (!info) {
      return null;
    }
    info = [].concat(info);
    return _react2.default.createElement(
      'span',
      { className: 'ap-chat-item-info' },
      info.map(function (info) {
        return _react2.default.createElement(
          'span',
          { key: info },
          info
        );
      })
    );
  },
  _renderBalloon: function _renderBalloon(text) {
    var s = this;
    if (!text) {
      return null;
    }
    return _react2.default.createElement(
      'span',
      { className: 'ap-chat-item-balloon' },
      _react2.default.createElement('span', { className: 'ap-chat-item-balloon-arrow' }),
      _react2.default.createElement(
        'span',
        { className: 'ap-chat-item-balloon-content' },
        text
      )
    );
  }
});

exports.default = ApChatItem;
