// apeman-react-menu/lib/ap_menu.jsx
/**
 * apeman react package for menu components
 * @class ApMenu
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'

/** @lends ApMenu */
const ApMenu = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {},

  mixins: [],

  statics: {
    SIGNOUT_ICON: 'fa fa-sign-out',
    ABOUT_ICON: 'fa fa-info-circle',
    SUPPORT: 'fa fa-life-ring',
    TAG: 'fa fa-tag',
    SETTING: 'fa fa-cog'
  },

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {}
  },

  render () {
    const s = this
    let { props } = s

    let className = classnames('ap-menu', props.className)
    return (
      <div className={ className }
           style={ Object.assign({}, props.style) }>
        { props.children }
        <span className="ap-menu-end">
        </span>
      </div>
    )
  }

  // --------------------
  // Lifecycle
  // --------------------

  // ------------------
  // Custom
  // ------------------

  // ------------------
  // Private
  // ------------------
})

export default ApMenu
