// apeman-react-menu/lib/ap_menu_style.jsx
/**
 * Style for ApMenu.
 * @class ApMenuStyle
 */

'use strict'

import React, {PropTypes as types} from 'react'
import {ApStyle} from 'apeman-react-style'

/** @lends ApMenuStyle */
const ApMenuStyle = React.createClass({
  propTypes: {
    style: types.object,
    highlightColor: types.string
  },
  getDefaultProps () {
    return {
      style: {},
      highlightColor: ApStyle.DEFAULT_HIGHLIGHT_COLOR,
      backgroundColor: ApStyle.DEFAULT_BACKGROUND_COLOR
    }
  },
  statics: {
    styleData (config) {
      let {
        highlightColor,
        backgroundColor
      } = config

      return {
        all: {
          '.ap-menu': {
            display: 'flex',
            flexFlow: 'row wrap',
            justifyContent: 'flex-start',
            maxWidth: `${ApStyle.CONTENT_WIDTH}px`,
            margin: '0 auto'
          },
          '.ap-menu-item': {
            border: 'none',
            backgroundColor: 'transparent',
            minWidth: '86px',
            textAlign: 'center',
            color: 'inherit'
          },
          '.ap-menu-item-hidden': {
            display: 'none !important'
          },
          '.ap-menu-item:hover': {
            cursor: 'pointer',
            opacity: 0.9
          },
          '.ap-menu-item:active': {
            boxShadow: 'none',
            opacity: 0.66
          },
          '.ap-menu-end': {
            flex: 'auto',
            display: 'inline-block',
            width: '0px',
            opacity: 0
          }
        },
        small: {},
        medium: {},
        large: {}
      }
    }
  },
  render () {
    const s = this
    let { props } = s

    let { all, small, medium, large } = ApMenuStyle.styleData(props)

    return (
      <ApStyle data={ Object.assign(all, props.style) }
               smallMediaData={ small }
               mediumMediaData={ medium }
               largeMediaData={ large }
      >{ props.children }</ApStyle>
    )
  }
})

export default ApMenuStyle
