// apeman-react-button/lib/ap_cell_button.jsx
/**
 * Cell button component.
 * @class ApCellButton
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import ApButton from './ap_button'

import {shallowEqual} from 'asobj'

/** @lends ApCellButton */
const ApCellButton = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    disabled: types.bool,
    onTap: types.func,
    text: types.string
  },

  mixins: [
  ],

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {
      disabled: false,
      onTap: null,
      text: null
    }
  },

  render () {
    const s = this
    let { props } = s
    return (
      <ApButton { ...props }
                className={ classnames('ap-cell-button', props.className) }
                wide={ false }
      >
        <span className='ap-cell-button-aligner'>&nbsp;</span>
        <span className='ap-cell-button-text'>{ props.text }</span>
      </ApButton>
    )
  },

  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate (nextProps, nextState) {
    const s = this
    let { props, state } = s
    return !shallowEqual(props, nextProps) || !shallowEqual(state, nextState)
  }
})

export default ApCellButton
