// apeman-react-button/lib/ap_prev_button.jsx
/**
 * Prev button component.
 * @class ApPrevButton
 */

'use strict'

import React, {PropTypes as types} from 'react'
import classnames from 'classnames'
import ApButton from './ap_button'
import {ApIcon} from 'apeman-react-icon'
import {shallowEqual} from 'asobj'

/** @lends ApPrevButton */
const ApPrevButton = React.createClass({

  // --------------------
  // Specs
  // --------------------

  propTypes: {
    disabled: types.bool,
    onTap: types.func,
    text: types.string,
    size: types.number,
    icon: types.string
  },

  mixins: [],

  getInitialState () {
    return {}
  },

  getDefaultProps () {
    return {
      disabled: false,
      onTap: null,
      text: null,
      icon: 'fa fa-caret-left'
    }
  },

  render () {
    const s = this
    let { props } = s
    return (
      <ApButton { ...props }
                className={ classnames('ap-prev-button', props.className) }
                wide={ false }
                style={Object.assign({}, props.style)}
      >
        <ApIcon className={ classnames('ap-prev-button-icon', props.icon) }/>
        <span className='ap-prev-button-text'>
                    { props.text }
                </span>
        { props.children }
      </ApButton>
    )
  },

  // --------------------
  // Lifecycle
  // --------------------

  shouldComponentUpdate (nextProps, nextState) {
    const s = this
    let { props, state } = s
    return !shallowEqual(props, nextProps) || !shallowEqual(state, nextState)
  }

})

export default ApPrevButton
