// apeman-react-tag/shim/node/ap_tag_style.js
/**
 * Style for ApTag.
 * @class ApTagStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** @lends ApTagStyle */
var ApTagStyle = function (_Component) {
  _inherits(ApTagStyle, _Component);

  function ApTagStyle() {
    _classCallCheck(this, ApTagStyle);

    return _possibleConstructorReturn(this, (ApTagStyle.__proto__ || Object.getPrototypeOf(ApTagStyle)).apply(this, arguments));
  }

  _createClass(ApTagStyle, [{
    key: 'render',
    value: function render() {
      var s = this;
      var props = s.props;

      var _ApTagStyle$styleData = ApTagStyle.styleData(props);

      var all = _ApTagStyle$styleData.all;
      var small = _ApTagStyle$styleData.small;
      var medium = _ApTagStyle$styleData.medium;
      var large = _ApTagStyle$styleData.large;


      return _react2.default.createElement(
        _apemanReactStyle.ApStyle,
        { data: Object.assign(all, props.style),
          smallMediaData: small,
          mediumMediaData: medium,
          largeMediaData: large
        },
        props.children
      );
    }
  }]);

  return ApTagStyle;
}(_react.Component);

Object.assign(ApTagStyle, {
  propTypes: {
    style: _react.PropTypes.object,
    height: _react.PropTypes.number
  },
  defaultProps: {
    style: {},
    height: 16
  },
  styleData: function styleData(config) {
    var height = config.height;

    var dotSize = 4;
    var backgroundColor = '#888';
    return {
      all: {
        '.ap-tag': {
          display: 'inline-block',
          fontSize: '12px',
          padding: '0px 4px 0 0',
          margin: '0 8px 0 4px',
          textAlign: 'center',
          minWidth: '18px',
          lineHeight: height + 'px',
          color: 'white',
          backgroundColor: '' + backgroundColor,
          boxSizing: 'border-box',
          position: 'relative'
        },
        '.ap-tag-corner': {
          position: 'absolute',
          display: 'inline-block',
          width: '1px',
          height: 0,
          top: 0,
          left: -height / 2 + 'px',
          borderStyle: 'solid',
          boxSizing: 'border-box',
          borderWidth: height / 2 + 'px ' + height / 2 + 'px ' + height / 2 + 'px 0',
          borderColor: 'transparent ' + backgroundColor + ' transparent transparent'
        },
        '.ap-tag-dot': {
          display: 'inline-block',
          background: 'white',
          width: dotSize + 'px',
          height: dotSize + 'px',
          borderRadius: '50%',
          verticalAlign: 'middle',
          position: 'relative',
          top: '-1px',
          left: '-2px'
        },
        '.ap-tag-text': {
          display: 'inline-block',
          margin: '0 2px 0 0'
        }
      }
    };
  }
});

exports.default = ApTagStyle;
